---
title: "Paralog Annotation Notes"
# output: rmarkdown::github_document
output: 
  word_document:
#    reference_docx: template.docx

editor_options: 
  chunk_output_type: inline
bibliography: bibliography.bib
always_allow_html: yes
---

<!--Load Packages and function-->
```{r echo=FALSE, message=FALSE, warning=FALSE, include=FALSE, results='hide'}
source("src/Paralogous_var_align_functions.R")
```


### Introduction
With the advancements of sequencing technology, new potential variants are being discovered constantly. However to be able to identify said variants as pathogenic or benign requires supporting evidence, which does not always exists especially if the variant novel. 
Previously Ware *et al.* have developed **Paralogue Annotation** [@Ware2012; @Walsh2014], which utilizes information from paralogues (evolutionarily related genes from the same species) to help classify pathogenic variants. They verified its use in LQTS genes on variants acquired from patient cohorts.

Here **Paralogue Annotation** is tested further on a (Likely) Pathogenic/Benign varaint dataset from Clinvar.

<span style="color:red">Also have a look at @Barshir2018 for more info about paralogs in diseases.</span>


### Material and Methods Notes

#### Datasets
Clinvar Likely Pathogenic/Pathogenic and Likely Benign/Benign variant vcf files were extracted and downloaded via the method developed by Zhang *et al.* [@Zhang2017]. 

EFs calculated from OMGL and LMM datasets taken from previous @Walsh2017 publication.

Exac and Gnomad data from @Lek2016. 


#### Statistical terms
In context of is there a pathogenic paralogue alignment? A TP = pathogenic query variant with a paralogous pathogenic hit; FP = benign query variant with a paralogous pathogenic hit; FN = pathogenic query variant with no paralogous pathogenic hit; and TN= benign query variant with no paralogous pathogenic hit.

Likewise for a benign paralogous alignment, a TP = benign query variant with a paralogous benign hit; FP = pathogenic query variant with a paralogous benign hit; FN = benign query variant with no paralogous benign hit; and TN = pathogenic query variant with no paralogous benign hit.

#### Annotation of Clinvar
The Clinvar file __clinvar_20171029.vcf__ was downloaded from ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/. Note that since the initial look at what was available there's been updated Clinvar files. 

NOTE that I have noticed some descrepencies between the plugin annotations which call REFID = 1/0 and that of comparing the REF amino acid by VEP in the dataset to itself. This is due to the fact that the paralogous variant VEP is referring to is simply not in the dataset that I am annotating back to. As a result, it is best to make sure that the ref alleles are indeed the same when processing in R.

The annotataion of the entire clinvar set as of 2018:
```{r echo = FALSE}
#noQC
#pathogenic to pathogenic
p.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# p.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized")
#benign to pathogenic
b.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#stats
conf = conf_matrix(p.paralogous_var_align$num_of_paralog_anno, p.paralogous_var_align$paralog_data, b.paralogous_var_align$num_of_paralog_anno, b.paralogous_var_align$paralog_data)

#para_con
p2.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b2.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.paralogous_var_align$num_of_paralog_anno, p2.paralogous_var_align$paralog_data, b2.paralogous_var_align$num_of_paralog_anno, b2.paralogous_var_align$paralog_data)

#all_con
p3.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b3.paralogous_var_align = Paralogous_var_align("data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.paralogous_var_align$num_of_paralog_anno, p3.paralogous_var_align$paralog_data, b3.paralogous_var_align$num_of_paralog_anno, b3.paralogous_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.paralogous_var_align$Total_paralog_annotations[p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# p4.Total_paralog_annotations = p4.Total_paralog_annotations[!duplicated(p4.Total_paralog_annotations$ID.x),]
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))

#benign to pathogenic
b4.Total_paralog_annotations = b3.paralogous_var_align$Total_paralog_annotations[b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# b4.Total_paralog_annotations = b4.Total_paralog_annotations[!duplicated(b4.Total_paralog_annotations$ID.x),]
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))

conf4 = conf_matrix(p4.num_of_paralog_anno, p3.paralogous_var_align$paralog_data, b4.num_of_paralog_anno, b3.paralogous_var_align$paralog_data)
```


Taking only the 8 sarcomeric genes:
```{r echo = FALSE}
#noQC
p.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf = conf_matrix(p.sarco_var_align$num_of_paralog_anno, p.sarco_var_align$paralog_data, b.sarco_var_align$num_of_paralog_anno, b.sarco_var_align$paralog_data)
#para_con
p2.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b2.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.sarco_var_align$num_of_paralog_anno, p2.sarco_var_align$paralog_data, b2.sarco_var_align$num_of_paralog_anno, b2.sarco_var_align$paralog_data)
#all_con
p3.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b3.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.sarco_var_align$num_of_paralog_anno, p3.sarco_var_align$paralog_data, b3.sarco_var_align$num_of_paralog_anno, b3.sarco_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.sarco_var_align$Total_paralog_annotations[p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# p4.Total_paralog_annotations = p4.Total_paralog_annotations[!duplicated(p4.Total_paralog_annotations$ID.x),]
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))

#benign to pathogenic
b4.Total_paralog_annotations = b3.sarco_var_align$Total_paralog_annotations[b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# b4.Total_paralog_annotations = b4.Total_paralog_annotations[!duplicated(b4.Total_paralog_annotations$ID.x),]
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))

conf4 = conf_matrix(p4.num_of_paralog_anno, p3.sarco_var_align$paralog_data, b4.num_of_paralog_anno, b3.sarco_var_align$paralog_data)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.sarco_var_align$paralog_data$ID),length(b.sarco_var_align$paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),conf$PPV,conf$Sensitivty,conf$Pvalue$p.value)
Variants_remaining_after_PA_QC1 = c(format(conf2$con_table[2], scientific = FALSE),format(conf2$con_table[4], scientific = FALSE),conf2$PPV, conf2$Sensitivity, conf2$Pvalue$p.value)

Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

sarcomeric_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )
```

Using only the 8 sarcomeric genes and joining to the whole clinvar dataset did not provide many annotations which could suggest either PA does not perform well on sarcomeric genes (paralogues to sarcomeric genes are not involed in disease) or that there is a lack of data. Therefore, it is not yet certain that PA does not work on sarcomeric genes and annotataion of additional sarcomeric data is required. See below.

Taking only the 5 channelopathy genes:
```{r echo = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)
#noQC
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data$Variant_pos = paste(p.tableized_data$CHROM,p.tableized_data$POS, sep = " ")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],unlist)
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],function(x) x[1])
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],unlist)
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# p.tableized_data = p.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
p.ref_data = p.tableized_data
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data$Variant_pos = paste(b.tableized_data$CHROM,b.tableized_data$POS, sep = " ")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],unlist)
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],function(x) x[1])
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],unlist)
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# b.tableized_data = b.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
b.ref_data = b.tableized_data
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table_TP = ptop.num_of_paralog_anno
con_table_FP = btop.num_of_paralog_anno
con_table_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table_PPV = con_table_TP/(con_table_TP+con_table_FP)
con_table_Sensitivty = con_table_TP/(con_table_TP+con_table_FN)
con_table = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table) = c("Pathogenic", "Benign")
rownames(con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table_p_value = fisher.test(con_table)

#para_con
#pathogenic to pathogenic
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table2_TP = ptop.num_of_paralog_anno
con_table2_FP = btop.num_of_paralog_anno
con_table2_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table2_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table2_PPV = con_table2_TP/(con_table2_TP+con_table2_FP)
con_table2_Sensitivty = con_table2_TP/(con_table2_TP+con_table2_FN)
con_table2 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table2) = c("Pathogenic", "Benign")
rownames(con_table2) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table2_p_value = fisher.test(con_table2)
#Variants_removed_after_PA_QC1
var_rem_para_con_TP = con_table[2]-con_table2[2]
var_rem_para_con_FP = con_table[4]-con_table2[4]
var_rem_para_con_FN = length(p.paralog_data$ID)-var_rem_para_con_TP
var_rem_para_con_PPV = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FP)
var_rem_para_con_Sensitivity = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FN)
var_rem_para_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_para_con_TP,
    length(b.paralog_data$ID),
    var_rem_para_con_FP
  ), ncol = 2
)
colnames(var_rem_para_con) = c("Pathogenic", "Benign")
rownames(var_rem_para_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_para_con_p_value = fisher.test(var_rem_para_con)

#all_con
#pathogenic to pathogenic
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table3_TP = ptop.num_of_paralog_anno
con_table3_FP = btop.num_of_paralog_anno
con_table3_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table3_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table3_PPV = con_table3_TP/(con_table3_TP+con_table3_FP)
con_table3_Sensitivty = con_table3_TP/(con_table3_TP+con_table3_FN)
con_table3 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table3) = c("Pathogenic", "Benign")
rownames(con_table3) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table3_p_value = fisher.test(con_table3)
#Variants_removed_after_PA_QC2
var_rem_all_con_TP = con_table2[2]-con_table3[2]
var_rem_all_con_FP = con_table2[4]-con_table3[4]
var_rem_all_con_FN = length(p.paralog_data$ID)-var_rem_all_con_TP
var_rem_all_con_PPV = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FP)
var_rem_all_con_Sensitivity = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FN)
var_rem_all_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_all_con_TP,
    length(b.paralog_data$ID),
    var_rem_all_con_FP
  ), ncol = 2
)
colnames(var_rem_all_con) = c("Pathogenic", "Benign")
rownames(var_rem_all_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_all_con_p_value = fisher.test(var_rem_all_con)

#alt_con
#pathogenic to pathogenic
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$ALT_Amino_acids.x==ptop.Total_paralog_annotations$ALT_Amino_acids.y,]
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[!duplicated(ptop.Total_paralog_annotations$ID.x),]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$ALT_Amino_acids.x==btop.Total_paralog_annotations$ALT_Amino_acids.y,]
btop.Total_paralog_annotations = btop.Total_paralog_annotations[!duplicated(btop.Total_paralog_annotations$ID.x),]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table4_TP = ptop.num_of_paralog_anno
con_table4_FP = btop.num_of_paralog_anno
con_table4_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table4_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table4_PPV = con_table4_TP/(con_table4_TP+con_table4_FP)
con_table4_Sensitivty = con_table4_TP/(con_table4_TP+con_table4_FN)
con_table4 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table4) = c("Pathogenic", "Benign")
rownames(con_table4) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table4_p_value = fisher.test(con_table4)
#Variants_removed_after_PA_QC3
var_rem_alt_con_TP = con_table3[2]-con_table4[2]
var_rem_alt_con_FP = con_table3[4]-con_table4[4]
var_rem_alt_con_FN = length(p.paralog_data$ID)-var_rem_alt_con_TP
var_rem_alt_con_PPV = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FP)
var_rem_alt_con_Sensitivity = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FN)
var_rem_alt_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_alt_con_TP,
    length(b.paralog_data$ID),
    var_rem_alt_con_FP
  ), ncol = 2
)
colnames(var_rem_alt_con) = c("Pathogenic", "Benign")
rownames(var_rem_alt_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_alt_con_p_value = fisher.test(var_rem_alt_con)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.paralog_data$ID),length(b.paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(con_table[2], scientific = FALSE),format(con_table[4], scientific = FALSE),con_table2_PPV,con_table_Sensitivty,con_table_p_value$p.value)
Variants_remaining_after_PA_QC1 = c(format(con_table2[2], scientific = FALSE),format(con_table2[4], scientific = FALSE),con_table2_PPV,con_table2_Sensitivty,con_table2_p_value$p.value)
Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

channelopathy_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )

```

On the other hand, channelopathy genes did annotate well suggesting that their paralogues are involved in disease. 

Looking at alt alleles. Taking only pairwise alignments where the alt allele is conserved leaves only 1115 individual pairwise alignments. The number of actual unique variants this equates to is less - 825.

#### Annotation of all possible missense variations in the 8 sarcomeric genes and calculation of EF

Limitations of this current framework are the reliance on 1) genes with paralogues and 2) for those paralogues to have pathogenic variants. This is not always the case, for example in HCM (DOUBLE CHECK THIS CLAIM), where there are not many pathogenic paralogous variants since most paralogues of HCM disease genes are not associated with diesase (according roddy….). Hence we calculated EFs in order to see for those few variants that are predicted to be pathogenic, how often do they appear to be causative of disease in a disease cohort case control study. Segway to HCM validation.

For calculating the EFs, run the all possible missense variants through VEP+plugin and return paraloc locations. Then join those locations with pathogenic clinvar variants as before. This indicates which variants from all possible missense variants are likely to be pathogenic. Then we check to see if any of these variants are present in the cases and controls. Hopefully the controls will be less but there is more control data than cases bare in mind. Calculate the EFs using that. Remember though the EFs are based on how many times an allele is seen, not the number of different alleles by themselves.

```{r echo = FALSE}
#noQC
#all missense data
art.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_art_var = art.paralogous_var_align$Total_paralog_annotations[!is.na(art.paralogous_var_align$Total_paralog_annotations$ID.y),]
#case data
case.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.vcf", sep = "\t", skip = 4)
case.data$Variant_pos = paste(case.data$X.CHROM, case.data$POS, sep = " ")
patho_case.data = case.data[case.data$Variant_pos %in% patho_art_var$Variant_pos,]
#control data
control.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.vcf", sep = "\t", skip = 4)
control.data$Variant_pos = paste(control.data$X.CHROM, control.data$POS, sep = " ")
patho_control.data = control.data[control.data$Variant_pos %in% patho_art_var$Variant_pos,]

#case data VEP
case.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_case_var = case.paralogous_var_align$Total_paralog_annotations[!is.na(case.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_case_var_IDs = patho_case_var$ID.x
# write(patho_case_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_case_var_IDs.txt", sep = "\t", ncolumns = 1)
case.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/HCM_missense_LMM_OMGL_chr.csv", sep = ",")
patho_case.data = case.data[case.data$mut_id %in% patho_case_var_IDs,]
case_cases = sum(patho_case.data$sum.vs_case_count.)
Total_case_cases = sum(case.data$sum.vs_case_count.)

#control data VEP
control.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_control_var = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_control_var_IDs = patho_control_var$ID.x
# write(patho_control_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_control_var_IDs.txt", sep = "\t", ncolumns = 1)
control.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/ExAC_missense_sarcomeric_chr.csv", sep = ",")
patho_control.data = control.data[control.data$mut_id %in% patho_control_var_IDs,]
control_cases = sum(patho_control.data$mut_exac_count)
Total_control_cases = sum(control.data$mut_exac_count)

```


Total cases: 6140
number of affected cases: 39

Total controls: 60678
number of affected controls: 28

Odds ratio 	13.7648
95 % CI:	8.4648 to 22.3833
z statistic	10.570
Significance level	P < 0.0001
Attributable Risk Percent: 92.7% 
95 % CI: 79.6 to 100

#### Paralogue stats
According to ensembl, 92096 protein coding genes are defined to have paralogues. While 7958 protein genes do not have paralogues


#### Plan for overall annotations table
REF              | All | Alt matches |Alt no match
-----------------|-----|-------------|-------------
                 |     |             |
No QC            |     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
Paralog Conserved|     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
All Conserved    |     |             |
                 |     |             |
                 
#### Para-Z scores
For the para-z scores, will need to extract amino acid position from VEP output as well. Then look up the gene in question in para-z score folder, and using the position identify the para-z score. From my understanding, the para-z score is the same across aligned amino acids in the same gene family. Therefore, we could use a cut-off threshold to further improve our confidence in calling variants pathogenic etc. We could also then calculate ROC curves by altering the cut-off to see how that affects sensitivity/PPV.

```{r eval = FALSE, include = FALSE}
Sensitivity_list = vector()
FPR_list = vector()
con_table_list = matrix(, ncol = 2)
for (i in seq(0,11)){
#noQC
#pathogenic to pathogenic
p.paraz_var_align = ParaZ_var_align(i,"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#benign to pathogenic
b.paraz_var_align = ParaZ_var_align(i,"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
paraz.conf = conf_matrix(p.paraz_var_align$num_of_paralog_anno, p.paraz_var_align$paralog_data, b.paraz_var_align$num_of_paralog_anno, b.paraz_var_align$paralog_data)
Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
FPR_list = c(FPR_list, paraz.conf$FPR)
con_table_list = rbind(con_table_list, paraz.conf$con_table)
}
roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list, cutoff = seq(0,11)))
gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
gg = gg + geom_abline(intercept = 0, slope = 1, linetype = "dotted")

gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = cutoff)) + geom_point() + xlim(0,0.25) + ylim(0.7,1)
```

#### References