---
title: "Further applications of Paralog Annotation"
# output: rmarkdown::github_document
output: 

  # word_document:
  #  reference_docx: template.docx

  pdf_document:
    toc: true
    number_sections: true
    fig_caption: yes
    
editor_options: 
  chunk_output_type: inline
bibliography: bibliography.bib
always_allow_html: yes
---

<!--Load Packages and function-->
<!-- ```{r echo=FALSE, message=FALSE, warning=FALSE, include=FALSE, results='hide'} -->
```{r echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
source("src/Paralogous_var_align_functions.R")
```
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.pos = 'H')
```
\newpage
# Introduction
With the advancements of sequencing technology, new potential variants are being discovered constantly. However to be able to identify said variants as being pathogenic requires supporting evidence, which does not always exists especially if the variant is novel. Several commonly used bioinformatic tools for predicting the functional effect of variants, such as SIFT and REVEL [@sim2012sift,@ioannidis2016revel], utilize information from conservation and/or changes in biophysiocochemical properties of the substituting amino acid. However these predictions cannot be considered as sufficient evidence for pathogenicity if predicted as such and need to be used with care, especially in a clinical context, as results typically have low specificity [@flanagan2010using].

Previously Ware *et al.* developed the concept of **Paralogue Annotation** [@Ware2012; @Walsh2014], which utilizes known variant information from paralogues (evolutionarily related genes arising from gene duplication from the same species) to help classify pathogenic variants. They verified its use in LQT syndrome (LQTS) and Brugada syndrome associated genes on variants acquired from patient cohorts.

Here **Paralogue Annotation** is tested further using a Pathogenic/Likely Pathogenic (P/LP) and Benign/Likely Benign (B/LB) varaint dataset acquired from Clinvar. These variants covered a wider range of genes, and were used to demonstrate and validate the application of **Paralogue Annotation** in comparison to common variant interpretation tools used today.

\newpage
# Material and Methods

## Datasets
Clinvar Likely Pathogenic/Pathogenic and Likely Benign/Benign variant vcf files were extracted and downloaded via the method developed by Zhang *et al.* [@Zhang2017]. Variants that were positioned in any of the following 8 sarcomeric: MYH7; MYBPC3; TNNT2; TPM1; MYL2; MYL3; TNNI3; and ACTC1, were also subsetted for further analysis as the case example and used for etiological fraction calculations.

Clinical significance definitions of variants - Pathogenic; Likely Pathogenic; Likely Benign; Benign - were defined according to Clinvar [@landrum2015clinvar]. Variants of unknown significance were not used in the study.

For the case and control study, HCM diseased cohorts data were gathered from the OMGL and LMM datasets, which were taken from previous @Walsh2017 publication. Data from ExAC were used as the control cohort @Lek2016.

## Annotation of variants and transfer of annotations across paralogues

The data was prepared by removing frameshift mutations

## Statistical measures
For a pathogenic paralogue alignment:

true positive (TP) = pathogenic query variant with a paralogous pathogenic hit;

false positive (FP) = benign query variant with a paralogous pathogenic hit;

false negative (FN) = pathogenic query variant with no paralogous pathogenic hit;

and true negative (TN) = benign query variant with no paralogous pathogenic hit.\newline


Likewise for a benign paralogous alignment:

TP = benign query variant with a paralogous benign hit; 

FP = pathogenic query variant with a paralogous benign hit; 

FN = benign query variant with no paralogous benign hit; 

and TN = pathogenic query variant with no paralogous benign hit.\newline

Positive Predictive Values (PPV) and Sensitivties are therefore calculated by:

$$PPV = \frac{TP}{TP+FP}$$
$$Sensitivity = \frac{TP}{TP+FN}$$

P values are calculated via a Fisher's exact test on a 2x2 contingency table tabulating the number of pathogenic and benign variants of interest and how many of those are predicted as pathogenic or benign.

## Calculation of Etiological Fractions

Odds ratios (OR) are calculated by:

$$ OR =  \frac{(a/b)}{(c/d)} $$

where:

$a =$ number of variants predicted to be pathogenic by paralogue annotation in the diseased cohort

$b =$ number of varaints predicted to be pathogenic by paralogue annotation in the control cohort

$c =$ number of variants not predicted to be pathogenic by paralogue annotation in the diseased cohort

$d =$ number of variants not predicted to be pathogenic by paralogue annotation in the control cohort

and the 95% confidence intervals for OR values are calculated according to @altman1991practical via:

$$ 95\% \ CI = [e^{\ln(OR) - 1.96 \cdot SE(\ln(OR))},e^{\ln(OR) + 1.96 \cdot SE(\ln(OR))}] $$
where:

$$ SE(\ln(OR)) =  \sqrt{\frac{1}{a}+\frac{1}{b}+\frac{1}{c}+\frac{1}{d}} $$

Etiological Fractions (EF) can then be calculated by:

$$ EF = \frac{OR - 1}{OR} $$
and the 95% confidence intervals for EF values are calculated according to @hildebrandt2006calculating via:

$$ 95\% \ CI = [\widehat{EF} - z_{1-\frac{a}{2}} \sqrt{\widehat{VAR}(\widehat{\Phi})}, \ min( \widehat{EF} + z_{1-\frac{a}{2}} \sqrt{\widehat{VAR}(\widehat{\Phi})}, \ 1)]  $$
where the asymptomatic variance estimator of $\widehat{\Phi}$ is given by:

$$ \widehat{VAR}(\widehat{\Phi}) = \widehat{\Phi}^2 [\frac{1-\widehat{\pi}_{0}}{N_0\widehat{\pi}_{0}}+\frac{1-\widehat{\pi}_{1}}{N_1\widehat{\pi}_{1}}]$$
and:
$\widehat{\pi}_{1} = a/(b + a)$
$\widehat{\pi}_{0} = c/(d + c)$ 
$N_0 = c + d$ 
$N_1 = a + b$ 
$\widehat{\Phi}^2 = (\widehat{\pi}_{0}/\widehat{\pi}_{1})^2$

NOTE ACCORDING TO https://www.medcalc.org/calc/odds_ratio.php - "Where zeros cause problems with computation of the odds ratio or its standard error, 0.5 is added to all cells (a, b, c, d) (Pagano & Gauvreau, 2000; Deeks & Higgins, 2010)."

## Paralogue stats
The additional statistics were calculated by programmatically extracting the genes of interest (using `src/check_what_clinvar_genes.py` and `src/Find_unique_genes.py`) and then retrieving relevant information manually from [Ensembl's Bioimart](https://www.ensembl.org/biomart)

Alternatively, this can be reproduced using biomaRt package
                 
## Para-Z scores
For the para-z scores, will need to extract amino acid position from VEP output as well. Then look up the gene in question in para-z score folder, and using the position identify the para-z score. From my understanding, the para-z score is the same across aligned amino acids in the same gene family. Therefore, we could use a cut-off threshold to further improve our confidence in calling variants pathogenic etc. We could also then calculate ROC curves by altering the cut-off to see how that affects sensitivity/PPV.

All available para-z scores were retreived from https://git-r3lab.uni.lu/genomeanalysis/paralogs/tree/master/data [@Lal2017]. Para-Z score cutoff thresholds were used to remove any annotation alignments in question. Amino Acid positions that had a para-z score below the chosen cutoff threshold were not used for annotations.

\newpage

## Comparison to SIFT/REVEL scores
SIFT and Mcap scores were retreived using ANNOVAR with dbnsfp30a [@wang2010annovar,@liu2016dbnsfp]. Revel scores were downloaded from https://sites.google.com/site/revelgenomics/downloads [@ioannidis2016revel].

Revel did not provide scores for all available variants in the dataset and therefore to ensure the same variants were being annotated by all tools

# Results and Discussion

## Paralogue stats

```{r paralog_dist, cache=TRUE, echo=FALSE, message=FALSE, warning=FALSE, fig.width=7,fig.height=6,fig.cap="\\label{fig:paralog_dist}Distribution of genes with paralogues by the number of paralogues they're related to"}
ensembl = useMart("ENSEMBL_MART_ENSEMBL", host = "jul2018.archive.ensembl.org")
ens_human <- useDataset("hsapiens_gene_ensembl",mart=ensembl)
protein_genes_w_paralogues = getBM(attributes = c("external_gene_name", "hsapiens_paralog_associated_gene_name"), filters = c("biotype", "with_hsapiens_paralog"), values = list("protein_coding", TRUE), mart = ens_human)
protein_genes_w_paralogues_genes = as.vector(unique(protein_genes_w_paralogues$external_gene_name))

# eg = bitr(protein_genes_w_paralogues_genes, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Hs.eg.db")
# ego = groupGO(gene = eg$ENTREZID, OrgDb = org.Hs.eg.db, keyType = "ENTREZID", readable = TRUE, ont = "MF", level = 2)
# barplot(ego)

# length(protein_genes_w_paralogues)
protein_genes_w_paralogues_wide = plyr::count(protein_genes_w_paralogues, "external_gene_name")
protein_genes_w_paralogues_wide$paralogue = "Yes"
colnames(protein_genes_w_paralogues_wide)[2] = "number_of_paralogues"
par(mfrow=c(2,2))
# hist(protein_genes_w_paralogues_wide$number_of_paralogues, breaks = seq(0,50,1), xlab = "number of paralogues", ylab = "number of genes", main="A) Overall distribution of genes with paralogues", cex.main = 0.7)

protein_genes_wo_paralogues = getBM(attributes = c("external_gene_name", "hsapiens_paralog_associated_gene_name"), filters = c("biotype", "with_hsapiens_paralog"), values = list("protein_coding", FALSE), mart = ens_human)
protein_genes_wo_paralogues = protein_genes_wo_paralogues[!(protein_genes_wo_paralogues$external_gene_name %in% protein_genes_w_paralogues_genes),]
protein_genes_wo_paralogues_genes = as.vector(unique(protein_genes_wo_paralogues$external_gene_name))

# length(protein_genes_wo_paralogues)
protein_genes_wo_paralogues_wide = protein_genes_wo_paralogues
protein_genes_wo_paralogues_wide[is.na(protein_genes_wo_paralogues_wide)] = 0
protein_genes_wo_paralogues_wide$paralogue = "No"
colnames(protein_genes_wo_paralogues_wide)[2] = "number_of_paralogues"
protein_genes_wo_paralogues_wide = protein_genes_wo_paralogues_wide[!(protein_genes_wo_paralogues_wide$external_gene_name %in% protein_genes_w_paralogues_wide$external_gene_name),]


#testing to see how many genes in the clinvar dataset have paralogues
clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
### To only consider snv missense ###
# clinvar_P_LP_tableized = clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & 
#                                                   !(grepl("-", clinvar_P_LP_tableized$Protein_position)) &
#                                                   grepl("/", clinvar_P_LP_tableized$Amino_acids) &
#                                                   str_count(clinvar_P_LP_tableized$Amino_acids) == 3,]
####
clinvar_P_LP_tableized_genes = as.vector(unique(clinvar_P_LP_tableized$SYMBOL))
clinvar_P_LP_vcf_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_vcf_tableized_genes = as.vector(unique(clinvar_P_LP_vcf_tableized$GENEINFO))
clinvar_P_LP_vcf_tableized_genes = unique(unlist(sapply(clinvar_P_LP_vcf_tableized_genes, strsplit, "\\|")))
clinvar_P_LP_vcf_tableized_genes = sapply(strsplit(clinvar_P_LP_vcf_tableized_genes, "\\:"), "[", 1) #counts genes that overlap as single gene #4494

clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
### To only consider snv missense ###
# clinvar_B_LB_tableized = clinvar_B_LB_tableized[!is.na(clinvar_B_LB_tableized$Protein_position) & 
#                                                   !(grepl("-", clinvar_B_LB_tableized$Protein_position)) &
#                                                   grepl("/", clinvar_B_LB_tableized$Amino_acids) &
#                                                   str_count(clinvar_B_LB_tableized$Amino_acids) == 3,]
####
clinvar_B_LB_tableized_genes = as.vector(unique(clinvar_B_LB_tableized$SYMBOL))
clinvar_B_LB_vcf_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_vcf_tableized_genes = as.vector(unique(clinvar_B_LB_vcf_tableized$GENEINFO))
clinvar_B_LB_vcf_tableized_genes = unique(unlist(sapply(clinvar_B_LB_vcf_tableized_genes, strsplit, "\\|")))
clinvar_B_LB_vcf_tableized_genes = sapply(strsplit(clinvar_B_LB_vcf_tableized_genes, "\\:"), "[", 1) #counts genes that overlap as single gene 

#total clinvar genes in both P/LP and B/LB
all_clinvar_genes = unique(c(clinvar_P_LP_tableized_genes, clinvar_B_LB_tableized_genes)) #6284 here
write.table(all_clinvar_genes,file="data/clinvar/6284_all_clinvar_genes.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)


#ALL clinvar genes including VUS, not just P/LP and B/LB
full_clinvar_tableized = read.csv(file="data/clinvar/clinvar_20190114.vcf_tableized_org", sep = "\t", header = TRUE, stringsAsFactors = FALSE)

overlapped_genes = vector()
non_overlapped_genes = vector()
for (x in unique(full_clinvar_tableized$GENEINFO)){
  if (grepl("\\|", x)){
    overlapped_genes = c(overlapped_genes, x)
  } else {
    non_overlapped_genes = c(non_overlapped_genes, x)
  }
}
overlapped_genes = unique(unlist(sapply(overlapped_genes, strsplit, "\\|")))
overlapped_genes = sapply(strsplit(overlapped_genes, "\\:"), "[", 1)
non_overlapped_genes = unique(unlist(sapply(non_overlapped_genes, strsplit, "\\|")))
non_overlapped_genes = sapply(strsplit(non_overlapped_genes, "\\:"), "[", 1)
full_genes = unique(c(overlapped_genes, non_overlapped_genes)) 
full_genes_w_paralogues = intersect(full_genes, protein_genes_w_paralogues_genes)


# Total protein genes
all_protein_genes = dplyr::bind_rows(protein_genes_wo_paralogues_wide, protein_genes_w_paralogues_wide)
his = ggplot(all_protein_genes, aes(x=number_of_paralogues, fill=paralogue)) + geom_histogram(binwidth = 1, colour = "black") + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    plot.title = element_text(hjust = 0.5)) +labs(title = "Overall distribution of human protein coding genes in Ensembl and their associated paralogues", 
    x = "number of paralogues", y = "number of genes", 
    fill = "Has paralogues")+labs(title = "Overall distribution of human protein coding
genes in Ensembl and their associated paralogues")+ylim(0,6000) + theme(axis.title = element_text(size = 15), 
    axis.text = element_text(size = 15), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 12))
ggsave("images/Total_genes_paralogs.png")
# Subset disease genes only
disease_genes = vector()
# for (gene in clinvar_P_LP_vcf_tableized_genes){
for (gene in clinvar_P_LP_tableized_genes){
  # print(gene)
  if (gene %in% all_protein_genes$external_gene_name){
    disease_genes = c(gene, disease_genes)
  }
}
# Disease genes
disease_genes_wide = all_protein_genes[all_protein_genes$external_gene_name %in% disease_genes,]
his2 = ggplot(disease_genes_wide, aes(x=number_of_paralogues, fill=paralogue)) + geom_histogram(binwidth = 1, colour = "black") + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    plot.title = element_text(hjust = 0.5)) +labs(title = "Distribution of human protein coding genes involved in disease in Ensembl and their associated paralogues", 
    x = "number of paralogues", y = "number of genes", 
    fill = "Has paralogues")+labs(title = "Distribution of human protein coding genes involved
in disease in Ensembl and their associated paralogues") + theme(axis.title = element_text(size = 15), 
    axis.text = element_text(size = 15), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 12))
ggsave("images/Disease_genes_paralogs.png")

#combined total protein and disease gene dsitributions

disease_genes_wide$disease = "Yes"
all_protein_genes$disease = "No"

total_plot_data = rbind(all_protein_genes, disease_genes_wide)
total_plot_data[total_plot_data$disease == "No" & total_plot_data$paralogue == "No", "colour"] = "no_dis_no_para"
total_plot_data[total_plot_data$disease == "No" & total_plot_data$paralogue == "Yes", "colour"] = "no_dis_yes_para"
total_plot_data[total_plot_data$disease == "Yes" & total_plot_data$paralogue == "No", "colour"] = "yes_dis_no_para"
total_plot_data[total_plot_data$disease == "Yes" & total_plot_data$paralogue == "Yes", "colour"] = "yes_dis_yes_para"


cols = c("no_dis_no_para" = "purple", "yes_dis_no_para" = "orange", "no_dis_yes_para" = "blue", "yes_dis_yes_para" = "red")
dens = ggplot(total_plot_data, aes(x=number_of_paralogues)) + 
  geom_histogram(data = total_plot_data[total_plot_data$disease=="No" & total_plot_data$paralogue == "No",], aes(fill = "no_dis_no_para"), alpha = 0.4, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="Yes" & total_plot_data$paralogue == "No",], aes(fill = "yes_dis_no_para"), alpha = 0.9, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="No" & total_plot_data$paralogue == "Yes",], aes(fill = "no_dis_yes_para"), alpha = 0.4, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="Yes" & total_plot_data$paralogue == "Yes",], aes(fill = "yes_dis_yes_para"), alpha = 0.9, binwidth = 1) +
  scale_fill_manual(values = cols, labels = c("All genes,\nno paralogues (n=5644)", "All genes,\nhas paralogues (n=14514)", "Clinvar disease genes,\nno paralogues (n=1496)",  "Clinvar disease genes,\nhas paralogues (n=3621)")) +
  # scale_y_continuous(expand = c(0,0)) +
  # scale_x_continuous(expand = c(0,0)) +
  theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    plot.title = element_text(hjust = 0.5)) +
  labs(x = "number of paralogues", y = "number of genes", fill = "") + 
    # labs(title = "Distribution of human protein coding genes involved in disease in Ensembl and their associated paralogues") + 
  theme(axis.title = element_text(size = 15), 
    axis.text = element_text(size = 15), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 12),
    legend.position = c(0.875,0.75)) +
  labs(fill = NULL) + 
  guides(fill=guide_legend(
                 keywidth=0.2,
                 keyheight=0.5,
                 default.unit="inch")
      )
ggsave("images/Total_and_Disease_genes_paralogs.png", height = 5, width = 10)

dens = ggplot(total_plot_data, aes(x=number_of_paralogues)) + 
  geom_histogram(data = total_plot_data[total_plot_data$disease=="No" & total_plot_data$paralogue == "No",], aes(fill = "no_dis_no_para"), alpha = 0.4, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="Yes" & total_plot_data$paralogue == "No",], aes(fill = "yes_dis_no_para"), alpha = 0.9, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="No" & total_plot_data$paralogue == "Yes",], aes(fill = "no_dis_yes_para"), alpha = 0.4, binwidth = 1) +
  geom_histogram(data = total_plot_data[total_plot_data$disease=="Yes" & total_plot_data$paralogue == "Yes",], aes(fill = "yes_dis_yes_para"), alpha = 0.9, binwidth = 1) +
  scale_fill_manual(values = cols, labels = c("All genes,\nno paralogues (n=5644)", "All genes,\nhas paralogues (n=14514)", "Disease genes,\nno paralogues (n=1496)",  "Disease genes,\nhas paralogues (n=3621)")) +
  scale_y_continuous(limits = c(0,6000)) +
  # scale_x_continuous(expand = c(0,0.5)) +
  theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    plot.title = element_text(hjust = 0.5)) +
  labs(x = "number of paralogues", y = "number of genes", fill = "") + 
    # labs(title = "Distribution of human protein coding genes involved in disease in Ensembl and their associated paralogues") + 
  theme(axis.title = element_text(size = 25), 
    axis.text = element_text(size = 25), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 17), 
    legend.title = element_text(size = 12),
    legend.position = c(0.825,0.675)) +
  labs(fill = NULL) + 
  guides(fill=guide_legend(
                 keywidth=0.2,
                 keyheight=0.7,
                 default.unit="inch")
      )
ggsave("images/Total_and_Disease_genes_paralogs_for_poster_big.png", height = 5, width = 10)

# dens2 = ggplot(disease_genes_wide, aes(x=number_of_paralogues)) + geom_density()

#see if Patho and Benign sets have same distribution of variants in genes with paralogues
clinvar_P_LP_genes_w_paralogs = vector()
clinvar_P_LP_genes_wo_paralogs = vector()
clinvar_B_LB_genes_w_paralogs = vector()
clinvar_B_LB_genes_wo_paralogs= vector()
for (gene in clinvar_P_LP_tableized_genes){
  # print(gene)
  if (gene %in% protein_genes_w_paralogues_genes){
    clinvar_P_LP_genes_w_paralogs = c(gene, clinvar_P_LP_genes_w_paralogs) #P/LP clinvar genes that have at least 1 paralogue according to ensembl
  } else {
    clinvar_P_LP_genes_wo_paralogs = c(gene, clinvar_P_LP_genes_wo_paralogs)
  }
}
clinvar_P_LP_genes_w_paralogs_wide = protein_genes_w_paralogues_wide[protein_genes_w_paralogues_wide$external_gene_name %in% clinvar_P_LP_genes_w_paralogs,]
clinvar_P_LP_genes_wo_paralogs_wide = protein_genes_wo_paralogues_wide[protein_genes_wo_paralogues_wide$external_gene_name %in% clinvar_P_LP_genes_wo_paralogs,]
# clinvar_P_LP_genes_wo_paralogs_wide = data.frame( #as some genes from clinvar_P_LP_genes_wo_paralogs do not appear in protein_genes_wo_paralogues_wide, just make own arbitary one
#   external_gene_name = as.vector(clinvar_P_LP_genes_wo_paralogs),
#   number_of_paralogues = 0,
#   paralogue = "No"
# )
# hist(clinvar_P_LP_genes_w_paralogs_wide$number_of_paralogues, breaks = seq(0,50,1), xlab = "number of paralogues", ylab = "number of genes", main="B) Distribution of genes with pathogenic/likely pathogenic variants", cex.main = 0.7)


for (gene in clinvar_B_LB_tableized_genes){
  # print(gene)
  if (gene %in% protein_genes_w_paralogues_genes){
    clinvar_B_LB_genes_w_paralogs = c(gene, clinvar_B_LB_genes_w_paralogs)
  } else {
    clinvar_B_LB_genes_wo_paralogs = c(gene, clinvar_B_LB_genes_wo_paralogs)
  }
}
clinvar_B_LB_genes_w_paralogs_wide = protein_genes_w_paralogues_wide[protein_genes_w_paralogues_wide$external_gene_name %in% clinvar_B_LB_genes_w_paralogs,]
# hist(clinvar_B_LB_genes_w_paralogs_wide$number_of_paralogues, breaks = seq(0,50,1), xlab = "number of paralogues", ylab = "number of genes", main="C) Distribution of genes with benign/likely benign variants", cex.main = 0.7)

clinvar_P_LP_variants_wo_paralogs = vector()
for (variant in clinvar_P_LP_vcf_tableized$SYMBOL){
  if (variant %in% protein_genes_wo_paralogues_genes){
    clinvar_P_LP_variants_wo_paralogs = c(variant, clinvar_P_LP_variants_wo_paralogs)
  }
}

clinvar_B_LB_variants_wo_paralogs = vector()
for (variant in clinvar_B_LB_vcf_tableized$SYMBOL){
  if (variant %in% protein_genes_wo_paralogues_genes){
    clinvar_B_LB_variants_wo_paralogs = c(variant, clinvar_B_LB_variants_wo_paralogs)
  }
}

ks = ks.test(clinvar_P_LP_genes_w_paralogs_wide$number_of_paralogues, clinvar_B_LB_genes_w_paralogs_wide$number_of_paralogues)

# cap = "\\label{fig:paralog_dist}Distribution of genes with paralogues by the number of paralogues they're related to"

#total number of paralogs in clinvar set
all_clinvar_gene_wo_paralogs = unique(c(clinvar_P_LP_genes_wo_paralogs, clinvar_B_LB_genes_wo_paralogs))
all_clinvar_gene_w_paralogs = unique(c(clinvar_P_LP_genes_w_paralogs, clinvar_B_LB_genes_w_paralogs))
write.table(all_clinvar_gene_w_paralogs,file="data/clinvar/all_clinvar_gene_w_paralogs.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

#2 sided KS test for disease and non-disease genes
clinvar_P_LP_genes = dplyr::bind_rows(clinvar_P_LP_genes_wo_paralogs_wide, clinvar_P_LP_genes_w_paralogs_wide)
ks2 = ks.test(clinvar_P_LP_genes$number_of_paralogues, all_protein_genes$number_of_paralogues)


#discrepency between clinvar genes and ensembl genes
genes_not_in_ens = setdiff(clinvar_P_LP_vcf_tableized_genes, disease_genes)
write.table(genes_not_in_ens,file="data/clinvar/genes_not_in_ens.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE) #list of genes not in ensembl but in clinvar


#aggregated paralogues 
protein_genes_w_paralogues_aggregate = aggregate(hsapiens_paralog_associated_gene_name ~., protein_genes_w_paralogues, toString)
# Number of P/LP missense variants in paralogues of genes that have P/LP 
clinvar_P_LP_only_snv_missense_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_only_snv_missense_tableized = clinvar_P_LP_only_snv_missense_tableized[!is.na(clinvar_P_LP_only_snv_missense_tableized$Protein_position) &
                                                                                      !is.na(clinvar_P_LP_only_snv_missense_tableized$Amino_acids) &
                                                                                      !(grepl("-", clinvar_P_LP_only_snv_missense_tableized$Protein_position)) &
                                                                                      !(grepl("-", clinvar_P_LP_only_snv_missense_tableized$Amino_acids)) &
                                                                                      !(grepl("\\*", clinvar_P_LP_only_snv_missense_tableized$Amino_acids)) &
                                                                                      grepl("/", clinvar_P_LP_only_snv_missense_tableized$Amino_acids) &
                                                                                      str_count(clinvar_P_LP_only_snv_missense_tableized$REF) == str_count(clinvar_P_LP_only_snv_missense_tableized$ALT) &
                                                                                      str_count(clinvar_P_LP_only_snv_missense_tableized$Amino_acids) == 3,]
potential_anno = 0
for (row in 1:nrow(protein_genes_w_paralogues_aggregate)){
# for (row in 1:5){
  gene = protein_genes_w_paralogues_aggregate$external_gene_name[row]
  paralogs = unlist(str_split(protein_genes_w_paralogues_aggregate$hsapiens_paralog_associated_gene_name[row], ", "))
  for (paralog in paralogs){
    missense_var = clinvar_P_LP_only_snv_missense_tableized[clinvar_P_LP_only_snv_missense_tableized$SYMBOL == paralog,"ID"]
    potential_anno = potential_anno + length(missense_var) #There's 178378 potentially informative annotations
  }
}
#looking only at 1947 disease genes
x = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:147, sep = "")), quote="", na.strings=".", comment.char="#", stringsAsFactors = 0)
dis_genes_w_dis_para = vector()
for (row in 1:nrow(protein_genes_w_paralogues_aggregate)){
# for (row in 1:5){
  gene = protein_genes_w_paralogues_aggregate$external_gene_name[row]
  paralogs = unlist(str_split(protein_genes_w_paralogues_aggregate$hsapiens_paralog_associated_gene_name[row], ", "))
  if (gene %in% unique(x$Gene)){
    para_check = 0
    for (paralog in paralogs){
      if (paralog %in% clinvar_P_LP_tableized_genes){
        para_check = 1
      }
    }
    if (para_check == 1){
      dis_genes_w_dis_para = c(dis_genes_w_dis_para,gene)
    }
  }
}
potential_anno = 0
for (row in 1:nrow(protein_genes_w_paralogues_aggregate)){
  gene = protein_genes_w_paralogues_aggregate$external_gene_name[row]
  if (gene %in% dis_genes_w_dis_para){
    paralogs = unlist(str_split(protein_genes_w_paralogues_aggregate$hsapiens_paralog_associated_gene_name[row], ", "))
    for (paralog in paralogs){
      missense_var = clinvar_P_LP_only_snv_missense_tableized[clinvar_P_LP_only_snv_missense_tableized$SYMBOL == paralog,"ID"]
      potential_anno = potential_anno + length(missense_var) #There's 83125 potentially informative annotations
    }
  }
}

```

According to ensembl, `r length(protein_genes_w_paralogues_genes)` protein coding genes are defined to have paralogues. While `r length(protein_genes_wo_paralogues_genes)` protein coding genes did not have paralogues. Of those genes with paralogues (**fig.** \ref{fig:paralog_dist}a) the mean had `r round(mean(protein_genes_w_paralogues_wide$number_of_paralogues), 3)` paralogues with a standard deviation of `r round(sd(protein_genes_w_paralogues_wide$number_of_paralogues), 3)`. The maximum number of paralogues a gene had was `r max(protein_genes_w_paralogues_wide$number_of_paralogues)`. The zinc finger family of proteins have highest number of paralogues

In the clinvar pathogenic and likely pathogenic dataset, there were `r length(unique(clinvar_P_LP_tableized$ID))` variants from `r length(clinvar_P_LP_tableized_genes)` genes. `r length(clinvar_P_LP_genes_wo_paralogs)` of these did not have paralogs and therefore the `r length(clinvar_P_LP_variants_wo_paralogs)` variants lying within these genes were not used for annotation, leaving `r length(clinvar_P_LP_tableized$CHROM) - length(clinvar_P_LP_variants_wo_paralogs)` for use in the analysis. The distribution of number of paralogues for these set of genes is shown in **fig.** \ref{fig:paralog_dist}b. The mean number of paralogues was `r round(mean(clinvar_P_LP_genes_w_paralogs_wide$number_of_paralogues), 3)` with a standard deviation `r round(sd(clinvar_P_LP_genes_w_paralogs_wide$number_of_paralogues), 3)`.

For variants in the clinvar benign and likely benign dataset, there were `r length(clinvar_B_LB_tableized$CHROM)` variants from `r length(clinvar_B_LB_tableized_genes)` genes. `r length(clinvar_B_LB_tableized$CHROM) - length(clinvar_B_LB_variants_wo_paralogs)` variants resided in `r length(clinvar_B_LB_genes_w_paralogs)` genes with paralogs. Their respective distribution is shown in **fig.** \ref{fig:paralog_dist}c, with a mean of `r round(mean(clinvar_B_LB_genes_w_paralogs_wide$number_of_paralogues), 3)` paralogues and a standard deviation of `r round(sd(clinvar_B_LB_genes_w_paralogs_wide$number_of_paralogues), 3)`.

Performing a simple kolmogorov smirnov test between the distribution of pathogenic variants in genes with paralogues and benign variants shows a p-value of `r ks$p.value` suggesting that the null hypothesis of the distributions being identical cannot be rejected. From this, there appears to be no statistical difference between pathogenic variants being more likely to lie in genes that have more paralogs compared to benign, at least in regards to the definitions of clinical significance made by clinvar.

## Annotation of Clinvar
```{r clinvar_table, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE, fig.width=7,fig.height=6,fig.cap="\\label{fig:clinvar_table}Total number of known P/LP variants and known B/LB variants predicted to be pathogenic"}
# library(knitr)
#noQC
#pathogenic to pathogenic
p.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#benign to pathogenic
b.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#pathogenic to benign
ptob.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")
#benign to benign
btob.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")
#stats
conf = conf_matrix(p.paralogous_var_align$true_num_of_paralog_anno, p.paralogous_var_align$input_variants, b.paralogous_var_align$true_num_of_paralog_anno, b.paralogous_var_align$input_variants)
conf_b = conf_matrix_benign(ptob.paralogous_var_align$true_num_of_paralog_anno, ptob.paralogous_var_align$input_variants, btob.paralogous_var_align$true_num_of_paralog_anno, btob.paralogous_var_align$input_variants)

genes_before_PA = length(unique(p.paralogous_var_align$Total_paralog_annotations$Gene))
write.table(unique(p.paralogous_var_align$Total_paralog_annotations$Gene), file = "data/clinvar/genes_before_paralogue_annotation.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
genes_with_PA_noQC = length(unique(p.paralogous_var_align$Total_paralog_annotations$SYMBOL))
write.table(unique(p.paralogous_var_align$Total_paralog_annotations[!is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),"Gene"]), file = "data/clinvar/genes_with_paralogue_annotation_noQC.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

b.genes_before_PA = length(unique(b.paralogous_var_align$Total_paralog_annotations$Gene))
b.genes_with_PA_noQC = length(unique(b.paralogous_var_align$Total_paralog_annotations$SYMBOL))


#finding out what genes the variants are in
# var_in_genes = Paralogous_var_align("data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")#,Overlap = 0)
clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_P_LP_tableized = clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & 
#                                                   !(grepl("-", clinvar_P_LP_tableized$Protein_position)) &
#                                                   grepl("/", clinvar_P_LP_tableized$Amino_acids) &
#                                                   str_count(clinvar_P_LP_tableized$Amino_acids) == 3,]
clinvar_P_LP_tableized_genes = as.vector(unique(clinvar_P_LP_tableized$SYMBOL))
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized_genes = as.vector(unique(clinvar_B_LB_tableized$SYMBOL))


var_in_genes_unique_var = data.frame(table(p.paralogous_var_align$Unique_variant_gene_annotations$Gene))
p.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
p.var_in_genes_unique_var = p.var_in_genes_unique_var[p.var_in_genes_unique_var$Var1 %in% clinvar_P_LP_tableized_genes,] #643 HERE!!! 624 w/o indel annotations
write.table(p.var_in_genes_unique_var, file = "data/clinvar/genes_with_number_of_annotations_noQC.csv", na="", row.names = FALSE, col.names = TRUE, sep = ",", quote=FALSE)
not_var_in_genes_unique_var = data.frame(table(p.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
p.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
p.not_var_in_genes_unique_var = p.not_var_in_genes_unique_var[p.not_var_in_genes_unique_var$Var1 %in% clinvar_P_LP_tableized_genes,]
# length(intersect(p.var_in_genes_unique_var$Var1, p.not_var_in_genes_unique_var$Var1))

#top 5 genes where PA performs best with clinvar data
COL3A1_annotations = p.paralogous_var_align$Total_paralog_annotations[p.paralogous_var_align$Total_paralog_annotations$Gene == "COL3A1" & !is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),]
COL3A1_annotations_genes = data.frame(table(COL3A1_annotations$SYMBOL))
COL3A1_annotations_genes = arrange(COL3A1_annotations_genes, desc(Freq))
COL3A1_annotations_genes$Gene = "COL3A1"
COL4A5_annotations = p.paralogous_var_align$Total_paralog_annotations[p.paralogous_var_align$Total_paralog_annotations$Gene == "COL4A5" & !is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),]
COL4A5_annotations_genes = data.frame(table(COL4A5_annotations$SYMBOL))
COL4A5_annotations_genes = arrange(COL4A5_annotations_genes, desc(Freq))
COL4A5_annotations_genes$Gene = "COL4A5"
SCN1A_annotations = p.paralogous_var_align$Total_paralog_annotations[p.paralogous_var_align$Total_paralog_annotations$Gene == "SCN1A" & !is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),]
SCN1A_annotations_genes = data.frame(table(SCN1A_annotations$SYMBOL))
SCN1A_annotations_genes = arrange(SCN1A_annotations_genes, desc(Freq))
SCN1A_annotations_genes$Gene = "SCN1A"
COL1A1_annotations = p.paralogous_var_align$Total_paralog_annotations[p.paralogous_var_align$Total_paralog_annotations$Gene == "COL1A1" & !is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),]
COL1A1_annotations_genes = data.frame(table(COL1A1_annotations$SYMBOL))
COL1A1_annotations_genes = arrange(COL1A1_annotations_genes, desc(Freq))
COL1A1_annotations_genes$Gene = "COL1A1"
COL1A2_annotations = p.paralogous_var_align$Total_paralog_annotations[p.paralogous_var_align$Total_paralog_annotations$Gene == "COL1A2" & !is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),]
COL1A2_annotations_genes = data.frame(table(COL1A2_annotations$SYMBOL))
COL1A2_annotations_genes = arrange(COL1A2_annotations_genes, desc(Freq))
COL1A2_annotations_genes$Gene = "COL1A2"
top_5_paralogues = rbind(COL3A1_annotations_genes, COL4A5_annotations_genes, SCN1A_annotations_genes, COL1A1_annotations_genes, COL1A2_annotations_genes)
top_5_paralogues = top_5_paralogues[,c(3,1,2)]
names(top_5_paralogues) = c("Gene", "Paralogue", "number of predictions")
write.table(top_5_paralogues, file = "data/clinvar/Top_5_genes_with_number_of_annotations_noQC.csv", na="", row.names = FALSE, col.names = TRUE, sep = ",", quote=FALSE)

#carrying on finding out what genes are in the variants being annotated and not for the other parts of the confusion matrixes
var_in_genes_unique_var = data.frame(table(b.paralogous_var_align$Unique_variant_gene_annotations$Gene))
b.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
b.var_in_genes_unique_var = b.var_in_genes_unique_var[b.var_in_genes_unique_var$Var1 %in% clinvar_B_LB_tableized_genes,]
dim(b.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(b.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
b.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
b.not_var_in_genes_unique_var = b.not_var_in_genes_unique_var[b.not_var_in_genes_unique_var$Var1 %in% clinvar_B_LB_tableized_genes,]
# length(intersect(b.var_in_genes_unique_var$Var1, b.not_var_in_genes_unique_var$Var1))

var_in_genes_unique_var = data.frame(table(ptob.paralogous_var_align$Unique_variant_gene_annotations$Gene))
ptob.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(ptob.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(ptob.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
ptob.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(ptob.var_in_genes_unique_var$Var1, ptob.not_var_in_genes_unique_var$Var1))

var_in_genes_unique_var = data.frame(table(btob.paralogous_var_align$Unique_variant_gene_annotations$Gene))
btob.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(btob.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(btob.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
btob.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(btob.var_in_genes_unique_var$Var1, btob.not_var_in_genes_unique_var$Var1))

#para_con
p2.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b2.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
ptob2.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")
btob2.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")

conf2 = conf_matrix(p2.paralogous_var_align$true_num_of_paralog_anno, p2.paralogous_var_align$input_variants, b2.paralogous_var_align$true_num_of_paralog_anno, b2.paralogous_var_align$input_variants)
conf_b2 = conf_matrix_benign(ptob2.paralogous_var_align$true_num_of_paralog_anno, ptob2.paralogous_var_align$input_variants, btob2.paralogous_var_align$true_num_of_paralog_anno, btob2.paralogous_var_align$input_variants)
genes_after_QC1 = length(unique(p2.paralogous_var_align$Total_paralog_annotations$Gene))
write.table(unique(p2.paralogous_var_align$Total_paralog_annotations$Gene), file = "data/clinvar/genes_after_QC1.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
genes_with_PA_QC1 = length(unique(p2.paralogous_var_align$Total_paralog_annotations$SYMBOL))
write.table(unique(p2.paralogous_var_align$Total_paralog_annotations$SYMBOL), file = "data/clinvar/genes_with_PA_QC1.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

var_removed = var_rem_matrix(conf$con_table, conf2$con_table, p2.paralogous_var_align$paralog_data, b2.paralogous_var_align$paralog_data)
var_removed_b = var_rem_matrix(conf_b$con_table, conf_b2$con_table, ptob2.paralogous_var_align$paralog_data, btob2.paralogous_var_align$paralog_data)

#finding out how many genes in para_con
var_in_genes_unique_var = data.frame(table(p2.paralogous_var_align$Unique_variant_gene_annotations$Gene))
p2.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(p2.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(p2.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
p2.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(p2.var_in_genes_unique_var$Var1, p2.not_var_in_genes_unique_var$Var1))

var_in_genes_unique_var = data.frame(table(b2.paralogous_var_align$Unique_variant_gene_annotations$Gene))
b2.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(b2.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(b2.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
b2.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(b2.var_in_genes_unique_var$Var1, b2.not_var_in_genes_unique_var$Var1))

var_in_genes_unique_var = data.frame(table(ptob2.paralogous_var_align$Unique_variant_gene_annotations$Gene))
ptob2.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(ptob2.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(ptob2.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
ptob2.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(ptob2.var_in_genes_unique_var$Var1, ptob2.not_var_in_genes_unique_var$Var1))

var_in_genes_unique_var = data.frame(table(btob2.paralogous_var_align$Unique_variant_gene_annotations$Gene))
btob2.var_in_genes_unique_var = arrange(var_in_genes_unique_var, desc(Freq))
dim(btob2.paralogous_var_align$Unique_variants_not_annotated)[1]
not_var_in_genes_unique_var = data.frame(table(btob2.paralogous_var_align$Unique_variant_genes_not_annotated$Gene))
btob2.not_var_in_genes_unique_var = arrange(not_var_in_genes_unique_var, desc(Freq))
length(intersect(btob2.var_in_genes_unique_var$Var1, btob2.not_var_in_genes_unique_var$Var1))

#all_con
p3.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b3.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
ptob3.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")
btob3.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")

conf3 = conf_matrix(p3.paralogous_var_align$true_num_of_paralog_anno, p3.paralogous_var_align$input_variants, b3.paralogous_var_align$true_num_of_paralog_anno, b3.paralogous_var_align$input_variants)
conf_b3 = conf_matrix_benign(ptob3.paralogous_var_align$true_num_of_paralog_anno, ptob3.paralogous_var_align$input_variants, btob3.paralogous_var_align$true_num_of_paralog_anno, btob3.paralogous_var_align$input_variants)
genes_after_QC2 = length(unique(p3.paralogous_var_align$Total_paralog_annotations$Gene))
write.table(unique(p3.paralogous_var_align$Total_paralog_annotations$Gene), file = "data/clinvar/genes_after_QC2.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
genes_with_PA_QC2 = length(unique(p3.paralogous_var_align$Total_paralog_annotations$SYMBOL))
write.table(unique(p3.paralogous_var_align$Total_paralog_annotations$SYMBOL), file = "data/clinvar/genes_with_PA_QC2.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

var_removed2 = var_rem_matrix(conf2$con_table, conf3$con_table, p3.paralogous_var_align$paralog_data, b3.paralogous_var_align$paralog_data)
var_removed_b2 = var_rem_matrix(conf_b2$con_table, conf_b3$con_table, ptob3.paralogous_var_align$paralog_data, btob3.paralogous_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.paralogous_var_align$Total_paralog_annotations[p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# p4.Total_paralog_annotations = p4.Total_paralog_annotations[!duplicated(p4.Total_paralog_annotations$ID.x),]
p4.Total_paralog_annotations = distinct(p4.Total_paralog_annotations)

p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))
p4.Unique_variant_annotations = p4.Total_paralog_annotations[!is.na(p4.Total_paralog_annotations$ID.y),]
p4.Unique_variant_annotations = distinct(p4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p4.true_num_of_paralog_anno = length(p4.Unique_variant_annotations$Variant_pos)

#benign to pathogenic
b4.Total_paralog_annotations = b3.paralogous_var_align$Total_paralog_annotations[b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# b4.Total_paralog_annotations = b4.Total_paralog_annotations[!duplicated(b4.Total_paralog_annotations$ID.x),]
b4.Total_paralog_annotations = distinct(b4.Total_paralog_annotations)

b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))
b4.Unique_variant_annotations = b4.Total_paralog_annotations[!is.na(b4.Total_paralog_annotations$ID.y),]
b4.Unique_variant_annotations = distinct(b4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b4.true_num_of_paralog_anno = length(b4.Unique_variant_annotations$Variant_pos)

#pathogenic to benign
ptob4.Total_paralog_annotations = ptob3.paralogous_var_align$Total_paralog_annotations[ptob3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==ptob3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
ptob4.Total_paralog_annotations = distinct(ptob4.Total_paralog_annotations)

ptob4.num_of_paralog_anno = sum(!is.na(ptob4.Total_paralog_annotations$ID.y))
ptob4.Unique_variant_annotations = ptob4.Total_paralog_annotations[!is.na(ptob4.Total_paralog_annotations$ID.y),]
ptob4.Unique_variant_annotations = distinct(ptob4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
ptob4.true_num_of_paralog_anno = length(ptob4.Unique_variant_annotations$Variant_pos)

#benign to benign
btob4.Total_paralog_annotations = btob3.paralogous_var_align$Total_paralog_annotations[btob3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==btob3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
btob4.Total_paralog_annotations = distinct(btob4.Total_paralog_annotations)

btob4.num_of_paralog_anno = sum(!is.na(btob4.Total_paralog_annotations$ID.y))
btob4.Unique_variant_annotations = btob4.Total_paralog_annotations[!is.na(btob4.Total_paralog_annotations$ID.y),]
btob4.Unique_variant_annotations = distinct(btob4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
btob4.true_num_of_paralog_anno = length(btob4.Unique_variant_annotations$Variant_pos)


conf4 = conf_matrix(p4.true_num_of_paralog_anno, p3.paralogous_var_align$input_variants, b4.true_num_of_paralog_anno, b3.paralogous_var_align$input_variants)
conf_b4 = conf_matrix_benign(ptob4.true_num_of_paralog_anno, ptob3.paralogous_var_align$input_variants, btob4.true_num_of_paralog_anno, btob3.paralogous_var_align$input_variants)
genes_after_QC3 = length(unique(p4.Total_paralog_annotations$Gene))
write.table(unique(p4.Total_paralog_annotations$Gene), file = "data/clinvar/genes_after_QC3.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
genes_with_PA_QC3 = length(unique(p4.Total_paralog_annotations$SYMBOL))
write.table(unique(p4.Total_paralog_annotations$SYMBOL), file = "data/clinvar/genes_with_PA_QC3.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

var_removed3 = var_rem_matrix(conf3$con_table, conf4$con_table, p3.paralogous_var_align$paralog_data, b3.paralogous_var_align$paralog_data)
var_removed_b3 = var_rem_matrix(conf_b3$con_table, conf_b4$con_table, ptob3.paralogous_var_align$paralog_data, btob3.paralogous_var_align$paralog_data)

#benign test for only alt con
ptob5.Total_paralog_annotations = ptob.paralogous_var_align$Total_paralog_annotations[ptob.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==ptob.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
ptob5.Total_paralog_annotations = distinct(ptob5.Total_paralog_annotations)

ptob5.num_of_paralog_anno = sum(!is.na(ptob5.Total_paralog_annotations$ID.y))
ptob5.Unique_variant_annotations = ptob5.Total_paralog_annotations[!is.na(ptob5.Total_paralog_annotations$ID.y),]
ptob5.Unique_variant_annotations = distinct(ptob5.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
ptob5.true_num_of_paralog_anno = length(ptob5.Unique_variant_annotations$Variant_pos)

btob5.Total_paralog_annotations = btob.paralogous_var_align$Total_paralog_annotations[btob.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==btob.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
btob5.Total_paralog_annotations = distinct(btob5.Total_paralog_annotations)

btob5.num_of_paralog_anno = sum(!is.na(btob5.Total_paralog_annotations$ID.y))
btob5.Unique_variant_annotations = btob5.Total_paralog_annotations[!is.na(btob5.Total_paralog_annotations$ID.y),]
btob5.Unique_variant_annotations = distinct(btob5.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
btob5.true_num_of_paralog_anno = length(btob5.Unique_variant_annotations$Variant_pos)

conf_b5 = conf_matrix_benign(ptob5.true_num_of_paralog_anno, ptob.paralogous_var_align$input_variants, btob5.true_num_of_paralog_anno, btob.paralogous_var_align$input_variants)

####
#alt_con with para_con but without all_con
#pathogenic to pathogenic
p6.Total_paralog_annotations = p2.paralogous_var_align$Total_paralog_annotations[p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p6.Total_paralog_annotations = distinct(p6.Total_paralog_annotations)

p6.num_of_paralog_anno = sum(!is.na(p6.Total_paralog_annotations$ID.y))
p6.Unique_variant_annotations = p6.Total_paralog_annotations[!is.na(p6.Total_paralog_annotations$ID.y),]
p6.Unique_variant_annotations = distinct(p6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p6.true_num_of_paralog_anno = length(p6.Unique_variant_annotations$Variant_pos)

#benign to pathogenic
b6.Total_paralog_annotations = b2.paralogous_var_align$Total_paralog_annotations[b2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
b6.Total_paralog_annotations = distinct(b6.Total_paralog_annotations)

b6.num_of_paralog_anno = sum(!is.na(b6.Total_paralog_annotations$ID.y))
b6.Unique_variant_annotations = b6.Total_paralog_annotations[!is.na(b6.Total_paralog_annotations$ID.y),]
b6.Unique_variant_annotations = distinct(b6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b6.true_num_of_paralog_anno = length(b6.Unique_variant_annotations$Variant_pos)

#pathogenic to benign
ptob6.Total_paralog_annotations = ptob2.paralogous_var_align$Total_paralog_annotations[ptob2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==ptob2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
ptob6.Total_paralog_annotations = distinct(ptob6.Total_paralog_annotations)

ptob6.num_of_paralog_anno = sum(!is.na(ptob6.Total_paralog_annotations$ID.y))
ptob6.Unique_variant_annotations = ptob6.Total_paralog_annotations[!is.na(ptob6.Total_paralog_annotations$ID.y),]
ptob6.Unique_variant_annotations = distinct(ptob6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
ptob6.true_num_of_paralog_anno = length(ptob6.Unique_variant_annotations$Variant_pos)

#benign to benign
btob6.Total_paralog_annotations = btob2.paralogous_var_align$Total_paralog_annotations[btob2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==btob2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
btob6.Total_paralog_annotations = distinct(btob6.Total_paralog_annotations)

btob6.num_of_paralog_anno = sum(!is.na(btob6.Total_paralog_annotations$ID.y))
btob6.Unique_variant_annotations = btob6.Total_paralog_annotations[!is.na(btob6.Total_paralog_annotations$ID.y),]
btob6.Unique_variant_annotations = distinct(btob6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
btob6.true_num_of_paralog_anno = length(btob6.Unique_variant_annotations$Variant_pos)


conf6 = conf_matrix(p6.true_num_of_paralog_anno, p2.paralogous_var_align$input_variants, b6.true_num_of_paralog_anno, b2.paralogous_var_align$input_variants)
conf_b6 = conf_matrix_benign(ptob6.true_num_of_paralog_anno, ptob2.paralogous_var_align$input_variants, btob6.true_num_of_paralog_anno, btob2.paralogous_var_align$input_variants)
# genes_after_QC3_w_QC1_no_QC2 = length(unique(p6.Total_paralog_annotations$Gene))
# write.table(unique(p6.Total_paralog_annotations$Gene), file = "data/clinvar/genes_after_QC3_w_QC1_no_QC2.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
# genes_with_PA_QC3_w_QC1_no_QC2 = length(unique(p6.Total_paralog_annotations$SYMBOL))
# write.table(unique(p6.Total_paralog_annotations$SYMBOL), file = "data/clinvar/genes_with_PA_QC3_w_QC1_no_QC2.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

var_removed4 = var_rem_matrix(conf2$con_table, conf6$con_table, p2.paralogous_var_align$paralog_data, b2.paralogous_var_align$paralog_data)
var_removed_b4 = var_rem_matrix(conf_b2$con_table, conf_b6$con_table, ptob2.paralogous_var_align$paralog_data, btob2.paralogous_var_align$paralog_data)

####

#Overall_table for predicting pathogenic variants
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
Total = c(dim(p.paralogous_var_align$input_variants)[1],dim(b.paralogous_var_align$input_variants)[1],"NA","NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),signif(conf$PPV, digits = 2),signif(conf$Sensitivity, digits = 2),signif(conf$Specificity, digits = 2),signif(conf$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC1 = c(format(conf2$con_table[2], scientific = FALSE),format(conf2$con_table[4], scientific = FALSE),signif(conf2$PPV, digits = 2), signif(conf2$Sensitivity, digits = 2),signif(conf2$Specificity, digits = 2), signif(conf2$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC2 = c(format(conf3$con_table[2], scientific = FALSE),format(conf3$con_table[4], scientific = FALSE),signif(conf3$PPV, digits = 2),signif(conf3$Sensitivity, digits = 2),signif(conf3$Specificity, digits = 2),signif(conf3$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC3 = c(format(conf4$con_table[2], scientific = FALSE),format(conf4$con_table[4], scientific = FALSE),signif(conf4$PPV, digits = 2),signif(conf4$Sensitivity, digits = 2),signif(conf4$Specificity, digits = 2),signif(conf4$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC3_w_QC1_no_QC2 = c(format(conf6$con_table[2], scientific = FALSE),format(conf6$con_table[4], scientific = FALSE),signif(conf6$PPV, digits = 2),signif(conf6$Sensitivity, digits = 2),signif(conf6$Specificity, digits = 2),signif(conf6$Pvalue$p.value, digits = 2))

Variants_removed_after_PA_QC1 = c(format(var_removed$con_table[2], scientific = FALSE),format(var_removed$con_table[4], scientific = FALSE),signif(var_removed$PPV, digits = 2),signif(var_removed$Sensitivity, digits = 2),signif(var_removed$Specificity, digits = 2),signif(var_removed$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC2 = c(format(var_removed2$con_table[2], scientific = FALSE),format(var_removed2$con_table[4], scientific = FALSE),signif(var_removed2$PPV, digits = 2),signif(var_removed2$Sensitivity, digits = 2),signif(var_removed2$Specificity, digits = 2),signif(var_removed2$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC3 = c(format(var_removed3$con_table[2], scientific = FALSE),format(var_removed3$con_table[4], scientific = FALSE),signif(var_removed3$PPV, digits = 2),signif(var_removed3$Sensitivity, digits = 2),signif(var_removed3$Specificity, digits = 2),signif(var_removed3$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC3_w_QC1_no_QC2 = c(format(var_removed4$con_table[2], scientific = FALSE),format(var_removed4$con_table[4], scientific = FALSE),signif(var_removed4$PPV, digits = 2),signif(var_removed4$Sensitivity, digits = 2),signif(var_removed4$Specificity, digits = 2),signif(var_removed4$Pvalue$p.value, digits = 2))

# Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),conf$PPV,conf$Sensitivity,conf$Pvalue$p.value)
# Variants_remaining_after_PA_QC1 = c(conf2$con_table[2],conf2$con_table[4],conf2$PPV, conf2$Sensitivity, conf2$Pvalue$p.value)
# 
# Variants_removed_after_PA_QC1 = c(var_removed$con_table[2],var_removed$con_table[4],var_removed$PPV,"NA",var_removed$Pvalue$p.value)
# Variants_remaining_after_PA_QC2 = c(conf3$con_table[2],conf3$con_table[4],conf3$PPV,conf3$Sensitivity,conf3$Pvalue$p.value)
# Variants_removed_after_PA_QC2 = c(var_removed2$con_table[2],var_removed2$con_table[4],var_removed2$PPV,"NA",var_removed2$Pvalue$p.value)
# Variants_remaining_after_PA_QC3 = c(conf4$con_table[2],conf4$con_table[4],conf4$PPV,conf4$Sensitivity,conf4$Pvalue$p.value)
# Variants_removed_after_PA_QC3 = c(var_removed3$con_table[2],var_removed3$con_table[4],var_removed3$PPV,"NA",var_removed3$Pvalue$p.value)

clinvar_set = data.frame(#Variant,
                         Total,
                         Paralogue_Annotation_no_QC,
                         Variants_remaining_after_PA_QC1,
                         Variants_remaining_after_PA_QC2,
                         Variants_remaining_after_PA_QC3_w_QC1_no_QC2,
                         Variants_remaining_after_PA_QC3,
                         Variants_removed_after_PA_QC1,
                         Variants_removed_after_PA_QC2,
                         Variants_removed_after_PA_QC3_w_QC1_no_QC2,
                         Variants_removed_after_PA_QC3
                         )

clinvar_set = t(clinvar_set)
colnames(clinvar_set) = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
rownames(clinvar_set) = c("Total variants", 
                          "Paralogue Annotation", 
                          "Variants remaining after QC1",
                          "Variants remaining after QC1+QC2", 
                          "Variants remaining after QC1+QC3", 
                          "Variants remaining after QC1+QC2+QC3",
                          "Variants removed from QC1", 
                          "Variants removed from QC1+QC2",
                          "Variants removed from QC1+QC3",
                          "Variants removed from QC1+QC2+QC3")
write.table(clinvar_set, file = "data/clinvar/clinvar_set.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)
kable(clinvar_set)
# pander(clinvar_set2)
# pander(clinvar_set3)

clinvar_set_genes = data.frame(c("Total","Genes with Paralogue Annotations - no QC", "Genes remaining after QC1", "Genes with Paralogue Annotations - QC1", "Genes remaining after QC2", "Genes with Paralogue Annotations - QC2", "Genes remaining after QC3", "Genes with Paralogue Annotations - QC3"),
                               c(genes_before_PA,genes_with_PA_noQC,genes_after_QC1,genes_with_PA_QC1,genes_after_QC2,genes_with_PA_QC2,genes_after_QC3,genes_with_PA_QC3))
colnames(clinvar_set_genes) = c("","Number of genes")
kable(clinvar_set_genes)

#Overall_table for predicting Benign variants
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
Total = c(dim(p.paralogous_var_align$input_variants)[1],dim(b.paralogous_var_align$input)[1],"NA","NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(conf_b$con_table[2], scientific = FALSE),format(conf_b$con_table[4], scientific = FALSE),signif(conf_b$PPV, digits = 2),signif(conf_b$Sensitivity, digits = 2),signif(conf_b$Specificity, digits = 2),signif(conf_b$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC1 = c(format(conf_b2$con_table[2], scientific = FALSE),format(conf_b2$con_table[4], scientific = FALSE),signif(conf_b2$PPV, digits = 2), signif(conf_b2$Sensitivity, digits = 2),signif(conf_b2$Specificity, digits = 2), signif(conf_b2$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC1 = c(format(var_removed_b$con_table[2], scientific = FALSE),format(var_removed_b$con_table[4], scientific = FALSE),signif(var_removed_b$PPV, digits = 2),signif(var_removed_b$Sensitivity, digits = 2),signif(var_removed_b$Specificity, digits = 2),signif(var_removed_b$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC2 = c(format(conf_b3$con_table[2], scientific = FALSE),format(conf_b3$con_table[4], scientific = FALSE),signif(conf_b3$PPV, digits = 2),signif(conf_b3$Sensitivity, digits = 2),signif(conf_b3$Specificity, digits = 2),signif(conf_b3$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC2 = c(format(var_removed_b2$con_table[2], scientific = FALSE),format(var_removed_b2$con_table[4], scientific = FALSE),signif(var_removed_b2$PPV, digits = 2),signif(var_removed_b2$Sensitivity, digits = 2),signif(var_removed_b2$Specificity, digits = 2),signif(var_removed_b2$Pvalue$p.value, digits = 2))
Variants_remaining_after_PA_QC3 = c(format(conf_b4$con_table[2], scientific = FALSE),format(conf_b4$con_table[4], scientific = FALSE),signif(conf_b4$PPV, digits = 2),signif(conf_b4$Sensitivity, digits = 2),signif(conf_b4$Specificity, digits = 2),signif(conf_b4$Pvalue$p.value, digits = 2))
Variants_removed_after_PA_QC3 = c(format(var_removed_b3$con_table[2], scientific = FALSE),format(var_removed_b3$con_table[4], scientific = FALSE),signif(var_removed_b3$PPV, digits = 2),signif(var_removed_b3$Sensitivity, digits = 2),signif(var_removed_b3$Specificity, digits = 2),signif(var_removed_b3$Pvalue$p.value, digits = 2))

# Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),conf$PPV,conf$Sensitivity,conf$Pvalue$p.value)
# Variants_remaining_after_PA_QC1 = c(conf2$con_table[2],conf2$con_table[4],conf2$PPV, conf2$Sensitivity, conf2$Pvalue$p.value)
# 
# Variants_removed_after_PA_QC1 = c(var_removed$con_table[2],var_removed$con_table[4],var_removed$PPV,"NA",var_removed$Pvalue$p.value)
# Variants_remaining_after_PA_QC2 = c(conf3$con_table[2],conf3$con_table[4],conf3$PPV,conf3$Sensitivity,conf3$Pvalue$p.value)
# Variants_removed_after_PA_QC2 = c(var_removed2$con_table[2],var_removed2$con_table[4],var_removed2$PPV,"NA",var_removed2$Pvalue$p.value)
# Variants_remaining_after_PA_QC3 = c(conf4$con_table[2],conf4$con_table[4],conf4$PPV,conf4$Sensitivity,conf4$Pvalue$p.value)
# Variants_removed_after_PA_QC3 = c(var_removed3$con_table[2],var_removed3$con_table[4],var_removed3$PPV,"NA",var_removed3$Pvalue$p.value)

clinvar_set_benign = data.frame(#Variant,
                         Total,
                         Paralogue_Annotation_no_QC,
                         Variants_remaining_after_PA_QC1,
                         Variants_remaining_after_PA_QC2,
                         Variants_remaining_after_PA_QC3,
                         Variants_removed_after_PA_QC1,
                         Variants_removed_after_PA_QC2,
                         Variants_removed_after_PA_QC3
                         )

clinvar_set_benign = t(clinvar_set_benign)
colnames(clinvar_set_benign) = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
rownames(clinvar_set_benign) = c("Total variants", "Paralogue Annotation", "Variants remaining after QC1", "Variants remaining after QC2", "Variants remaining after QC3", "Variants removed from QC1", "Variants removed from QC2", "Variants removed from QC3")
write.table(clinvar_set_benign, file = "data/clinvar/clinvar_set_benign.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)
# kable(clinvar_set_benign)

### Analysing average number of annotations per input
num_of_anno = vector()
for (i in p.paralogous_var_align$Total_paralog_annotations[!is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),"ID.x"]){
    num_of_anno = c(num_of_anno, i)
}
summary_num_of_anno = as.data.frame(table(num_of_anno))
p.count_num_of_anno_per_var = as.data.frame(table(summary_num_of_anno$Freq))
colnames(p.count_num_of_anno_per_var) = c("annotations","variants")
p.count_num_of_anno_per_var$annotations = as.numeric(as.character(p.count_num_of_anno_per_var$annotations))
p.count_num_of_anno_per_var$label = "pathogenic"
p.data = data.frame(
  variant = "pathogenic",
  annotations = summary_num_of_anno$Freq
)
p.data = rbind(p.data, data.frame(
  variant = "pathogenic",
  annotations = integer(19660)
))
p.data$variant = as.factor(p.data$variant)

num_of_anno = vector()
for (i in b.paralogous_var_align$Total_paralog_annotations[!is.na(b.paralogous_var_align$Total_paralog_annotations$ID.y),"ID.x"]){
    num_of_anno = c(num_of_anno, i)
}
summary_num_of_anno = as.data.frame(table(num_of_anno))
b.count_num_of_anno_per_var = as.data.frame(table(summary_num_of_anno$Freq))
colnames(b.count_num_of_anno_per_var) = c("annotations","variants")
b.count_num_of_anno_per_var$annotations = as.numeric(as.character(b.count_num_of_anno_per_var$annotations))
b.count_num_of_anno_per_var$label = "benign"
b.data = data.frame(
  variant = "benign",
  annotations = summary_num_of_anno$Freq
)
b.data = rbind(b.data, data.frame(
  variant = "benign",
  annotations = integer(14251)
))
b.data$variant = as.factor(b.data$variant)

data = rbind(p.data, b.data)
data2 = rbind(p.count_num_of_anno_per_var, b.count_num_of_anno_per_var)

dens = ggplot(data, aes(x = variant, y = annotations)) +
  # geom_violin(trim = T) +
  # geom_dotplot(binaxis='y', stackdir='center', dotsize=0.5, binwidth = 10) +
  geom_jitter(aes(colour = variant, shape = variant),size = 1) +
  # scale_shape_manual(values = c(1,2)) +
  coord_trans(y="log10") +
  scale_y_continuous(breaks = c(1,2,3,4,5,10,20,30,40,50,100,200,300,400,500,600))

data2 = rbind(data2, 
              data.frame(annotations = c(0,0),
                         variants = c(19660,14251),
                         label = c("pathogenic","benign")))

dens = ggplot(data2, aes(x = variants, y = annotations)) +
  geom_point(aes(colour = label, shape = label)) + 
  # geom_hline(yintercept = mean(p.data$annotations),linetype="dashed", color = "blue") +
  # geom_hline(yintercept = mean(b.data$annotations),linetype="dashed", color = "red") +
  # coord_trans(y="log2") +
  # scale_y_continuous(breaks = c(1,2,3,4,5,10,20,30,40,50,100,200,300,400,500,1000)) +
  # scale_x_continuous(breaks = c(0,500,1000,1500,2000,2500,3000), limits = c(0,3000)) +
  ylab("number of annotations per variant") + xlab("number of variants") + theme(axis.title = element_text(size = 15), 
    axis.text = element_text(size = 15), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 12))
  
ggsave("images/average_number_of_annotations_per_variant.png", height = 10, width = 8)
```

Prior to annotation, there were `r length(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Paralogue_Vars),1])` variants that aligned to at least one paralogous equivalent position according to ensembl (`r length(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Paralogue_Vars),1])/length(clinvar_P_LP_tableized$CHROM)`)

The full analysis of clinvar variants is shown in **table \ref{fig:clinvar_table}**. In summary, `r conf$con_table[1]` Pathogenic and Likely Pathogenic (P/LP) variants and `r conf$con_table[2]` Benign and Likey Benign (B/LB) variants from clinvar had paralogue annotations. With no quality control, `r conf$con_table[2]` known P/LP  and `r conf$con_table[4]` known B/LB variants were predicted to be pathogenic, given a PPV and sensitivity of `r conf$PPV` and `r conf$Sensitivity` respectively. Comparatively, predicting benign variants was not as reliable. With `r conf_b$con_table[2]` known P/LP and `r conf_b$con_table[4]` known B/LB variants predicted to be benign. Though the proportional differene is statistically significant with a p-value of $`r conf_b$Pvalue$p.value`$, this lead to a PPV and sensitivity of `r conf_b$PPV` and `r conf_b$Sensitivity` respectively. 

Using the aforementioned quality control steps to increase the stringency of conservation across alignment columns in regards to reference and alternate amino acid alleles shows improvement to PPV and decrease in sensitivity over all for predicting pathogenic variants. But this does not help the case for predicting benign variants. The PPV does not improve significantly to a reliable level. Therefore, it can be concluded that at least with the dataset used in this study, paralogue annotation can be used as a variant classification method for predicting pathogenic variants, but not benign.


Looking at pipeline of study:

```{r, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)

num_only_missense = length(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Amino_acids) & 
                             !(grepl("-", clinvar_P_LP_tableized$Amino_acids)) &
                             !(grepl("-", clinvar_P_LP_tableized$Protein_position)) &
                             !(grepl("\\*", clinvar_P_LP_tableized$Amino_acids)) &
                             grepl("/", clinvar_P_LP_tableized$Amino_acids) &
                               str_count(clinvar_P_LP_tableized$REF) == str_count(clinvar_P_LP_tableized$ALT) &
                             str_count(clinvar_P_LP_tableized$Amino_acids) == 3,"ID"]))


x = read.csv(file="/media/nick/821ED5711ED55F2B/Users/Nick/Documents/PhD/Paralogues/data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
y = read.csv(file="/media/nick/821ED5711ED55F2B/Users/Nick/Documents/PhD/Paralogues/data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.vcf", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
length(unique(x$V3)) #short one
length(unique(y$V3)) #longer one
setdiff(unique(y$V3),unique(x$V3))

x1 = read.csv(file="/media/nick/821ED5711ED55F2B/Users/Nick/Documents/PhD/Paralogues/data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
y1 = read.csv(file="/media/nick/821ED5711ED55F2B/Users/Nick/Documents/PhD/Paralogues/data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.vcf", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
setdiff(unique(y1$V3),unique(x1$V3))
```
NB that in P/LP 11 variants with IDs: 16549; 525690; 12282; 3882; 29695; 13538; 7226; 17114; 7366; 4576; and 504370 are actually deletions so VEP does not predict anything and does not output them in results. Hence out_paraloc file has 11 less lines than the vcf file.
For B/LB theres 295 dels.

Finding overlapping genes and distribution of variants among genes in test set:
```{r, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
# x = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
x0 = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:147, sep = "")), quote="", na.strings=".", comment.char="#", stringsAsFactors = 0)
# x = distinct(x0[c("Variant_pos", "ID", "REF", "ALT")])
library(plyr)
y = ddply(x0[,c("Variant_pos", "ID", "REF", "ALT")],.(Variant_pos, ID, REF, ALT),nrow)
y = y[y$V1>1,]
dim(y)
z = x0[x0$ID %in% y$ID,]

out_paraloc_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep="\t", quote="", stringsAsFactors = 0)
p.overlapp_genes = Paralogous_var_align(Overlap = 1, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.overlapp_genes = p.overlapp_genes$paralog_data[,c("ID", "Gene")]
p.overlapp_genes = unique(p.overlapp_genes$ID)
overlapp_genes = data.frame(
  # CHROM = c(),
  # POS = c(),
  # REF = c(),
  # ALT = c(),
  # ID = c(),
  # SYMBOL = c()
)
for (i in p.overlapp_genes){
  x = out_paraloc_tableized[grepl(paste("\\b",i,"\\b", sep = ""),out_paraloc_tableized$ID, perl = T), c("CHROM","POS","REF","ALT","ID","SYMBOL")]
  # tmp_df = data.frame(
  #   id = i,
  #   genes = x
  # )
  overlapp_genes= rbind(overlapp_genes, x)
}
overlapp_genes = aggregate(SYMBOL ~., overlapp_genes, toString)
names(overlapp_genes)[6] = "GENES"
write.table(overlapp_genes, file = "data/clinvar/variants_in_overlapp_genes.csv", na=".", row.names = FALSE, col.names = TRUE, sep = ",", quote=FALSE) #list of overlapping genes

### Analysing average number of variants per gene
num_of_var = vector()
for (i in x0[,"Gene"]){
    num_of_var = c(num_of_var, i)
}
summary_num_of_var = as.data.frame(table(num_of_var))
# mean_var_per_gene = mean(summary_num_of_var$Freq)
p.count_num_of_var_per_gene = as.data.frame(table(summary_num_of_var$Freq))
colnames(p.count_num_of_var_per_gene) = c("variants","genes")
p.count_num_of_var_per_gene$variants = as.numeric(as.character(p.count_num_of_var_per_gene$variants))
p.count_num_of_var_per_gene$label = "pathogenic"
p.data = data.frame(
  gene = "pathogenic",
  variants = summary_num_of_var$Freq
)
p.data$gene = as.factor(p.data$gene)

b = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs", sep = "\t", header=0, stringsAsFactors = 0, fill=FALSE, quote="", na.strings=".", comment.char="#")
b0 = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:147, sep = "")), quote="", na.strings=".", comment.char="#", stringsAsFactors = 0)
b = distinct(b0[c("Variant_pos", "ID", "REF", "ALT")])
c = ddply(b0[,c("Variant_pos", "ID", "REF", "ALT")],.(Variant_pos, ID, REF, ALT),nrow)
c = c[c$V1>1,]
dim(c)

num_of_var = vector()
for (i in b0[,"Gene"]){
    num_of_var = c(num_of_var, i)
}
summary_num_of_var = as.data.frame(table(num_of_var))
b.count_num_of_var_per_gene = as.data.frame(table(summary_num_of_var$Freq))
colnames(b.count_num_of_var_per_gene) = c("variants","genes")
b.count_num_of_var_per_gene$variants = as.numeric(as.character(b.count_num_of_var_per_gene$variants))
b.count_num_of_var_per_gene$label = "benign"
b.data = data.frame(
  gene = "benign",
  variants = summary_num_of_var$Freq
)
b.data$gene = as.factor(b.data$gene)

data = rbind(p.data, b.data)
data2 = rbind(p.count_num_of_var_per_gene, b.count_num_of_var_per_gene)

dens2 = ggplot(data, aes(x = gene, y = variants)) +
  # geom_violin(trim = T, adjust = 0.01,aes(fill = gene)) +
  # geom_jitter(aes(colour = gene, shape = gene),size = 1) +
  # scale_shape_manual(values = c(1,2)) +
  coord_trans(y="log10") +
  scale_y_continuous(breaks = c(1,2,3,4,5,10,20,30,40,50,100,200,300,400,500,600))
  
dens2 = ggplot(data2, aes(x = genes, y = variants)) +
  geom_point(aes(colour = label, shape = label)) + 
  # geom_hline(yintercept = mean(p.data$variants),linetype="dashed", color = "blue") + 
  # geom_hline(yintercept = mean(b.data$variants),linetype="dashed", color = "red") + 
  coord_trans(y="log10") +
  scale_y_continuous(breaks = c(1,2,3,4,5,10,20,30,40,50,100,200,300,400,500,600)) +
  ylab("number of variants per gene") + xlab("number of genes") + theme(axis.title = element_text(size = 15), 
    axis.text = element_text(size = 15), 
    plot.title = element_text(size = 15), 
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 12))
  
ggsave("images/average_number_of_variants_per_genes.png", height = 10, width = 8)
```
NB there's 435 unique variants lying in overlapping genes in the P/LP. Overall there's 489 instances of variants in a position that covers two genes (i.e. one instance of two gene overlapp)

Getting numbers for experimental design flow chart:
```{r, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
clinvar_P_LP_only_snv_missense_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_only_snv_missense_tableized = clinvar_P_LP_only_snv_missense_tableized[!is.na(clinvar_P_LP_only_snv_missense_tableized$Protein_position) &
                                                                                      !is.na(clinvar_P_LP_only_snv_missense_tableized$Amino_acids) &
                                                                                      !(grepl("-", clinvar_P_LP_only_snv_missense_tableized$Protein_position)) &
                                                                                      !(grepl("-", clinvar_P_LP_only_snv_missense_tableized$Amino_acids)) &
                                                                                      !(grepl("\\*", clinvar_P_LP_only_snv_missense_tableized$Amino_acids)) &
                                                                                      grepl("/", clinvar_P_LP_only_snv_missense_tableized$Amino_acids) &
                                                                                      str_count(clinvar_P_LP_only_snv_missense_tableized$REF) == str_count(clinvar_P_LP_only_snv_missense_tableized$ALT) &
                                                                                      str_count(clinvar_P_LP_only_snv_missense_tableized$Amino_acids) == 3,]

num_of_var_clinvar_P_LP_only_snv_missense_tableized = length(unique(clinvar_P_LP_only_snv_missense_tableized$ID)) #43136 actually 30476
num_of_genes_clinvar_P_LP_only_snv_missense_tableized = length(unique(clinvar_P_LP_only_snv_missense_tableized$SYMBOL)) #3631 actually 3376

x = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:147, sep = "")), quote="", na.strings=".", comment.char="#", stringsAsFactors = 0)
setdiff(unique(clinvar_P_LP_genes_w_paralogs),unique(x$Gene))
setdiff(unique(x$Gene),unique(clinvar_P_LP_genes_w_paralogs))
# write.table(gene_EFs, file = "data/case_and_control/gene_EFs.csv", na="", row.names = FALSE, col.names = TRUE, sep = ",", quote=FALSE)
# Number of disease protein coding genes that have at least 1 paralogue and at least 1 P/LP variant aligning to a non-gap position
length(unique(x$Gene)) #2485
# Number of disease protein coding genes that have at least 1 paralogue and at least 1 P/LP variant aligning to a non-gap position of another disease gene
length(intersect(unique(x$Gene),unique(clinvar_P_LP_tableized_genes))) #2451 this is probably wrong see below
dis_genes_w_dis_para = vector()
for (row in 1:nrow(protein_genes_w_paralogues_aggregate)){
# for (row in 1:5){
  gene = protein_genes_w_paralogues_aggregate$external_gene_name[row]
  paralogs = unlist(str_split(protein_genes_w_paralogues_aggregate$hsapiens_paralog_associated_gene_name[row], ", "))
  if (gene %in% unique(x$Gene)){
    para_check = 0
    for (paralog in paralogs){
      if (paralog %in% clinvar_P_LP_tableized_genes){
        para_check = 1
      }
    }
    if (para_check == 1){
      dis_genes_w_dis_para = c(dis_genes_w_dis_para,gene)
    }
  }
}
length(dis_genes_w_dis_para) #1947
write.table(dis_genes_w_dis_para, file = "data/clinvar/1947_dis_clinvar_genes_w_dis_para.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)


# write.table(setdiff(unique(clinvar_P_LP_tableized_genes),unique(x$Gene)), file = "data/clinvar/genes_in_out_paraloc_paralogs2.noQC.txt", row.names = FALSE, col.names = FALSE)
length(unique(intersect(x$Gene, clinvar_P_LP_only_snv_missense_tableized$SYMBOL))) #2451 actually 2450

p.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
length(unique(intersect(p.paralogous_var_align$Total_paralog_annotations$SYMBOL, clinvar_P_LP_tableized$SYMBOL)))

clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized$Variant_pos = paste(clinvar_P_LP_tableized$CHROM,clinvar_P_LP_tableized$POS, sep = " ")
paralog_data = left_join(x,clinvar_P_LP_tableized, by =  c("Variant_pos", "ID", "REF", "ALT", "Gene" = "SYMBOL"))
paralog_data = paralog_data[!is.na(paralog_data$Protein_position) & !(grepl("-",paralog_data$Protein_position)),]
length(unique(paralog_data$ID)) #24134


clinvar_B_LB_only_snv_missense_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_only_snv_missense_tableized = clinvar_B_LB_only_snv_missense_tableized[!is.na(clinvar_B_LB_only_snv_missense_tableized$Protein_position) &
                                                                                      !is.na(clinvar_B_LB_only_snv_missense_tableized$Amino_acids) &
                                                                                      !(grepl("-", clinvar_B_LB_only_snv_missense_tableized$Protein_position)) &
                                                                                      !(grepl("-", clinvar_B_LB_only_snv_missense_tableized$Amino_acids)) &
                                                                                      !(grepl("\\*", clinvar_B_LB_only_snv_missense_tableized$Amino_acids)) &
                                                                                      grepl("/", clinvar_B_LB_only_snv_missense_tableized$Amino_acids) &
                                                                                      str_count(clinvar_B_LB_only_snv_missense_tableized$REF) == str_count(clinvar_B_LB_only_snv_missense_tableized$ALT) &
                                                                                      str_count(clinvar_B_LB_only_snv_missense_tableized$Amino_acids) == 3,]

num_of_var_clinvar_B_LB_only_snv_missense_tableized = length(unique(clinvar_B_LB_only_snv_missense_tableized$ID)) #19112 actually 19039
num_of_genes_clinvar_B_LB_only_snv_missense_tableized = length(unique(clinvar_B_LB_only_snv_missense_tableized$SYMBOL)) #3007 actually 2987

total_num_of_genes_clinvar_P_LP_B_LB_only_snv_missense_tableized = length(unique(c(unique(clinvar_P_LP_only_snv_missense_tableized$SYMBOL), unique(clinvar_B_LB_only_snv_missense_tableized$SYMBOL)))) #4340 actually 4225
write.table(unique(c(unique(clinvar_P_LP_only_snv_missense_tableized$SYMBOL), unique(clinvar_B_LB_only_snv_missense_tableized$SYMBOL))),file="data/clinvar/4225_clinvar_genes_w_only_missense.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)

clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized$Variant_pos = paste(clinvar_B_LB_tableized$CHROM,clinvar_B_LB_tableized$POS, sep = " ")
y = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:147, sep = "")), quote="", na.strings=".", comment.char="#", stringsAsFactors = 0)
b.paralog_data = left_join(y,clinvar_B_LB_tableized, by =  c("Variant_pos", "ID", "REF", "ALT", "Gene" = "SYMBOL"))
b.paralog_data = b.paralog_data[!is.na(b.paralog_data$Protein_position) & !(grepl("-",b.paralog_data$Protein_position)) & str_count(b.paralog_data$REF) == str_count(b.paralog_data$ALT),]
length(unique(b.paralog_data$ID)) #14788


length(intersect(unique(y$Gene),unique(clinvar_B_LB_tableized_genes)))
length(unique(intersect(y$Gene, clinvar_B_LB_only_snv_missense_tableized$SYMBOL))) # 2210

length(unique(c(intersect(unique(x$Gene),unique(clinvar_P_LP_tableized_genes)), intersect(unique(y$Gene),unique(clinvar_B_LB_tableized_genes))))) #3123 here
write.table(unique(c(intersect(unique(x$Gene),unique(clinvar_P_LP_tableized_genes)), intersect(unique(y$Gene),unique(clinvar_B_LB_tableized_genes)))),file="data/clinvar/3123_genes_w_paralogues_w_aln_pos.txt", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
setdiff(intersect(unique(x$Gene),unique(clinvar_P_LP_tableized_genes)), intersect(unique(y$Gene),unique(clinvar_B_LB_tableized_genes)))


```
NB in clinvar_P_LP_genes_w_paralogs there are 3621 genes with paralogues but in the out_paraloc_paralogs file there are only 2451 genes. This is because there 1170 genes in clinvar_P_LP_genes_w_paralogs that are not in out_paraloc_paralogs because though there are variants in these genes they do not align to anything. There are also cases of over-lapping genes and so while one gene is reported the other might not. Also the same for some readthrough transcripts. Several genes with designation "AC######.#" go through non-sense mediated decay.

Reclassifying VUS:
```{r clinvar_table, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE, fig.width=7,fig.height=6,fig.cap="\\label{fig:clinvar_table}Total number of known P/LP variants and known B/LB variants predicted to be pathogenic"}
#noQC
p.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_paralogs2.noQC",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_tableized",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p2.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_paralogs2.para_con",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_tableized",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p3.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_paralogs2.all_con",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyVUS.out_paraloc_tableized",
                                              "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

p4.Total_paralog_annotations = p2.paralogous_var_align$Total_paralog_annotations[p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p4.Total_paralog_annotations = distinct(p4.Total_paralog_annotations)
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))
p4.Unique_variant_annotations = p4.Total_paralog_annotations[!is.na(p4.Total_paralog_annotations$ID.y),]
p4.Unique_variant_annotations = distinct(p4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p4.true_num_of_paralog_anno = length(p4.Unique_variant_annotations$Variant_pos)

p5.Total_paralog_annotations = p3.paralogous_var_align$Total_paralog_annotations[p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p5.Total_paralog_annotations = distinct(p5.Total_paralog_annotations)
p5.num_of_paralog_anno = sum(!is.na(p5.Total_paralog_annotations$ID.y))
p5.Unique_variant_annotations = p5.Total_paralog_annotations[!is.na(p5.Total_paralog_annotations$ID.y),]
p5.Unique_variant_annotations = distinct(p5.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p5.true_num_of_paralog_anno = length(p5.Unique_variant_annotations$Variant_pos)


df = data.frame(Type = factor(c("none", "conserved AA", "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants"), levels = c("none", "conserved AA", "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants")),
                reclassified_vus = c(p.paralogous_var_align$true_num_of_paralog_anno,p2.paralogous_var_align$true_num_of_paralog_anno,p3.paralogous_var_align$true_num_of_paralog_anno,p4.true_num_of_paralog_anno,p5.true_num_of_paralog_anno),
                bar = c("VUS","VUS","VUS","VUS","VUS"))

bar_chart = ggplot(df, aes(y=reclassified_vus,x=bar,fill=Type)) + geom_bar(stat="identity",position = "identity",width = 0.1) +
  scale_fill_manual(values = c("none"="#648FFF", "conserved AA"="#785EF0", "full conserved AA"="#DC267F", "conserved AA +\nconserved variants"="#FE6100", "full conserved AA +\nconserved variants"="#FFB000"), labels = c("none\n(n=3508)", "conserved AA\n(n=1871)", "full conserved AA\n(n=708)", "conserved AA +\nconserved variants\n(n=510)", "full conserved AA +\nconserved variants\n(n=185)")) +
  # scale_color_manual(values = c("#000000","#000000","#000000","#000000","#000000")) +
  theme(legend.title = element_text(size=25),
        legend.key.size = unit(6,"line"),
        # legend.position = c(0.85,0.91),
        legend.text = element_text(size=25),
        axis.text = element_text(size=25),
        axis.text.x = element_text(colour = "black"),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=25)) + ylab("Variants predicted pathogenic") +
  scale_y_continuous(breaks = c(0,1000,2000,3000,4000), limits = c(0,4100), expand = c(0,0)) +
  labs(fill="conservation stringency") + 
  # scale_x_discrete(expand = c(0,0.5)) +
  geom_text(aes(label=reclassified_vus), size = 10, color = "black")
ggsave("images/reclassified_vus_clinvar_set_bar_chart_with_legend.png", height = 14, width = 12)


bar_chart = ggplot(df, aes(y=reclassified_vus,x=bar,fill=Type)) + geom_bar(stat="identity",position = "identity",width = 0.5) +
  scale_fill_manual(values = c("none"="#648FFF", "conserved AA"="#785EF0", "full conserved AA"="#DC267F", "conserved AA +\nconserved variants"="#FE6100", "full conserved AA +\nconserved variants"="#FFB000")) + coord_flip() +
  # scale_color_manual(values = c("Pathogenic"="#B85450", "Benign"="#82B366"), labels = c("Pathogenic/\nLikely Pathogenic","Benign/\nLikely Benign")) +
  theme(legend.title = element_blank(),
        plot.margin = unit(c(1,1,1,1), "cm"),
        legend.key.size = unit(5,"line"),
        legend.position = "none",
        legend.text = element_text(size=20),
        axis.text = element_text(size=30),
        axis.text.y = element_text(colour = "black"),
        axis.title.y = element_blank(),
        axis.title.x = element_text(size=30)) + ylab("Variants predicted pathogenic") +
  scale_y_continuous(breaks = c(0,1000,2000,3000,4000), limits = c(0,4100), expand = c(0,0), position = "right")
  # scale_x_discrete(expand = c(0,0)) 
ggsave("images/reclassified_vus_clinvar_set_bar_chart.png", height = 3, width = 14)
```
127080 missense VUS in clinvar


## Para-Z scores
The filtering steps outlined above take a more binaray approach to taking account the conservativeness of amino acid positions in the alignments. They only consider if amino acids in question share the the same amino acid or not. The Para-Z scores defined by @Lal2017 on the other hand takes a more quantitative approach to this by representing a numeric integer value of how conserved each amino acid position is across the same paralogue family. As shown by **fig.** \ref{fig:paraz}, one can use the Para-Z scores as cutoff thresholds for how paralogue conserved alignments must be before considering annotations. The more stringent the cutoff the lower the false positive rate is, but as expected the lower the sensitivity is also. Regardless both methods validate the concept that the more conserved amino acid positions are when transfering annotation the more likely annotations will be true positives as one would expect.
\newpage

```{r paraz, message=FALSE, cache=TRUE,  results='asis', warning=FALSE, echo=FALSE, fig.width=4,fig.height=3,fig.cap="\\label{fig:paraz}ROC space plot of pathogenic/likely pathogenic clinvar variants predicted as pathogenic by paralogue annotation at varying degrees of Para-Z scores"}
Sensitivity_list = vector()
Sens_CI_list = vector()
FPR_list = vector()
con_table_list = matrix(, ncol = 2)
PPV_list = vector()
PPV_CI_list = vector()
P_value_list = vector()
Specificity_list = vector()
##Table##
paraz_set = t(data.frame(Variant,Total))
roc_data = data.frame()
####
for (i in seq(0,11)){
#noQC
#pathogenic to pathogenic
# p.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  p.paraz_var_align = Paralogous_var_align(paraz_cutoff = i,"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

#benign to pathogenic
# b.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  b.paraz_var_align = Paralogous_var_align(paraz_cutoff = i,"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

  paraz.conf = conf_matrix(p.paraz_var_align$true_num_of_paralog_anno, p.paraz_var_align$input_variants, b.paraz_var_align$true_num_of_paralog_anno, b.paraz_var_align$input_variants)
  Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
  Sens_CI_list = c(Sens_CI_list, 1.96*paraz.conf$Sensitivity_SE)
  FPR_list = c(FPR_list, paraz.conf$FPR)
  con_table_list = rbind(con_table_list, paraz.conf$con_table)
  PPV_list = c(PPV_list, paraz.conf$PPV)
  PPV_CI_list= c(PPV_CI_list, 1.96*paraz.conf$PPV_SE)
  P_value_list = c(P_value_list, paraz.conf$Pvalue$p.value)
  Specificity_list = c(Specificity_list, paraz.conf$Specificity)
  
  PA_threshold_row = c(format(paraz.conf$con_table[2], scientific = FALSE), format(paraz.conf$con_table[2,2], scientific = FALSE), round(paraz.conf$PPV, digits = 3), round(paraz.conf$Sensitivity, digits = 3), round(paraz.conf$Specificity,digits = 3), signif(paraz.conf$Pvalue$p.value, digits = 3))
  paraz_set = rbind(paraz_set, PA_threshold_row)
}
##Graph
# roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list, cutoff = seq(0,11)))
roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list))
roc_data$Type = "Para_z"

# colnames(roc_data) = c("Sensitivity", "FPR", "type")
# roc_data$type="Para_z"
roc_data = rbind(roc_data, c(as.numeric(conf$Sensitivity),as.numeric(conf$FPR)))
roc_data[13,3]="noQC"
roc_data = rbind(roc_data, c(as.numeric(conf2$Sensitivity),as.numeric(conf2$FPR)))
roc_data[14,3]="QC1"
roc_data = rbind(roc_data, c(as.numeric(conf3$Sensitivity),as.numeric(conf3$FPR)))
roc_data[15,3]="QC2"
roc_data = rbind(roc_data, c(as.numeric(conf4$Sensitivity),as.numeric(conf4$FPR)))
roc_data[16,3]="QC3"

# gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour=cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
# gg = gg + geom_abline(intercept = 0, slope = 1, linetype = "dotted")
# 
# ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = type)) + geom_point() + xlim(0,0.05) + ylim(0,0.25) + scale_color_gradient(low="blue", high="red", limits=c(0, 11))
gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = Type)) + geom_point(size=2) + xlim(0,0.05) + ylim(0,0.25) + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14, 
        hjust = 0.25)) +labs(title = "Receiver operating characteristic space plot of\nParalogue Annotation") + theme(axis.text = element_text(size = 13))
ggsave("images/ROC_plot.png")

# prc_data = do.call(rbind, Map(data.frame, Precision = PPV_list, Sensitivity = Sensitivity_list, cutoff = seq(0,11)))
prc_data = do.call(rbind, Map(data.frame, Precision = PPV_list, Precision_CI = PPV_CI_list, Sensitivity = Sensitivity_list, Sensitivity_CI = Sens_CI_list))
prc_data$Type = "Para_z"

prc_data = rbind(prc_data, c(as.numeric(conf$PPV),as.numeric(1.96*conf$PPV_SE),as.numeric(conf$Sensitivity),as.numeric(1.96*conf$Sensitivity_SE)))
prc_data[13,5]="noQC"
prc_data = rbind(prc_data, c(as.numeric(conf2$PPV),as.numeric(1.96*conf2$PPV_SE),as.numeric(conf2$Sensitivity),as.numeric(1.96*conf2$Sensitivity_SE)))
prc_data[14,5]="QC1"
prc_data = rbind(prc_data, c(as.numeric(conf3$PPV),as.numeric(1.96*conf3$PPV_SE),as.numeric(conf3$Sensitivity),as.numeric(1.96*conf3$Sensitivity_SE)))
prc_data[15,5]="QC2"
prc_data = rbind(prc_data, c(as.numeric(conf4$PPV),as.numeric(1.96*conf4$PPV_SE),as.numeric(conf4$Sensitivity),as.numeric(1.96*conf$Sensitivity_SE)))
prc_data[16,5]="QC3"

# gg2 = ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour=cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
# gg2 = gg2 + geom_abline(intercept = 0, slope = 1, linetype = "dotted")
# 
# ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour = cutoff)) + geom_point() + xlim(0.1,0.2) + ylim(0.85,1) + scale_color_gradient(low="blue", high="red", limits=c(0, 11))
gg2 = ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=5) + geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,width=5) +
  scale_shape_manual(values=c(15, 1, 16, 17, 18)) + xlim(0,0.25) + ylim(0.85,1)  + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(title = "Precision-Sensitivity space plot of Paralogue Annotation", 
    x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.5))
ggsave("images/PRC_plot.png")

prc_data2 = prc_data[13:16,]
# prc_data2 = rbind(prc_data2, c(0.619,0.032))
# prc_data2[5,3] = "removed by QC1"
# prc_data2 = rbind(prc_data2, c(0.931,0.083))
# prc_data2[6,3] = "removed by QC2"
# prc_data2 = rbind(prc_data2, c(0.981,0.058))
# prc_data2[7,3] = "removed by QC3"
gg2_5 = ggplot(prc_data2, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=.5) +
  scale_shape_manual(values=c(15, 16, 17, 18)) + xlim(0,0.25) + ylim(0.85,1)  + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(title = "Precision-Sensitivity space plot of Paralogue Annotation", 
    x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.5))
ggsave("images/PRC_with_removed_plot.png")



benign_data = data.frame(Precision = c(0.391,0.258,0.169,0.963), Precision_CI = c(0.0161531,0.02014389,0.02741531,0.0356178), Sensitivity = c(0.092,0.032,0.008,0.007), Sensitivity_CI = c(0.004658095,0.002811256,0.001446604,0.001341913),Type = c("noQC","QC1","QC2","QC3"))
gg3 = ggplot(benign_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) +
  geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  scale_shape_manual(values=c(15, 16, 17, 18)) + xlim(0,0.25) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
                                                                                                                                  plot.caption = element_text(vjust = 1),
                                                                                                                                  axis.title = element_text(size = 16), 
plot.title = element_text(size = 14)) + labs(title = "Precision-Sensitivity space plot of Paralogue Annotation", x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.5))
ggsave("images/benign_PRC_plot.png")


kable(paraz_set)
```

Can also apply Paraz scores after QC1

```{r paraz1, message=FALSE, cache=TRUE,  results='asis', warning=FALSE, echo=FALSE, fig.width=4,fig.height=3,fig.cap="\\label{fig:paraz}ROC space plot of pathogenic/likely pathogenic clinvar variants predicted as pathogenic by paralogue annotation at varying degrees of Para-Z scores"}
Sensitivity_list = vector()
Sens_CI_list = vector()
FPR_list = vector()
con_table_list = matrix(, ncol = 2)
PPV_list = vector()
PPV_CI_list = vector()
P_value_list = vector()
Specificity_list = vector()

#### PA methods with para_z setting to 0 in order to have same number of starting variants
p.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf = conf_matrix(p.paralogous_var_align$true_num_of_paralog_anno, p.paralogous_var_align$input_variants, b.paralogous_var_align$true_num_of_paralog_anno, b.paralogous_var_align$input_variants)
p2.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b2.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.paralogous_var_align$true_num_of_paralog_anno, p2.paralogous_var_align$input_variants, b2.paralogous_var_align$true_num_of_paralog_anno, b2.paralogous_var_align$input_variants)
p3.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b3.paralogous_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.paralogous_var_align$true_num_of_paralog_anno, p3.paralogous_var_align$input_variants, b3.paralogous_var_align$true_num_of_paralog_anno, b3.paralogous_var_align$input_variants)

p4.Total_paralog_annotations = p3.paralogous_var_align$Total_paralog_annotations[p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p4.Total_paralog_annotations = distinct(p4.Total_paralog_annotations)
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))
p4.Unique_variant_annotations = p4.Total_paralog_annotations[!is.na(p4.Total_paralog_annotations$ID.y),]
p4.Unique_variant_annotations = distinct(p4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p4.true_num_of_paralog_anno = length(p4.Unique_variant_annotations$Variant_pos)
b4.Total_paralog_annotations = b3.paralogous_var_align$Total_paralog_annotations[b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
b4.Total_paralog_annotations = distinct(b4.Total_paralog_annotations)
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))
b4.Unique_variant_annotations = b4.Total_paralog_annotations[!is.na(b4.Total_paralog_annotations$ID.y),]
b4.Unique_variant_annotations = distinct(b4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b4.true_num_of_paralog_anno = length(b4.Unique_variant_annotations$Variant_pos)
conf4 = conf_matrix(p4.true_num_of_paralog_anno, p3.paralogous_var_align$input_variants, b4.true_num_of_paralog_anno, b3.paralogous_var_align$input_variants)

p6.Total_paralog_annotations = p2.paralogous_var_align$Total_paralog_annotations[p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p6.Total_paralog_annotations = distinct(p6.Total_paralog_annotations)
p6.num_of_paralog_anno = sum(!is.na(p6.Total_paralog_annotations$ID.y))
p6.Unique_variant_annotations = p6.Total_paralog_annotations[!is.na(p6.Total_paralog_annotations$ID.y),]
p6.Unique_variant_annotations = distinct(p6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p6.true_num_of_paralog_anno = length(p6.Unique_variant_annotations$Variant_pos)
b6.Total_paralog_annotations = b2.paralogous_var_align$Total_paralog_annotations[b2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b2.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
b6.Total_paralog_annotations = distinct(b6.Total_paralog_annotations)
b6.num_of_paralog_anno = sum(!is.na(b6.Total_paralog_annotations$ID.y))
b6.Unique_variant_annotations = b6.Total_paralog_annotations[!is.na(b6.Total_paralog_annotations$ID.y),]
b6.Unique_variant_annotations = distinct(b6.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b6.true_num_of_paralog_anno = length(b6.Unique_variant_annotations$Variant_pos)
conf6 = conf_matrix(p6.true_num_of_paralog_anno, p2.paralogous_var_align$input_variants, b6.true_num_of_paralog_anno, b2.paralogous_var_align$input_variants)

Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
Total = c(dim(p.paralogous_var_align$input_variants)[1],dim(b.paralogous_var_align$input_variants)[1],"NA","NA","NA","NA")
##Table##
paraz_set = t(data.frame(Variant,Total))
roc_data = data.frame()
prc_data = data.frame()
####

for (i in seq(0,11)){
#noQC
#pathogenic to pathogenic
# p.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  p.paraz_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = i,"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

#benign to pathogenic
# b.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  b.paraz_var_align = Paralogous_var_align(Overlap = 0, paraz_cutoff = i,"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

  paraz.conf = conf_matrix(p.paraz_var_align$true_num_of_paralog_anno, p.paraz_var_align$input_variants, b.paraz_var_align$true_num_of_paralog_anno, b.paraz_var_align$input_variants)
  Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
  Sens_CI_list = c(Sens_CI_list, 1.96*paraz.conf$Sensitivity_SE)
  FPR_list = c(FPR_list, paraz.conf$FPR)
  con_table_list = rbind(con_table_list, paraz.conf$con_table)
  PPV_list = c(PPV_list, paraz.conf$PPV)
  PPV_CI_list= c(PPV_CI_list, 1.96*paraz.conf$PPV_SE)
  P_value_list = c(P_value_list, paraz.conf$Pvalue$p.value)
  Specificity_list = c(Specificity_list, paraz.conf$Specificity)
  
  PA_threshold_row = c(format(paraz.conf$con_table[2], scientific = FALSE), format(paraz.conf$con_table[2,2], scientific = FALSE), signif(paraz.conf$PPV, digits = 2), signif(paraz.conf$Sensitivity, digits = 2), signif(paraz.conf$Specificity,digits = 2), signif(paraz.conf$Pvalue$p.value, digits = 2))
  paraz_set = rbind(paraz_set, PA_threshold_row)
}
##Graph
# roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list, cutoff = seq(0,11)))
roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list))
roc_data$Type = "Para_z"

# colnames(roc_data) = c("Sensitivity", "FPR", "type")
# roc_data$type="Para_z"
roc_data = rbind(roc_data, c(as.numeric(conf$Sensitivity),as.numeric(conf$FPR)))
roc_data[13,3]="noQC"
roc_data = rbind(roc_data, c(as.numeric(conf2$Sensitivity),as.numeric(conf2$FPR)))
roc_data[14,3]="QC1"
roc_data = rbind(roc_data, c(as.numeric(conf3$Sensitivity),as.numeric(conf3$FPR)))
roc_data[15,3]="QC2"
roc_data = rbind(roc_data, c(as.numeric(conf4$Sensitivity),as.numeric(conf4$FPR)))
roc_data[16,3]="QC3"

# gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour=cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
# gg = gg + geom_abline(intercept = 0, slope = 1, linetype = "dotted")
# 
# ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = type)) + geom_point() + xlim(0,0.05) + ylim(0,0.25) + scale_color_gradient(low="blue", high="red", limits=c(0, 11))
gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = Type)) + geom_point(size=2) + xlim(0,0.05) + ylim(0,0.25) + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14, 
        hjust = 0.25)) +labs(title = "Receiver operating characteristic space plot of\nParalogue Annotation") + theme(axis.text = element_text(size = 13))
ggsave("images/ROC_plot_paraz_with_QC1.png")

# prc_data = do.call(rbind, Map(data.frame, Precision = PPV_list, Sensitivity = Sensitivity_list, cutoff = seq(0,11)))
prc_data = do.call(rbind, Map(data.frame, Precision = PPV_list, Precision_CI = PPV_CI_list, Sensitivity = Sensitivity_list, Sensitivity_CI = Sens_CI_list))
prc_data$Type = "conserved AA +\nPara_z"

prc_data = rbind(prc_data, c(as.numeric(conf$PPV),as.numeric(1.96*conf$PPV_SE),as.numeric(conf$Sensitivity),as.numeric(1.96*conf$Sensitivity_SE)))
prc_data[13,5]="none"
prc_data = rbind(prc_data, c(as.numeric(conf2$PPV),as.numeric(1.96*conf2$PPV_SE),as.numeric(conf2$Sensitivity),as.numeric(1.96*conf2$Sensitivity_SE)))
prc_data[14,5]="conserved AA"
prc_data = rbind(prc_data, c(as.numeric(conf3$PPV),as.numeric(1.96*conf3$PPV_SE),as.numeric(conf3$Sensitivity),as.numeric(1.96*conf3$Sensitivity_SE)))
prc_data[15,5]="full conserved AA"
prc_data = rbind(prc_data, c(as.numeric(conf4$PPV),as.numeric(1.96*conf4$PPV_SE),as.numeric(conf4$Sensitivity),as.numeric(1.96*conf4$Sensitivity_SE)))
prc_data[16,5]="full conseerved AA +\nconserved variants"
prc_data = rbind(prc_data, c(as.numeric(conf6$PPV),as.numeric(1.96*conf6$PPV_SE),as.numeric(conf6$Sensitivity),as.numeric(1.96*conf6$Sensitivity_SE)))
prc_data[17,5]="conserved AA +\nconserved variants"

# prc_data[18:21,5]=c("Sift", "MCap", "CADD", "Revel")

prc_data$Type = factor(prc_data$Type, levels = c("none", "conserved AA", "conserved AA +\nPara_z", "full conserved AA", "conserved AA +\nconserved variants", "full conseerved AA +\nconserved variants"))
# prc_data$Type = factor(prc_data$Type, levels = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1", "QC1+Para_Z", "QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

prc_data$Type[1:12] = "conserved AA +\nPara_z"
# gg2 = ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour=cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
# gg2 = gg2 + geom_abline(intercept = 0, slope = 1, linetype = "dotted")
# 
# ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour = cutoff)) + geom_point() + xlim(0.1,0.2) + ylim(0.85,1) + scale_color_gradient(low="blue", high="red", limits=c(0, 11))
cols = c("conserved AA +\nPara_z" = "#4D5656", "none" = "#58D68D", "conserved AA" = "#45B39D", "full conserved AA" = "#5DADE2", "conserved AA +\nconserved variants" = "#A569BD", "full conseerved AA +\nconserved variants" = "#EC7063")
shapes = c("conserved AA +\nPara_z" = 6, "none" = 15, "conserved AA" = 16, "full conserved AA" = 17, "conserved AA +\nconserved variants" = 18, "full conseerved AA +\nconserved variants" = 19)

# cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
# shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)

gg2 = ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  #  geom_point(size=4,data=prc_data[prc_data$Type=="conserved AA +\nPara_z",]) + 
  # geom_point(size=4,data=prc_data[prc_data$Type!="conserved AA +\nPara_z",]) + 
  
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=5) + 
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,width=5) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,0.25) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(y = "Precision", x = "Sensitivity") + 
  theme(axis.text = element_text(size = 13)) + 
  theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(legend.text=element_text(size=16), legend.title=element_text(size=16), legend.key.size = unit(3,"line")) +
  guides(shape = guide_legend(override.aes = list(size = 8))) +
  labs(shape="conservation stringency", color="conservation stringency")
  
 # theme(legend.text=element_text(size=25), axis.text = element_text(size = 16), axis.title.x=element_blank(), axis.title.y=element_blank()) + theme(plot.title = element_text(hjust = 0.25)) + theme(legend.key.size = unit(5,"line"))
ggsave("images/PRC_plot_paraz_with_QC1_and_QC3_no_QC2.png")

gg3 = ggplot(prc_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15) + 
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,0.2) + ylim(0.9,1)  + theme(plot.subtitle = element_text(vjust = 1), 
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 35), 
    plot.title = element_text(size = 14)) +labs(y = "Precision", x = "Sensitivity") + 
  theme(axis.text = element_text(size = 30)) + 
  theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(legend.text=element_text(size=16), legend.title=element_text(size=16), legend.key.size = unit(3,"line")) +
  guides(shape = guide_legend(override.aes = list(size = 8))) +
  labs(shape="conservation\nstringency", color="conservation\nstringency") + 
  theme(legend.position="none") 
ggsave("images/PRC_plot_paraz_with_QC1_and_QC3_no_QC2_for_poster.png", height = 10, width = 10)

write.table(paraz_set, file = "data/clinvar/paraz_set.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)

# kable(paraz_set)
```

## visual graph of main clinvar and para_z table
```{r plot_of_clinvar_paraz_set, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
clinvar_set_csv = read.csv("data/clinvar/clinvar_set.csv", sep=",", header = T)
clinvar_set_only_P = data.frame(bar = c("none", "conserved AA", "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants"),
           Pathogenic_var = clinvar_set_csv[2:6,1],
           Stage = "Pathogenic")
clinvar_set_only_P$Stage = factor(clinvar_set_only_P$Stage, levels = c("Pathogenic"))
clinvar_set_only_B = data.frame(bar = c("none", "conserved AA", "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants"),
           Pathogenic_var = clinvar_set_csv[2:6,2],
           Stage = "Benign")
clinvar_set_only_B$Stage = factor(clinvar_set_only_B$Stage, levels = c("Benign"))
Total_clinvar_set = rbind(clinvar_set_only_P,clinvar_set_only_B)
Total_clinvar_set$bar = factor(Total_clinvar_set$bar, levels = c("none", "conserved AA", "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants"))
# Total_clinvar_set[Total_clinvar_set$bar=="Total variants","Pathogenic_var"] = Total_clinvar_set[Total_clinvar_set$bar=="Total variants","Pathogenic_var"] - 10000

# clinvar_set_top = clinvar_set_csv[1:3,1:2]
# para_z_set_csv = read.csv("data/clinvar/paraz_set.csv", sep = ",", header = F)
# para_z_set_middle = para_z_set_csv[4:15,2:3]
# names(para_z_set_middle) = names(clinvar_set_top)
# row.names(para_z_set_middle) = paste("ParaZ_threshold_", seq(0,11), sep = "")
# clinvar_set_bottom = clinvar_set_csv[4:6,1:2]
# Overall_clinvar_paraz_set = rbind(clinvar_set_top,para_z_set_middle,clinvar_set_bottom)

Total_variants = data.frame(Clnsig = c("P/LP","B/LB"),
  Total.variants=t(clinvar_set_csv[1,1:2]))
Total_variants$Clnsig = factor(Total_variants$Clnsig, levels = c("P/LP","B/LB"))

Total_variants_bar = ggplot(Total_variants, aes(y=Total.variants, x=Clnsig, fill = Clnsig, colour = Clnsig)) + 
  geom_bar(stat="identity", width = 0.9) + 
  scale_fill_manual(values = c("P/LP"="#F8CECC", "B/LB"="#D5E8D4")) +
  scale_color_manual(values = c("P/LP"="#B85450", "B/LB"="#82B366")) +
  theme(aspect.ratio = 2/1,
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title.x = element_blank(),
        axis.text.x = element_text(colour = "black"),
        axis.title.y = element_text(size=20)) + ylab("Total variants") +
  scale_x_discrete(labels = c("Pathogenic/\nLikely pathogenic","Benign/\nLikely benign")) +
  scale_y_continuous(expand = c(0,0), limits = c(0,25000)) +
  geom_text(aes(label=Total.variants), position=position_dodge(width=0.9), vjust=-0.25, size = 7, color = "black")
ggsave("images/clinvar_total_variants_bar_chart.png")


bar_chart = ggplot(Total_clinvar_set, aes(y=Pathogenic_var, x=bar, fill = Stage, colour = Stage)) + geom_bar(stat="identity", position="dodge") +
  scale_fill_manual(values = c("Pathogenic"="#F8CECC", "Benign"="#D5E8D4"), labels = c("Pathogenic/\nLikely Pathogenic","Benign/\nLikely Benign")) +
  scale_color_manual(values = c("Pathogenic"="#B85450", "Benign"="#82B366"), labels = c("Pathogenic/\nLikely Pathogenic","Benign/\nLikely Benign")) +
  theme(legend.title = element_blank(),
        legend.key.size = unit(5,"line"),
        legend.position = c(0.81,0.89),
        legend.text = element_text(size=30),
        axis.text = element_text(size=35),
        axis.text.x = element_text(colour = "black", angle=35, hjust = 1),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=35, vjust = 6),
        plot.margin = unit(c(1,1,1,1), "cm")) + ylab("Variants predicted pathogenic") +
  scale_y_continuous(breaks = c(0,1000,2000,3000,4000), limits = c(0,4300), expand = c(0,0)) +
  geom_text(aes(label=Pathogenic_var), position=position_dodge(width=0.9), vjust=-0.25, size = 15, color = "black")
ggsave("images/clinvar_set_bar_chart.png", height = 14, width = 14)
  # scale_y_log10() 
  # scale_y_continuous(expand = c(0,0), breaks = c(0:4500,14000:24000))

All_in_one = Total_clinvar_set
All_in_one = rbind(
                   data.frame(bar="Total",Pathogenic_var=23894,Stage="Pathogenic"),
                   data.frame(bar="Total",Pathogenic_var=14617,Stage="Benign"),
                   Total_clinvar_set)

All_in_one_chart = ggplot(All_in_one, aes(y=Pathogenic_var, x=bar, fill = Stage, colour = Stage)) + geom_bar(stat="identity", position="dodge") +
  scale_fill_manual(values = c("Pathogenic"="#F8CECC", "Benign"="#D5E8D4"), labels = c("Pathogenic/\nLikely Pathogenic","Benign/\nLikely Benign")) +
  scale_color_manual(values = c("Pathogenic"="#B85450", "Benign"="#82B366"), labels = c("Pathogenic/\nLikely Pathogenic","Benign/\nLikely Benign")) +
  theme(legend.title = element_blank(),
        legend.key.size = unit(5,"line"),
        legend.position = c(0.91,0.91),
        legend.text = element_text(size=20),
        axis.text = element_text(size=25),
        axis.text.x = element_text(colour = "black", angle=45, hjust = 1),
        axis.title.x = element_blank(),
        axis.title.y = element_text(size=25)) + ylab("Variants predicted pathogenic") +
  scale_y_continuous(limits = c(0,25200), expand = c(0,0)) +
  geom_text(aes(label=Pathogenic_var), position=position_dodge(width=0.9), vjust=-0.25, size = 10, color = "black")
ggsave("images/clinvar_set_allinone_bar_chart.png", height = 14, width = 20)
```

## Case example: lack of paralogue annotation on 8 sarcomeric genes subset and use of etiological fractions as alternate validation 
```{r echo = FALSE, cache=TRUE, results='asis', warning=FALSE}
#noQC
# p.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# b.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# conf = conf_matrix(p.sarco_var_align$num_of_paralog_anno, p.sarco_var_align$paralog_data, b.sarco_var_align$num_of_paralog_anno, b.sarco_var_align$paralog_data)

p.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf = conf_matrix(p.sarco_var_align$true_num_of_paralog_anno, p.sarco_var_align$paralog_data, b.sarco_var_align$true_num_of_paralog_anno, b.sarco_var_align$paralog_data)


#para_con
# p2.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# b2.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# conf2 = conf_matrix(p2.sarco_var_align$num_of_paralog_anno, p2.sarco_var_align$paralog_data, b2.sarco_var_align$num_of_paralog_anno, b2.sarco_var_align$paralog_data)
p2.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b2.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.sarco_var_align$true_num_of_paralog_anno, p2.sarco_var_align$paralog_data, b2.sarco_var_align$true_num_of_paralog_anno, b2.sarco_var_align$paralog_data)

var_removed = var_rem_matrix(conf$con_table, conf2$con_table, p2.sarco_var_align$paralog_data, b2.sarco_var_align$paralog_data)

#all_con
# p3.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# b3.sarco_var_align = Subset_var_align_gene(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# conf3 = conf_matrix(p3.sarco_var_align$num_of_paralog_anno, p3.sarco_var_align$paralog_data, b3.sarco_var_align$num_of_paralog_anno, b3.sarco_var_align$paralog_data)
p3.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b3.sarco_var_align = Paralogous_var_align(Subset = c("Gene","MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.sarco_var_align$true_num_of_paralog_anno, p3.sarco_var_align$paralog_data, b3.sarco_var_align$true_num_of_paralog_anno, b3.sarco_var_align$paralog_data)

var_removed2 = var_rem_matrix(conf2$con_table, conf3$con_table, p3.sarco_var_align$paralog_data, b3.sarco_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.sarco_var_align$Total_paralog_annotations[p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
p4.Total_paralog_annotations = distinct(p4.Total_paralog_annotations)
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))
p4.Unique_variant_annotations = p4.Total_paralog_annotations[!is.na(p4.Total_paralog_annotations$ID.y),]
p4.Unique_variant_annotations = distinct(p4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p4.true_num_of_paralog_anno = length(p4.Unique_variant_annotations$Variant_pos)

#benign to pathogenic
b4.Total_paralog_annotations = b3.sarco_var_align$Total_paralog_annotations[b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
b4.Total_paralog_annotations = distinct(b4.Total_paralog_annotations)
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))
b4.Unique_variant_annotations = b4.Total_paralog_annotations[!is.na(b4.Total_paralog_annotations$ID.y),]
b4.Unique_variant_annotations = distinct(b4.Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b4.true_num_of_paralog_anno = length(b4.Unique_variant_annotations$Variant_pos)

conf4 = conf_matrix(p4.true_num_of_paralog_anno, p3.sarco_var_align$paralog_data, b4.true_num_of_paralog_anno, b3.sarco_var_align$paralog_data)

var_removed3 = var_rem_matrix(conf3$con_table, conf4$con_table, p3.sarco_var_align$paralog_data, b3.sarco_var_align$paralog_data)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificty","P value")
Total = c(dim(p.sarco_var_align$input_variants)[1],dim(b.sarco_var_align$input_variants)[1],"NA","NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),round(digits = 3, conf$PPV),round(digits = 3,conf$Sensitivity),round(digits = 3,conf$Specificity),signif(digits = 3,conf$Pvalue$p.value))
Variants_remaining_after_PA_QC1 = c(format(conf2$con_table[2], scientific = FALSE),format(conf2$con_table[4], scientific = FALSE),round(digits = 3,conf2$PPV), round(digits = 3,conf2$Sensitivity),round(conf2$Specificity,digits = 3), signif(digits = 3,conf2$Pvalue$p.value))

Variants_removed_after_PA_QC1 = c(var_removed$con_table[2],var_removed$con_table[4],round(digits = 3,var_removed$PPV),round(digits = 3,var_removed$Sensitivity),round(digits = 3,var_removed$Specificity),signif(digits = 3,var_removed$Pvalue$p.value))
Variants_remaining_after_PA_QC2 = c(format(conf3$con_table[2], scientific = FALSE),format(conf3$con_table[4], scientific = FALSE),round(digits = 3,conf3$PPV),round(digits = 3,conf3$Sensitivity),round(digits = 3,conf3$Specificity),signif(digits = 3,conf3$Pvalue$p.value))
Variants_removed_after_PA_QC2 = c(sapply(c(var_removed2$con_table[2],var_removed2$con_table[4]),format,scientific=FALSE),round(digits = 3,var_removed2$PPV),round(digits = 3,var_removed2$Sensitivity),round(digits = 3,var_removed2$Specificity),signif(digits = 3,var_removed2$Pvalue$p.value))
Variants_remaining_after_PA_QC3 = c(format(conf4$con_table[2], scientific = FALSE),format(conf4$con_table[4], scientific = FALSE),round(digits = 3,conf4$PPV),round(digits = 3,conf4$Sensitivity),round(digits = 3,conf4$Specificity),signif(digits = 3,conf4$Pvalue$p.value))
Variants_removed_after_PA_QC3 = c(var_removed3$con_table[2],var_removed3$con_table[4],round(digits = 3,var_removed3$PPV),round(digits = 3,var_removed3$Sensitivity),round(digits = 3,var_removed3$Specificity),signif(digits = 3,var_removed3$Pvalue$p.value))

sarcomeric_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )
sarcomeric_set_t = t(sarcomeric_set)
rownames(sarcomeric_set_t)  = c("", "Total variants", "Paralogue Annotation", "Variants remaining after QC1", "Variants removed from QC1", "Variants remaining after QC2", "Variants removed after QC2", "Variants remaining after QC3", "Variants removed after QC3")
kable(sarcomeric_set_t)

```


There are however limitations to this current framework. These can be listed as the following criteria: 1)the reliance on genes with paralogues; 2) for those paralogues to have pathogenic variants; and 3) for the paralogous variants to be aligned to corresponding equivalent positions. This is not always the case. 

As a specific example, consider Hypertrophic Cardiomyopathy (HCM) and the 8 sarcomeric genes commonly associated with the genetic basis of HCM: MYH7; MYBPC3; TNNT2; TPM1; MYL2; MYL3; TNNI3; and ACTC1.

Annotating only these 8 sarcomeric genes with the whole clinvar P/LP dataset as before did not provide many annotations - without any quality control there were only `r sarcomeric_set$Paralogue_Annotation_no_QC[1]` P/LP variants and `r sarcomeric_set$Paralogue_Annotation_no_QC[2]` B/LP variants predicted to be pathogenic.

This could suggest either PA does not perform well on sarcomeric genes for the reasons stated above (paralogues to sarcomeric genes are not involed in disease) or that there is a lack of data - given more pathogenic variants to annotate with would certainly increase the likelihood of paralogous alignments. 

\newpage

```{r sarco_para, cache=TRUE, echo=FALSE, message=FALSE, warning=FALSE, fig.width=7,fig.height=6,fig.cap="\\label{fig:sarco_para}Number of paralogues each sarcomeric gene has"}
ensembl = useMart("ENSEMBL_MART_ENSEMBL", host = "jul2018.archive.ensembl.org")
ens_human <- useDataset("hsapiens_gene_ensembl",mart=ensembl)
gene_id = c("MYH7", "MYBPC3", "TNNT2", "TPM1", "MYL2", "MYL3", "TNNI3", "ACTC1")

sarcomeric_genes = getBM(attributes = c("external_gene_name",
                                                  "hsapiens_paralog_associated_gene_name"), 
                                   filters = c("biotype", "with_hsapiens_paralog", "external_gene_name"), 
                                   values = list("protein_coding", TRUE, gene_id), mart = ens_human)

clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)

sarcomeric_variants = data.frame(Doubles=double(),
                                         Ints=integer(),
                                         Factors=factor(),
                                         Logicals=logical(),
                                         Characters=character(),
                                         stringsAsFactors=FALSE)
count = 1
for (variant in clinvar_P_LP_tableized$SYMBOL){
  if (variant %in% gene_id){
    sarcomeric_variants = rbind(sarcomeric_variants, clinvar_P_LP_tableized[count,])
  }
  count = count + 1
}

sarcomeric_paralog_variants = data.frame(Doubles=double(),
                                         Ints=integer(),
                                         Factors=factor(),
                                         Logicals=logical(),
                                         Characters=character(),
                                         stringsAsFactors=FALSE)
count = 1
for (variant in clinvar_P_LP_tableized$SYMBOL){
  if (variant %in% sarcomeric_genes$hsapiens_paralog_associated_gene_name){
    sarcomeric_paralog_variants = rbind(sarcomeric_paralog_variants, clinvar_P_LP_tableized[count,])
  }
  count = count + 1
}

# sarcomeric_genes_paralogs = spread(sarcomeric_genes, key = "hsapiens_paralog_associated_gene_name", "hsapiens_paralog_associated_gene_name")

sarcomeric_genes_paralogs = sarcomeric_genes %>%
  group_by(external_gene_name) %>%
  summarise(total = toString(hsapiens_paralog_associated_gene_name)) %>%
  ungroup()



kable(sarcomeric_genes_paralogs, "latex") %>%
  kable_styling(latex_options = "scale_down")
#kable(sarcomeric_paralog_variants)
```

Looking at how many paralogues the 8 sarcomeric genes have (table `sarcomeric_genes_paralogs`), there are at least 2 for each gene with MYH7 having the most - 13. This satisfies the first criteria.

In the clinvar P/LP dataset, there are `r length(sarcomeric_variants$CHROM)` known P/LP variants that lie in the 8 sarcomeric genes. But taking only the paralogues of these genes, there are only `r length(sarcomeric_paralog_variants$CHROM)` variants. Assuming the clinvar dataset is complete and considering it in isolation, this would suggest that in comparison to variants in the main 8 sarcomeric genes involved in HCM, their associated paralogous variants are not as frequently involved in disease. This still does however statisfy the second criteria.

Therefore, the third criteria is where the lack of as many annotations arises. Having additional data for both known reference variants to transfer annotations from and more query variants to transfer annotations to may resolve this as there would intuitively be more alignments available to transfer annotations. Collecting additional known pathogenic variants can be difficult as that would require established interpretation and verification of such variants. 

<!--Data in below code chunk too big. Unable to KnitR. Have to hard write numbers in paragraph.-->
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}
#noQC
#all missense data
# p.art.paralogous_var_align = Paralogous_var_align_compressed("data/all_possible_mutations/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_paralogs2.noQC.gz", "data/all_possible_mutations/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_tableized.gz", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
gene_subset = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
p.art.paralogous_var_align = Paralogous_var_align(Subset = c("Gene", gene_subset), "data/all_possible_mutations/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_paralogs2.noQC.gz", "data/all_possible_mutations/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_tableized.gz", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")



# p.art_Total_paralog_annotations = dplyr::filter(p.art.paralogous_var_align$Total_paralog_annotations, Gene %in% gene_subset)  
# p.art_num_of_paralog_anno = p.art_Total_paralog_annotations[!is.na(p.art_Total_paralog_annotations$ID.y),]
# 
# p.art_paralog_data = dplyr::filter(p.art.paralogous_var_align$paralog_data, Gene %in% gene_subset)

# patho_art_var = p.art.paralogous_var_align$Total_paralog_annotations[!is.na(p.art.paralogous_var_align$Total_paralog_annotations$ID.y),]
```

But querying every possible missense variant at all positions in the 8 sarcomeric genes can be done (30607 total variants). Doing so, predicts 1259 variants to pathogenic - a 78.7 fold increase. Though validating how many additional predictions are true positives would require additional known pathogenicity data.

```{r eval = FALSE, cache=TRUE, warning = FALSE}
##case data VEP##
case.paralogous_var_align = Paralogous_var_align("data/case_and_control/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_paralogs2.noQC", "data/case_and_control/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
patho_case_var = case.paralogous_var_align$Total_paralog_annotations[!is.na(case.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_case_var_IDs = patho_case_var$ID.x
# write(patho_case_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_case_var_IDs.txt", sep = "\t", ncolumns = 1)
case.data = read.csv("data/case_and_control/HCM_missense_LMM_OMGL_chr.csv", sep = ",") #only rare variants in the diseased cohort
patho_case.data = case.data[case.data$mut_id %in% patho_case_var_IDs,]
case_cases = sum(patho_case.data$sum.vs_case_count.)
Total_case_cases = sum(case.data$sum.vs_case_count.)

All_variant_case.cases = data.frame(genes = c("MYBPC3","MYH7","TNNT2","TNNI3","TPM1","ACTC1","MYL2","MYL3"), cases = c(6179,6112,6103,6047,4447,4185,4185,4185))

##control data VEP##
control.paralogous_var_align = Paralogous_var_align("data/case_and_control/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_paralogs2.noQC", "data/case_and_control/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") #only rare variants in the control cohort
patho_control_var = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_control_var_IDs = patho_control_var$ID.x
# write(patho_control_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_control_var_IDs.txt", sep = "\t", ncolumns = 1)
control.data = read.csv("data/case_and_control/ExAC_missense_sarcomeric_chr.csv", sep = ",")
patho_control.data = control.data[control.data$mut_id %in% patho_control_var_IDs,]
control_cases = sum(patho_control.data$mut_exac_count)
Total_control_cases = sum(control.data$mut_exac_count)

All_variant_control.cases = data.frame(genes = c("MYH7","TNNT2","TNNI3","MYBPC3","TPM1","ACTC1","MYL2","MYL3"), cases = c(60483,56800,53377,45646,58651,60271,60554,60611))

##Gene by gene analysis##
# example code for only ACTC1
# ACTC1.case.data = case.data[case.data$gene_symbol=="ACTC1",]
# patho_ACTC1.case.data = ACTC1.case.data[ACTC1.case.data$mut_id %in% patho_case_var_IDs,]
# patho_ACTC1.case.cases = sum(patho_ACTC1.case.data$sum.vs_case_count.)
# Total_ACTC1.case.cases = sum(ACTC1.case.data$sum.vs_case_count.)
#
# ACTC1.control.data = control.data[control.data$gene_symbol=="ACTC1",]
# patho_ACTC1.control.data = ACTC1.control.data[ACTC1.control.data$mut_id %in% patho_control_var_IDs,]
# patho_ACTC1.control.cases = sum(patho_ACTC1.control.data$mut_exac_count)
# Total_ACTC1.control.cases = sum(ACTC1.control.data$mut_exac_count)
# calc_EF(Total_ACTC1.case.cases,All_variant_case.cases[All_variant_case.cases$genes=="ACTC1","cases"],Total_ACTC1.control.cases,All_variant_control.cases[All_variant_control.cases$genes=="ACTC1","cases"]) #all rare variants
# calc_EF(patho_ACTC1.case.cases,All_variant_case.cases[All_variant_case.cases$genes=="ACTC1","cases"],patho_ACTC1.control.cases,All_variant_control.cases[All_variant_control.cases$genes=="ACTC1","cases"]) #rare variants predicted to be pathogenic by PA 

# x = case_control_gene_split(case.data, patho_case_var_IDs)
# y = case_control_gene_split(control.data, patho_control_var_IDs, "ACTC1")

# gene_EFs = list()

gene_EFs = data.frame()
for (gene in c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")){
# for (gene in c("MYBPC3")){
  cases = case_control_gene_split(case.data, patho_case_var_IDs, gene)
  controls = case_control_gene_split(control.data, patho_control_var_IDs, gene)
  all_rare_vars = calc_EF(cases$Total_gene_cases,
                          All_variant_case.cases[All_variant_case.cases$genes==gene,"cases"]-cases$Total_gene_cases,
                          controls$Total_gene_cases,
                          All_variant_control.cases[All_variant_control.cases$genes==gene,"cases"]-controls$Total_gene_cases)
  pa_predicted_rare_vars = calc_EF(cases$pa_predicted_gene_cases,
                                   All_variant_case.cases[All_variant_case.cases$genes==gene,"cases"]-cases$pa_predicted_gene_cases,
                                   controls$pa_predicted_gene_cases,
                                   All_variant_control.cases[All_variant_control.cases$genes==gene,"cases"]-controls$pa_predicted_gene_cases)
  # gene_EFs[[gene]] = list("all_rare_vars" = all_rare_vars, "pa_predicted_rare_vars" = pa_predicted_rare_vars)
  # tmp_df = data.frame(gene,toString(all_rare_vars$OR),toString(all_rare_vars$OR_CI),toString(all_rare_vars$EF),toString(all_rare_vars$EF_CI),toString(pa_predicted_rare_vars$OR),toString(pa_predicted_rare_vars$OR_CI),toString(pa_predicted_rare_vars$EF),toString(pa_predicted_rare_vars$EF_CI))
  tmp_df = data.frame(gene,cell_format(all_rare_vars$OR,all_rare_vars$OR_CI),cell_format(all_rare_vars$EF,all_rare_vars$EF_CI),cell_format(pa_predicted_rare_vars$OR,pa_predicted_rare_vars$OR_CI),cell_format(pa_predicted_rare_vars$EF,pa_predicted_rare_vars$EF_CI))
  gene_EFs = dplyr::bind_rows(gene_EFs,tmp_df)
}
##Overall##
all_rare_vars = calc_EF(Total_case_cases,
                        max(All_variant_case.cases$cases)-Total_case_cases,
                        Total_control_cases,
                        max(All_variant_control.cases$cases)-Total_control_cases)
pa_predicted_rare_vars = calc_EF(case_cases,
                                 max(All_variant_case.cases$cases)-case_cases,
                                 control_cases,
                                 max(All_variant_control.cases$cases)-control_cases)
gene_EFs = dplyr::bind_rows(gene_EFs,data.frame(gene="Overall",
                                                cell_format(all_rare_vars$OR,all_rare_vars$OR_CI),
                                                cell_format(all_rare_vars$EF,all_rare_vars$EF_CI),
                                                cell_format(pa_predicted_rare_vars$OR,pa_predicted_rare_vars$OR_CI),
                                                cell_format(pa_predicted_rare_vars$EF,pa_predicted_rare_vars$EF_CI)
                                                )
                            )
# gene_EFs = dplyr::bind_rows(gene_EFs,data.frame(gene="Overall",
#                                                 # sapply(c(
#                                                 toString(all_rare_vars$OR),
#                                                 toString(all_rare_vars$OR_CI),
#                                                 toString(all_rare_vars$EF),
#                                                 toString(all_rare_vars$EF_CI),
#                                                 toString(pa_predicted_rare_vars$OR),
#                                                 toString(pa_predicted_rare_vars$OR_CI),
#                                                 toString(pa_predicted_rare_vars$EF),
#                                                 toString(pa_predicted_rare_vars$EF_CI)
#                                                 # ),format,nsmall=3)
#                                                 )
#                             )
colnames(gene_EFs) = c("Gene","All rare variants OR","All rare variants EF","Predicted pathogenic OR","Predicted pathogenic EF")
write.table(gene_EFs, file = "data/case_and_control/gene_EFs.csv", na="", row.names = FALSE, col.names = TRUE, sep = ",", quote=FALSE)

```

However, it must be noted that even in genes with few predictions, paralogue annotation can still be functional and such predictions applicable. In the case of the 8 sarcomeric genes, one can still validate the use of paralogue annotation in a case control study.

Cases with positive (bad) outcome
Number in exposed group: 	
39
 	
Number in control group: 	
28
 	
Cases with negative (good) outcome
Number in exposed group: 	
41404
 	
Number in control group: 	
456365

Results
Odds ratio 	15.3524
95 % CI:	9.4468 to 24.9499
z statistic	11.024
Significance level	P < 0.0001

EFs show that even with few predictions, PA still works.

For example in the 8 sarcomeric genes involved in HCM [MYH7, MYBPC3, TNNT2, TPM1, MYL2, MYL3, TNNI3, ACTC1], taking MYH7 there were not many paralogous alignments.   since most paralogues of HCM disease genes are no.

Infact performing the analysis on all possible missesnse mutations for these set of genes still shows a lack of annotation...

Hence we calculated EFs in order to see for those few variants that are predicted to be pathogenic, how often do they appear to be causative of disease in a disease cohort case control study. Segway to HCM validation.

\newpage
## Calculations of EFs for LQTS and BrS
There are 15 genes that are associated with LQTS but the main 3 are: KCNH2; KCNQ1; and SCN5A.

The other 12 are: AKAP9; ANK2; CACNA1C; CALM1; CALM2;	CAV3;	KCNE1; KCNE2; KCNJ2; KCNJ5; SCN4B; and SNTA1. [@alders2018long]

For the OMGL data, Total disease cases would be considered as all variants in the spreadsheet. Need to first filter out non missense variants. Then take rare variants (Rare variants defined as https://www.nature.com/articles/gim201726; disease specific - in this case its LQTS; each disease has different threshold of "rarity") and variants that are predicted pathogenic by PA. Look at the same genes in gnomad, take variants from only exome (ask roddy - basically case data can only be extracted from vcfs https://gnomad.broadinstitute.org/downloads or from older version of gnomad browser). Extract only rare variants and ones predicted pathogenic by PA. Calculate EFs from this.

The Total case counts for gnomad would be the mean of all allele numbers of respective variants being considered divided by 2 (due to homo/heterozygous). I.e. for each variant take the AN number from gnomad vcf and sum them together, take the mean and divide by 2.

Using https://www.nature.com/articles/gim201726/tables/1 and using the upper CI level, e.g. for LQTS = 8.2 x 10−6. If the maximum filtering allele frequency (max out of all the sub populations, i.e. non-global) is smaller than 8.2 x 10-6 then that variant is considered "rare"

OMGL LQTS data - data incomplete, require case counts/allele numbers
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}
#Each line is a single patient count with the variant. exacAllelecount is the cell that has the same variant exac count
#For any variants where exac allele count is NA, just put as 0? Check this assumption

KCNQ1_data_ref = read.csv("data/LQTS/KCNQ1_JamesWare_270117.csv", sep = ",")
colnames(KCNQ1_data_ref)[1] = "gene"
KCNQ1_data = KCNQ1_data_ref[KCNQ1_data_ref$codingEffect == "missense",]
KCNQ1_gNomen_count = as.data.frame(table(KCNQ1_data$gNomen))
KCNQ1_gNomen_count = KCNQ1_gNomen_count[KCNQ1_gNomen_count$Freq > 0,]
colnames(KCNQ1_gNomen_count)[2] = "sum.vs_case_count"
KCNQ1_data = left_join(KCNQ1_data, KCNQ1_gNomen_count, by = c("gNomen" = "Var1"))
KCNQ1_data$exacAlleleCount[is.na(KCNQ1_data$exacAlleleCount)] = 0
unique_KCNQ1_gNomen = KCNQ1_data[match(unique(KCNQ1_data$gNomen), KCNQ1_data$gNomen),]

KCNH2_data_ref = read.csv("data/LQTS/KCNH2_JamesWare_270117.csv", sep = ",") 
# KCNH2_data = KCNH2_data[,c("gene", "chrom", "codingEffect", "assembly", "gDNAstart", "gDNAend", "pNomen", "pathogenicityClass", "rsId", "exacQuality", "exacFilter", "exacAlleleCount", "exacAllFreq", "exacAFRFreq", "exacAMRFreq", "exacEASFreq", "exacSASFreq", "exacNFEFreq", "exacFINFreq", "exacOTHFreq", "exacDP")]
KCNH2_data_ref = KCNH2_data_ref[,!(names(KCNH2_data_ref) %in% setdiff(colnames(KCNH2_data_ref), colnames(KCNQ1_data_ref)))]
KCNH2_data = KCNH2_data_ref[KCNH2_data_ref$codingEffect == "missense",]
KCNH2_gNomen_count = as.data.frame(table(KCNH2_data$gNomen))
KCNH2_gNomen_count = KCNH2_gNomen_count[KCNH2_gNomen_count$Freq > 0,]
colnames(KCNH2_gNomen_count)[2] = "sum.vs_case_count"
KCNH2_data = left_join(KCNH2_data, KCNH2_gNomen_count, by = c("gNomen" = "Var1"))
KCNH2_data$exacAlleleCount[is.na(KCNH2_data$exacAlleleCount)] = 0
unique_KCNH2_gNomen = KCNH2_data[match(unique(KCNH2_data$gNomen), KCNH2_data$gNomen),]

SCN5A_data_ref = read.csv("data/LQTS/SCN5A_JamesWare_270117.csv", sep = ",")
colnames(SCN5A_data_ref)[1] = "gene"
SCN5A_data_ref = SCN5A_data_ref[,!(names(SCN5A_data_ref) %in% setdiff(colnames(SCN5A_data_ref), colnames(KCNQ1_data_ref)))]
SCN5A_data = SCN5A_data_ref[SCN5A_data_ref$codingEffect == "missense",]
SCN5A_gNomen_count = as.data.frame(table(SCN5A_data$gNomen))
SCN5A_gNomen_count = SCN5A_gNomen_count[SCN5A_gNomen_count$Freq > 0,]
colnames(SCN5A_gNomen_count)[2] = "sum.vs_case_count"
SCN5A_data = left_join(SCN5A_data, SCN5A_gNomen_count, by = c("gNomen" = "Var1"))
SCN5A_data$exacAlleleCount[is.na(SCN5A_data$exacAlleleCount)] = 0
unique_SCN5A_gNomen = SCN5A_data[match(unique(SCN5A_data$gNomen), SCN5A_data$gNomen),]


Total_var = do.call("rbind", list(unique_KCNH2_gNomen, unique_KCNQ1_gNomen, unique_SCN5A_gNomen))
vcf_Total_var = Total_var[,c("chrom", "gDNAstart", "gNomen")]
vcf_Total_var$gNomen = as.character(vcf_Total_var$gNomen)
vcf_Total_var$REF = sapply(strsplit(vcf_Total_var$gNomen, ">"), "[", 1)
vcf_Total_var$REF = substr(vcf_Total_var$REF, nchar(vcf_Total_var$REF), nchar(vcf_Total_var$REF))
vcf_Total_var$ALT = sapply(strsplit(vcf_Total_var$gNomen, ">"), "[", 2)
vcf_Total_var$QUAl = "."
vcf_Total_var$FILTER = "."
vcf_Total_var$INFO = "."
colnames(vcf_Total_var)[1:3] = c("#CHROM", "POS", "ID")
write.table(vcf_Total_var, file = "data/LQTS/All_variants.vcf", sep = "\t", quote = FALSE, row.names = FALSE)
```

Ackerman data from ICC_Mutations
LQTS 
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/LQTS/Mayo_LQTS_variants_for_vcf_convertion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"
write.table(rare_disease_cohort_variants, file = "data/LQTS/rare_disease_cohort_variants.vcf", na="", row.names = FALSE, col.names = TRUE, sep = "\t", quote=FALSE) #RUN PA ON THIS FILE


###disease cohort
disease_data_ref = read.csv("data/LQTS/Mayo_LQTS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])
#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"), cases = c(2500,2500,2500,2500,2500))
#individual cases for rare variants
KCNE1_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNE2_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNH2_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNH2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNQ1_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNQ1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/LQTS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/LQTS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x,]
KCNE1_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE1" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNE2_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE2" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNH2_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNH2" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNQ1_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNQ1" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))

###control cohort
control_data_ref = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_with_customIDs.csv", sep = ",", stringsAsFactors = 0)
control_data_ref  = control_data_ref[control_data_ref$Gene != "Gene",] #remove the extra header lines from when concatenated the separate gene csv files together

gnomad_total_variants$X.CHROM = as.character(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.character(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID

KCNE1_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNE1",]
KCNE2_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNE2",]
KCNH2_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNH2",]
KCNQ1_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNQ1",]
SCN5A_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "SCN5A",]
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"), 
  cases = c(
    mean(as.numeric(KCNE1_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNE2_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNH2_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNQ1_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants
KCNE1_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE1" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNE2_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE2" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNH2_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNH2" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNQ1_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNQ1" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "SCN5A" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_paralogs2.noQC", "data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x,] 
KCNE1_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE1" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNE2_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE2" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNH2_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNH2" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNQ1_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNQ1" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "SCN5A" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))


###Calculating EFs
KCNE1_rare_EF = calc_EF(
  KCNE1_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_disease_rare_variant_cases,
  KCNE1_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_control_rare_variant_cases
  )
KCNE1_pa_EF = calc_EF(
  KCNE1_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_disease_pa_variant_cases,
  KCNE1_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_control_pa_variant_cases
  )

KCNE2_rare_EF = calc_EF(
  KCNE2_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_disease_rare_variant_cases,
  KCNE2_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_control_rare_variant_cases
  )
KCNE2_pa_EF = calc_EF(
  KCNE2_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_disease_pa_variant_cases,
  KCNE2_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_control_pa_variant_cases
  )

KCNH2_rare_EF = calc_EF(
  KCNH2_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_disease_rare_variant_cases,
  KCNH2_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_control_rare_variant_cases
  )
KCNH2_pa_EF = calc_EF(
  KCNH2_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_disease_pa_variant_cases,
  KCNH2_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_control_pa_variant_cases
  )

KCNQ1_rare_EF = calc_EF(
  KCNQ1_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_disease_rare_variant_cases,
  KCNQ1_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_control_rare_variant_cases
  )
KCNQ1_pa_EF = calc_EF(
  KCNQ1_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_disease_pa_variant_cases,
  KCNQ1_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_control_pa_variant_cases
  )

SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"),
  rare_variant_OR = c(cell_format(KCNE1_rare_EF$OR, KCNE1_rare_EF$OR_CI),
                      cell_format(KCNE2_rare_EF$OR, KCNE2_rare_EF$OR_CI),
                      cell_format(KCNH2_rare_EF$OR, KCNH2_rare_EF$OR_CI),
                      cell_format(KCNQ1_rare_EF$OR, KCNQ1_rare_EF$OR_CI),
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(cell_format(KCNE1_rare_EF$EF, KCNE1_rare_EF$EF_CI),
                      cell_format(KCNE2_rare_EF$EF, KCNE2_rare_EF$EF_CI),
                      cell_format(KCNH2_rare_EF$EF, KCNH2_rare_EF$EF_CI),
                      cell_format(KCNQ1_rare_EF$EF, KCNQ1_rare_EF$EF_CI),
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(cell_format(KCNE1_pa_EF$OR, KCNE1_pa_EF$OR_CI),
                    cell_format(KCNE2_pa_EF$OR, KCNE2_pa_EF$OR_CI),
                    cell_format(KCNH2_pa_EF$OR, KCNH2_pa_EF$OR_CI),
                    cell_format(KCNQ1_pa_EF$OR, KCNQ1_pa_EF$OR_CI),
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(cell_format(KCNE1_pa_EF$EF, KCNE1_pa_EF$EF_CI),
                    cell_format(KCNE2_pa_EF$EF, KCNE2_pa_EF$EF_CI),
                    cell_format(KCNH2_pa_EF$EF, KCNH2_pa_EF$EF_CI),
                    cell_format(KCNQ1_pa_EF$EF, KCNQ1_pa_EF$EF_CI),
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = sapply(c(KCNE1_rare_EF$OR_p_value, KCNE2_rare_EF$OR_p_value, KCNH2_rare_EF$OR_p_value, KCNQ1_rare_EF$OR_p_value, SCN5A_rare_EF$OR_p_value), signif, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = c(KCNE1_pa_EF$OR_p_value, KCNE2_pa_EF$OR_p_value, sapply(c(KCNH2_pa_EF$OR_p_value, KCNQ1_pa_EF$OR_p_value, SCN5A_pa_EF$OR_p_value), signif, digits = 2)), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/LQTS/Ackerman_LQTS_EFs.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)


### KCNQ1 EF plot
KCNQ1_EF_data = data.frame(
  OR = c(KCNQ1_rare_EF$OR, KCNQ1_pa_EF$OR),
  CI_lower = c(KCNQ1_rare_EF$OR_CI[1], KCNQ1_pa_EF$OR_CI[1]),
  CI_upper = c(KCNQ1_rare_EF$OR_CI[2], KCNQ1_pa_EF$OR_CI[2]),
  type = c("rare variants","rare and\nparalogue annotated")
)
KCNQ1_EF_data$type = factor(KCNQ1_EF_data$type, levels = c("rare and\nparalogue annotated","rare variants"))

gg = ggplot(KCNQ1_EF_data, aes(y=type, x=OR, 
                               #colour = type, 
                               #shape = type
                               )) + 
  geom_point(size=9) + 
  scale_color_manual(values=c("rare variants" = "#000000", "rare and\nparalogue annotated" = "#000000")) +
   geom_errorbarh(aes(xmin=CI_lower, xmax=CI_upper),colour="black",linetype=1,size=.2,height=.5) +
  # theme_bw() + 
  theme(#panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_blank(), 
        legend.position="none",
        axis.title.y = element_blank(),
        axis.text = element_text(size = 13),
        axis.title.x = element_text(size = 14)
        ) + 
  xlim(0,200) +
  xlab("Odds Ratios") +
  coord_fixed(ratio = 25)
ggsave("images/KCNQ1_ORs.png", width = 10, height = 6)

  
```
BrS 
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/BrS/Mayo_BrS_variants_for_vcf_conversion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"
write.table(rare_disease_cohort_variants, file = "data/BrS/rare_disease_cohort_variants.vcf", na="", row.names = FALSE, col.names = TRUE, sep = "\t", quote=FALSE) #RUN PA ON THIS FILE


###disease cohort
disease_data_ref = read.csv("data/BrS/Mayo_BrS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])
#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2111))
#individual cases for rare variants
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/BrS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/BrS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))

###control cohort
control_data_ref = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)

gnomad_total_variants$X.CHROM = as.numeric(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.numeric(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_paralogs2.noQC", "data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))


###Calculating EFs
SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_rare_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/BrS/Ackerman_BrS_EFs.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)


# disease_cohort_variants$check_in_rare_gnomad = ifelse(
#   is.na(
#     match(
#       paste0(disease_cohort_variants$X.CHROM, disease_cohort_variants$POS, disease_cohort_variants$REF, disease_cohort_variants$ALT),
#       paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
#       )
#     ),"No", "Yes"
#   )
# 
# disease_cohort_variants$check_in_total_gnomad

# disease_cohort_variants_tmp = disease_cohort_variants[,c(1,2,4,5)]
# gnomad_rare_variants_tmp = gnomad_rare_variants[,c(1,2,4,5)]

# vcf_conv = read.csv("data/LQTS/Mayo_LQTS_variants_for_vcf_convertion.csv", sep = ",", stringsAsFactors = 0)
```

LQTS 
not rare+pa vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}
#Finding only rare variants
disease_cohort_variants = read.csv("data/LQTS/Mayo_LQTS_variants_for_vcf_convertion.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_total_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE
rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"
###disease cohort
disease_data_ref = read.csv("data/LQTS/Mayo_LQTS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])
#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"), cases = c(2500,2500,2500,2500,2500))
#individual cases for rare variants
KCNE1_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNE2_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNH2_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNH2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
KCNQ1_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNQ1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/LQTS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/LQTS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x,]
KCNE1_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE1" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNE2_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE2" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNH2_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNH2" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
KCNQ1_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNQ1" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
#individual cases for rare variants and not annotated by PA
KCNE1_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))
KCNE2_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNE2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))
KCNH2_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNH2" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))
KCNQ1_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "KCNQ1" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))
SCN5A_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))


###control cohort
control_data_ref = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_with_customIDs.csv", sep = ",", stringsAsFactors = 0)
control_data_ref  = control_data_ref[control_data_ref$Gene != "Gene",] #remove the extra header lines from when concatenated the separate gene csv files together

gnomad_total_variants$X.CHROM = as.character(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.character(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID

KCNE1_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNE1",]
KCNE2_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNE2",]
KCNH2_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNH2",]
KCNQ1_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "KCNQ1",]
SCN5A_control_total_variant_cases = control_data_ref[control_data_ref$Gene == "SCN5A",]
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"), 
  cases = c(
    mean(as.numeric(KCNE1_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNE2_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNH2_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(KCNQ1_control_total_variant_cases$Allele.Number))/2,
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants
KCNE1_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE1" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNE2_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE2" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNH2_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNH2" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
KCNQ1_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNQ1" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "SCN5A" & control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_paralogs2.noQC", "data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x,] 
KCNE1_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE1" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNE2_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE2" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNH2_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNH2" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
KCNQ1_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNQ1" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "SCN5A" & control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
#individual cases for rare variants and not annotated by PA
KCNE1_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE1" & control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))
KCNE2_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNE2" & control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))
KCNH2_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNH2" & control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))
KCNQ1_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "KCNQ1" & control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))
SCN5A_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$Gene == "SCN5A" & control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))

###Calculating EFs
KCNE1_not_pa_EF = calc_EF(
  KCNE1_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_disease_not_pa_variant_cases,
  KCNE1_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_control_not_pa_variant_cases
  )
KCNE1_pa_EF = calc_EF(
  KCNE1_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_disease_pa_variant_cases,
  KCNE1_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE1", "cases"]-KCNE1_control_pa_variant_cases
  )

KCNE2_not_pa_EF = calc_EF(
  KCNE2_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_disease_not_pa_variant_cases,
  KCNE2_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_control_not_pa_variant_cases
  )
KCNE2_pa_EF = calc_EF(
  KCNE2_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_disease_pa_variant_cases,
  KCNE2_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNE2", "cases"]-KCNE2_control_pa_variant_cases
  )

KCNH2_not_pa_EF = calc_EF(
  KCNH2_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_disease_not_pa_variant_cases,
  KCNH2_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_control_not_pa_variant_cases
  )
KCNH2_pa_EF = calc_EF(
  KCNH2_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_disease_pa_variant_cases,
  KCNH2_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNH2", "cases"]-KCNH2_control_pa_variant_cases
  )

KCNQ1_not_pa_EF = calc_EF(
  KCNQ1_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_disease_not_pa_variant_cases,
  KCNQ1_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_control_not_pa_variant_cases
  )
KCNQ1_pa_EF = calc_EF(
  KCNQ1_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_disease_pa_variant_cases,
  KCNQ1_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "KCNQ1", "cases"]-KCNQ1_control_pa_variant_cases
  )

SCN5A_not_pa_EF = calc_EF(
  SCN5A_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_not_pa_variant_cases,
  SCN5A_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_not_pa_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("KCNE1","KCNE2","KCNH2","KCNQ1","SCN5A"),
  not_pa_variant_OR = c(cell_format(KCNE1_not_pa_EF$OR, KCNE1_not_pa_EF$OR_CI),
                      cell_format(KCNE2_not_pa_EF$OR, KCNE2_not_pa_EF$OR_CI),
                      cell_format(KCNH2_not_pa_EF$OR, KCNH2_not_pa_EF$OR_CI),
                      cell_format(KCNQ1_not_pa_EF$OR, KCNQ1_not_pa_EF$OR_CI),
                      cell_format(SCN5A_not_pa_EF$OR, SCN5A_not_pa_EF$OR_CI)),
  not_pa_variant_EF = c(cell_format(KCNE1_not_pa_EF$EF, KCNE1_not_pa_EF$EF_CI),
                      cell_format(KCNE2_not_pa_EF$EF, KCNE2_not_pa_EF$EF_CI),
                      cell_format(KCNH2_not_pa_EF$EF, KCNH2_not_pa_EF$EF_CI),
                      cell_format(KCNQ1_not_pa_EF$EF, KCNQ1_not_pa_EF$EF_CI),
                      cell_format(SCN5A_not_pa_EF$EF, SCN5A_not_pa_EF$EF_CI)),
  pa_variant_OR = c(cell_format(KCNE1_pa_EF$OR, KCNE1_pa_EF$OR_CI),
                    cell_format(KCNE2_pa_EF$OR, KCNE2_pa_EF$OR_CI),
                    cell_format(KCNH2_pa_EF$OR, KCNH2_pa_EF$OR_CI),
                    cell_format(KCNQ1_pa_EF$OR, KCNQ1_pa_EF$OR_CI),
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(cell_format(KCNE1_pa_EF$EF, KCNE1_pa_EF$EF_CI),
                    cell_format(KCNE2_pa_EF$EF, KCNE2_pa_EF$EF_CI),
                    cell_format(KCNH2_pa_EF$EF, KCNH2_pa_EF$EF_CI),
                    cell_format(KCNQ1_pa_EF$EF, KCNQ1_pa_EF$EF_CI),
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = sapply(c(KCNE1_not_pa_EF$OR_p_value, KCNE2_not_pa_EF$OR_p_value, KCNH2_not_pa_EF$OR_p_value, KCNQ1_not_pa_EF$OR_p_value, SCN5A_not_pa_EF$OR_p_value), signif, digits = 2), .after = "not_pa_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = c(KCNE1_pa_EF$OR_p_value, KCNE2_pa_EF$OR_p_value, sapply(c(KCNH2_pa_EF$OR_p_value, KCNQ1_pa_EF$OR_p_value, SCN5A_pa_EF$OR_p_value), signif, digits = 2)), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/LQTS/Ackerman_LQTS_EFs_pa_vs_not_pa.tsv", na="", row.names = FALSE, col.names = FALSE, sep = "\t", quote=FALSE)
```
BrS 
not rare+pa vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}
#Finding only rare variants
disease_cohort_variants = read.csv("data/BrS/Mayo_BrS_variants_for_vcf_conversion.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_total_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE
rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"

###disease cohort
disease_data_ref = read.csv("data/BrS/Mayo_BrS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])
#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2111))
#individual cases for rare variants
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID, "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/BrS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/BrS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x, "cases"]))
#individual cases for rare variants and not annotated by PA
SCN5A_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID & !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))


###control cohort
control_data_ref = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)

gnomad_total_variants$X.CHROM = as.numeric(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.numeric(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID, "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_paralogs2.noQC", "data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x, "Allele.Count"]))
#individual cases for rare variants and not annotated by PA
SCN5A_control_not_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID & !(control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x), "Allele.Count"]))

###Calculating EFs
SCN5A_not_pa_EF = calc_EF(
  SCN5A_disease_not_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_not_pa_variant_cases,
  SCN5A_control_not_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_not_pa_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_not_pa_EF$OR, SCN5A_not_pa_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_not_pa_EF$EF, SCN5A_not_pa_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_not_pa_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/BrS/Ackerman_BrS_EFs_pa_vs_not_pa.tsv", na="", row.names = FALSE, col.names = FALSE, sep = "\t", quote=FALSE)

```

looking at only variants in Voltage-gated Na+ ion channel, cytoplasmic domain in scn5a for LQTS
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/LQTS/Mayo_LQTS_variants_for_vcf_convertion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"
# write.table(rare_disease_cohort_variants, file = "data/BrS/rare_disease_cohort_variants.vcf", na="", row.names = FALSE, col.names = TRUE, sep = "\t", quote=FALSE) #RUN PA ON THIS FILE


###disease cohort
disease_data_ref = read.csv("data/LQTS/Mayo_LQTS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])

#isolate protein position
disease_data_ref$Protein_position = substr(disease_data_ref$mut_protein, 4, str_count(disease_data_ref$mut_protein)-1)

#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2500))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID &
                                                                     (disease_data_ref$Protein_position < 132 |
                                                                        disease_data_ref$Protein_position > 410) &
                                                                     (disease_data_ref$Protein_position < 718 |
                                                                        disease_data_ref$Protein_position > 938) &
                                                                     (disease_data_ref$Protein_position < 1207 |
                                                                        disease_data_ref$Protein_position > 1466) &
                                                                     (disease_data_ref$Protein_position < 1530 |
                                                                        disease_data_ref$Protein_position > 1771)
                                                                     , "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/LQTS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/LQTS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = disease.paralogous_var_align$Total_paralog_annotations[!is.na(disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 132 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 410) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 718 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 938) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1207 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1466) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1530 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771),]

disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x, "cases"]))

###control cohort
control_data_ref = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)
control_data_ref  = control_data_ref[control_data_ref$Gene != "Gene",] #remove the extra header lines from when concatenated the separate gene csv files together

gnomad_total_variants$X.CHROM = as.character(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.character(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

#isolate protein position
control_data_ref$Protein_position = substr(control_data_ref$Consequence, 6, str_count(control_data_ref$Consequence)-3)
control_data_ref[control_data_ref$Protein_position == "2017ArgextTe", "Protein_position"] = 2017

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID &
                                                                     (control_data_ref$Protein_position < 132 |
                                                                        control_data_ref$Protein_position > 410) &
                                                                     (control_data_ref$Protein_position < 718 |
                                                                        control_data_ref$Protein_position > 938) &
                                                                     (control_data_ref$Protein_position < 1207 |
                                                                        control_data_ref$Protein_position > 1466) &
                                                                     (control_data_ref$Protein_position < 1530 |
                                                                        control_data_ref$Protein_position > 1771)
                                                                     , "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_paralogs2.noQC", "data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 132 |
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 410) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 718 |
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 938) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1207|
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1466) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1530|
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771),]

control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x, "Allele.Count"]))

###raw numbers
LQTS_nTM_SCN5A_rare_rn = c(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
LQTS_nTM_SCN5A_pa_rn = c(SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )
###Calculating EFs
SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("LQTS_nTM_SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_rare_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/LQTS/Ackerman_LQTS_onlySCN5A_interlinker_domains_EFs.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
LQTS_nTM_SCN5A_EF = gene_EFs
```
looking at only variants in Voltage-gated Na+ ion channel, cytoplasmic domain in scn5a for BrS
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/BrS/Mayo_BrS_variants_for_vcf_conversion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"
# write.table(rare_disease_cohort_variants, file = "data/BrS/rare_disease_cohort_variants.vcf", na="", row.names = FALSE, col.names = TRUE, sep = "\t", quote=FALSE) #RUN PA ON THIS FILE


###disease cohort
disease_data_ref = read.csv("data/BrS/Mayo_BrS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])

#isolate protein position
disease_data_ref$Protein_position = substr(disease_data_ref$mut_protein, 4, str_count(disease_data_ref$mut_protein)-1)

#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2111))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID &
                                                                     (disease_data_ref$Protein_position < 132 |
                                                                        disease_data_ref$Protein_position > 410) &
                                                                     (disease_data_ref$Protein_position < 718 |
                                                                        disease_data_ref$Protein_position > 938) &
                                                                     (disease_data_ref$Protein_position < 1207 |
                                                                        disease_data_ref$Protein_position > 1466) &
                                                                     (disease_data_ref$Protein_position < 1530 |
                                                                        disease_data_ref$Protein_position > 1771)
                                                                     , "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/BrS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/BrS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = disease.paralogous_var_align$Total_paralog_annotations[!is.na(disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 132 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 410) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 718 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 938) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1207 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1466) &
                                                                              (disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1530 |
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771),]

disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x, "cases"]))

###control cohort
control_data_ref = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)

gnomad_total_variants$X.CHROM = as.numeric(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.numeric(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

#isolate protein position
control_data_ref$Protein_position = substr(control_data_ref$Consequence, 6, str_count(control_data_ref$Consequence)-3)
control_data_ref[control_data_ref$Protein_position == "2017ArgextTe", "Protein_position"] = 2017

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID &
                                                                     (control_data_ref$Protein_position < 132 |
                                                                        control_data_ref$Protein_position > 410) &
                                                                     (control_data_ref$Protein_position < 718 |
                                                                        control_data_ref$Protein_position > 938) &
                                                                     (control_data_ref$Protein_position < 1207 |
                                                                        control_data_ref$Protein_position > 1466) &
                                                                     (control_data_ref$Protein_position < 1530 |
                                                                        control_data_ref$Protein_position > 1771)
                                                                     , "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_paralogs2.noQC", "data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 132 |
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 410) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 718 |
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 938) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1207|
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1466) &
                                                                              (control.paralogous_var_align$Total_paralog_annotations$Protein_position.x < 1530|
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771),]

control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x, "Allele.Count"]))

###raw numbers
BrS_nTM_SCN5A_rare_rn = c(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
BrS_nTM_SCN5A_pa_rn = c(SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )
###Calculating EFs
SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("BrS_nTM_SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_rare_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/BrS/Ackerman_BrS_onlySCN5A_interlinker_domains_EFs.csv", na="", row.names = FALSE, col.names = FALSE, sep = ",", quote=FALSE)
BrS_nTM_SCN5A_EF = gene_EFs
```

looking at only variants in c terminal in scn5a for LQTS
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/LQTS/Mayo_LQTS_variants_for_vcf_convertion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"

###disease cohort
disease_data_ref = read.csv("data/LQTS/Mayo_LQTS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])

#isolate protein position
disease_data_ref$Protein_position = substr(disease_data_ref$mut_protein, 4, str_count(disease_data_ref$mut_protein)-1)

#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2500))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID &
                                                                        disease_data_ref$Protein_position > 1771, "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/LQTS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/LQTS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = disease.paralogous_var_align$Total_paralog_annotations[!is.na(disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771,]

disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x, "cases"]))

# SCN5A_disease_not_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID &
                                                                       #  disease_data_ref$Protein_position > 1771 &
                                                                       # !(disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x), "cases"]))


###control cohort
control_data_ref = read.csv("data/LQTS/gnomAD_v2.1.1_LQTS_variants_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)
control_data_ref  = control_data_ref[control_data_ref$Gene != "Gene",] #remove the extra header lines from when concatenated the separate gene csv files together

gnomad_total_variants$X.CHROM = as.character(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.character(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

#isolate protein position
control_data_ref$Protein_position = substr(control_data_ref$Consequence, 6, str_count(control_data_ref$Consequence)-3)
control_data_ref[control_data_ref$Protein_position == "2017ArgextTe", "Protein_position"] = 2017

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID &
                                                                        control_data_ref$Protein_position > 1771
                                                                     , "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_paralogs2.noQC", "data/LQTS/gnomAD_v2.1.1_LQTS_variants_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771,]

control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x, "Allele.Count"]))

###raw numbers
# LQTS_nTM_SCN5A_rare_rn = c(
#   SCN5A_disease_rare_variant_cases,
#   disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
#   SCN5A_control_rare_variant_cases,
#   control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
#   )
# LQTS_nTM_SCN5A_pa_rn = c(SCN5A_disease_pa_variant_cases,
#   disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
#   SCN5A_control_pa_variant_cases,
#   control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
#   )
###Calculating EFs
SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("LQTS_Cter_SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_rare_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/LQTS/Ackerman_LQTS_onlySCN5A_C_terminal_EFs.tsv", na="", row.names = FALSE, col.names = FALSE, sep = "\t", quote=FALSE)
```
looking at only variants in c terminal in scn5a for BrS
rare vs rare+pa
```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}

#Finding only rare variants
disease_cohort_variants = read.csv("data/BrS/Mayo_BrS_variants_for_vcf_conversion.vcf", sep = "\t", stringsAsFactors = 0)

gnomad_total_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A.vcf", sep = "\t", stringsAsFactors = 0)
gnomad_rare_variants = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.vcf", sep = "\t", stringsAsFactors = 0) #RUN PA ON THIS FILE

rare_disease_cohort_variants = data.frame()
# disease_cohort_variants = disease_cohort_variants[1:10,]
for (row in 1:nrow(disease_cohort_variants)){ #FINDING ONLY RARE VARIANTS BASED ON GNOMAD FREQUENCY
  if (
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_rare_variants$X.CHROM, gnomad_rare_variants$POS, gnomad_rare_variants$REF, gnomad_rare_variants$ALT)
    ){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  } else if (!(
    paste0(disease_cohort_variants[row,"X.CHROM"], disease_cohort_variants[row,"POS"], disease_cohort_variants[row,"REF"], disease_cohort_variants[row,"ALT"]) %in% paste0(gnomad_total_variants$X.CHROM, gnomad_total_variants$POS, gnomad_total_variants$REF, gnomad_total_variants$ALT)
  )){
    rare_disease_cohort_variants = rbind(rare_disease_cohort_variants, disease_cohort_variants[row,])
  }
}
colnames(rare_disease_cohort_variants)[1] = "#CHROM"

###disease cohort
disease_data_ref = read.csv("data/BrS/Mayo_BrS_variants.csv", sep = ",", stringsAsFactors = 0)
disease_data_ref$cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$cases = sapply(disease_data_ref$cases,function(x) x[1])
disease_data_ref$total_cases = sapply(disease_data_ref$rep_patients,strsplit,"/")
disease_data_ref$total_cases = sapply(disease_data_ref$total_cases,function(x) x[2])

#isolate protein position
disease_data_ref$Protein_position = substr(disease_data_ref$mut_protein, 4, str_count(disease_data_ref$mut_protein)-1)

#total cases sequenced for each gene
disease_total_variant_cases = data.frame(genes = c("SCN5A"), cases = c(2111))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_disease_rare_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% rare_disease_cohort_variants$ID &
                                                                        disease_data_ref$Protein_position > 1771
                                                                     , "cases"]))
#individual cases for variants predicted pathogenic by PA
disease.paralogous_var_align = Paralogous_var_align("data/BrS/rare_disease_cohort_variants.out_paraloc_paralogs2.noQC", "data/BrS/rare_disease_cohort_variants.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = disease.paralogous_var_align$Total_paralog_annotations[!is.na(disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                                 disease.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771,]

disease_cohort_data = disease_data_ref[disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x,]
SCN5A_disease_pa_variant_cases = sum(as.numeric(disease_data_ref[disease_data_ref$gene_symbol == "SCN5A" & disease_data_ref$mut_id %in% disease.paralogous_var_align$Unique_variant_annotations$ID.x & disease_data_ref$mut_id %in% Protein_pos_select$ID.x, "cases"]))

###control cohort
control_data_ref = read.csv("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_with_customIDs.csv", sep = ",", stringsAsFactors = 0, header = TRUE)

gnomad_total_variants$X.CHROM = as.numeric(gnomad_total_variants$X.CHROM)
gnomad_total_variants$POS = as.numeric(gnomad_total_variants$POS)
control_data_ref = full_join(control_data_ref,gnomad_total_variants,by = c("Chromosome" = "X.CHROM", "Position" = "POS", "Reference" = "REF", "Alternate" = "ALT")) #IN ORDER TO FIX THE GODDAM CUSTOM IDS, JOIN WITH ORIGINAL CSV FILE AND USE ID INSTEAD OF rsID ###I don't think I actually need this for BrS as there's only one gene and no need to identify which variants belong to which gene

#isolate protein position
control_data_ref$Protein_position = substr(control_data_ref$Consequence, 6, str_count(control_data_ref$Consequence)-3)
control_data_ref[control_data_ref$Protein_position == "2017ArgextTe", "Protein_position"] = 2017

SCN5A_control_total_variant_cases = control_data_ref
#total cases sequence for each gene
control_total_variant_cases = data.frame(
  genes = c("SCN5A"), 
  cases = c(
    mean(as.numeric(SCN5A_control_total_variant_cases$Allele.Number))/2))
#individual cases for rare variants #finding only protein positions that are not transmembrane domains identified by pfam
SCN5A_control_rare_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% gnomad_rare_variants$ID &
                                                                        control_data_ref$Protein_position > 1771
                                                                     , "Allele.Count"]))
#individual cases for variants predicted pathogenic by PA
control.paralogous_var_align = Paralogous_var_align("data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_paralogs2.noQC", "data/BrS/gnomAD_v2.1.1_ENSG00000183873_2019_03_21_17_12_57_SCN5A_rare.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized") 
#finding only protein positions that are not transmembrane domains identified by pfam
Protein_pos_select = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$Protein_position.x) &
                                                                                 control.paralogous_var_align$Total_paralog_annotations$Protein_position.x > 1771,]

control_cohort_data = control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x,] 
SCN5A_control_pa_variant_cases = sum(as.numeric(control_data_ref[control_data_ref$ID %in% control.paralogous_var_align$Unique_variant_annotations$ID.x & control_data_ref$ID %in% Protein_pos_select$ID.x, "Allele.Count"]))

###raw numbers
BrS_nTM_SCN5A_rare_rn = c(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
BrS_nTM_SCN5A_pa_rn = c(SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )
###Calculating EFs
SCN5A_rare_EF = calc_EF(
  SCN5A_disease_rare_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_rare_variant_cases,
  SCN5A_control_rare_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_rare_variant_cases
  )
SCN5A_pa_EF = calc_EF(
  SCN5A_disease_pa_variant_cases,
  disease_total_variant_cases[disease_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_disease_pa_variant_cases,
  SCN5A_control_pa_variant_cases,
  control_total_variant_cases[control_total_variant_cases$genes == "SCN5A", "cases"]-SCN5A_control_pa_variant_cases
  )

gene_EFs = data.frame(
  gene = c("BrS_nTM_SCN5A"),
  rare_variant_OR = c(
                      cell_format(SCN5A_rare_EF$OR, SCN5A_rare_EF$OR_CI)),
  rare_variant_EF = c(
                      cell_format(SCN5A_rare_EF$EF, SCN5A_rare_EF$EF_CI)),
  pa_variant_OR = c(
                    cell_format(SCN5A_pa_EF$OR, SCN5A_pa_EF$OR_CI)),
  pa_variant_EF = c(
                    cell_format(SCN5A_pa_EF$EF, SCN5A_pa_EF$EF_CI))
  )
gene_EFs = add_column(gene_EFs, "rare_p-values" = signif(SCN5A_rare_EF$OR_p_value, digits = 2), .after = "rare_variant_EF")
gene_EFs = add_column(gene_EFs, "pa_p-values" = signif(SCN5A_pa_EF$OR_p_value, digits = 2), .after = "pa_variant_EF")
write.table(gene_EFs, file = "data/BrS/Ackerman_BrS_onlySCN5A_C_terminal_EFs.tsv", na="", row.names = FALSE, col.names = FALSE, sep = "\t", quote=FALSE)
```

## EFs for NDD+E

```{r eval=FALSE, echo=FALSE, cache=TRUE, message=FALSE, warning=FALSE}
discovEHR = read.csv(file=gzfile("data/discovEHR_GRCh37.vcf.gz"), header=T, sep="\t")
NDD = read.csv("data/NDD+E/media-2.txt", sep="\t", header = T)
NDD = NDD[grepl("missense_variant", NDD$Consequence),]
NDD$Chrom = sapply(strsplit(as.character(NDD$Location),":"),"[",1)
```


## Comparison to other tools
Looking at clinvar variants:
```{r cf_sift_revel, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
# #First initial run to see what variants are not in overlapping genes
# # p.normal_PA2 = Paralogous_var_align_no_overlap("data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# p.normal_initial = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# # b.normal_PA2 = Paralogous_var_align_no_overlap("data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_initial = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_initial_conf = conf_matrix(p.normal_initial$true_num_of_paralog_anno, p.normal_initial$input_variants, b.normal_initial$true_num_of_paralog_anno, b.normal_initial$input_variants)
# 
# 
# ##Preparing and annotating with other classifiers
# #revel - as revel has the least number of variants annotated, all other methods were made sure to use the same variants that revel managed to annotate
# Revel_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized_w_REVEL", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# Revel_tableized_data = Revel_tableized_file
# Revel_tableized_data$Variant_pos = paste(Revel_tableized_data$CHROM,Revel_tableized_data$POS, sep = " ")
# Revel_tableized_data = Revel_tableized_data[!is.na(Revel_tableized_data$Codons)&!is.na(Revel_tableized_data$Paralogue_Vars),]
# Revel_tableized_data = filter(Revel_tableized_data, ID %in% p.normal_initial$n_occur$Var1)
# p.num_of_revel = length(which(Revel_tableized_data$REVEL_score > 0.5))
# 
# b.Revel_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized_w_REVEL", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# b.Revel_tableized_data = b.Revel_tableized_file
# b.Revel_tableized_data$Variant_pos = paste(b.Revel_tableized_data$CHROM,b.Revel_tableized_data$POS, sep = " ")
# b.Revel_tableized_data = b.Revel_tableized_data[!is.na(b.Revel_tableized_data$Codons)&!is.na(b.Revel_tableized_data$Paralogue_Vars),]
# b.Revel_tableized_data = filter(b.Revel_tableized_data, ID %in% b.normal_initial$n_occur$Var1)
# b.num_of_revel = length(which(b.Revel_tableized_data$REVEL_score > 0.5))
# 
# # Revel_TP = p.num_of_revel
# # Revel_FP = b.num_of_revel
# # Revel_TN = nrow(b.Revel_tableized_data)-b.num_of_revel
# # Revel_FN = nrow(Revel_tableized_data)-p.num_of_revel
# # Revel_PPV = Revel_TP/(Revel_TP+Revel_FP)
# # Revel_Sensitivty = Revel_TP/(Revel_TP+Revel_FN)
# # Revel_Specificity = Revel_TN/(Revel_TN+Revel_FP)
# # Revel_FPR = Revel_FP/(Revel_FP+Revel_TN)
# # Revel_ACC = (Revel_TP + Revel_TN)/(Revel_TP + Revel_TN + Revel_FP + Revel_FN)
# # Revel_con_table = matrix(
# #   c(nrow(Revel_tableized_data),
# #     p.num_of_revel,
# #     nrow(b.Revel_tableized_data),
# #     b.num_of_revel
# #     ), ncol = 2
# # )
# # colnames(Revel_con_table) = c("Pathogenic", "Benign")
# # rownames(Revel_con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
# 
# revel_conf = raw_conf_matrix(p.num_of_revel,Revel_tableized_data,b.num_of_revel,b.Revel_tableized_data)
# 
# #sift
# Sift_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# Sift_tableized_data = Sift_tableized_file
# Sift_tableized_data$Variant_pos = paste(Sift_tableized_data$CHROM,Sift_tableized_data$POS, sep = " ")
# 
# Sift_tableized_data = filter(Sift_tableized_data, ID %in% Revel_tableized_data$ID)
# p.num_of_sift = length(which(Sift_tableized_data$SIFT_pred == "D")) 
# 
# b.Sift_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# b.Sift_tableized_data = b.Sift_tableized_file
# b.Sift_tableized_data$Variant_pos = paste(b.Sift_tableized_data$CHROM,b.Sift_tableized_data$POS, sep = " ")
# 
# b.Sift_tableized_data = filter(b.Sift_tableized_data, ID %in% b.Revel_tableized_data$ID)
# b.num_of_sift = length(which(b.Sift_tableized_data$SIFT_pred == "D")) 
# 
# # Sift_TP = p.num_of_sift
# # Sift_FP = b.num_of_sift
# # Sift_TN = nrow(b.Sift_tableized_data)-b.num_of_sift
# # Sift_FN = nrow(Sift_tableized_data)-p.num_of_sift
# # Sift_PPV = Sift_TP/(Sift_TP+Sift_FP)
# # Sift_Sensitivty = Sift_TP/(Sift_TP+Sift_FN)
# # Sift_Specificity = Sift_TN/(Sift_TN+Sift_FP)
# # Sift_FPR = Sift_FP/(Sift_FP+Sift_TN)
# # Sift_ACC = (Sift_TP + Sift_TN)/(Sift_TP + Sift_TN + Sift_FP + Sift_FN)
# # Sift_con_table = matrix(
# #   c(nrow(Sift_tableized_data),
# #     p.num_of_sift,
# #     nrow(b.Sift_tableized_data),
# #     b.num_of_sift
# #     ), ncol = 2
# # )
# # colnames(Sift_con_table) = c("Pathogenic", "Benign")
# # rownames(Sift_con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
# 
# sift_conf = raw_conf_matrix(p.num_of_sift,Sift_tableized_data,b.num_of_sift,b.Sift_tableized_data)
# 
# #M-CAP
# Mcap_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc.hg19_multianno.vcf_tableized_org_M_CAP", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# Mcap_tableized_data = filter(Mcap_tableized_file, ID %in% Revel_tableized_data$ID)
# p.num_of_mcap = length(which(Mcap_tableized_data$M.CAP_pred == "D"))
# 
# b.Mcap_tableized_file = read.csv(file="data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc.hg19_multianno.vcf_tableized_org_M_CAP", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# b.Mcap_tableized_data = filter(b.Mcap_tableized_file, ID %in% b.Revel_tableized_data$ID)
# b.num_of_mcap = length(which(b.Mcap_tableized_data$M.CAP_pred == "D")) 
#  
# # Mcap_TP = p.num_of_mcap
# # Mcap_FP = b.num_of_mcap
# # Mcap_TN = nrow(b.Mcap_tableized_data)-b.num_of_mcap
# # Mcap_FN = nrow(Mcap_tableized_data)-p.num_of_mcap
# # Mcap_PPV = Mcap_TP/(Mcap_TP+Mcap_FP)
# # Mcap_Sensitivty = Mcap_TP/(Mcap_TP+Mcap_FN)
# # Mcap_Specificity = Mcap_TN/(Mcap_TN+Mcap_FP)
# # Mcap_FPR = Mcap_FP/(Mcap_FP+Mcap_TN)
# # Mcap_ACC = (Mcap_TP + Mcap_TN)/(Mcap_TP + Mcap_TN + Mcap_FP + Mcap_FN)
# # Mcap_con_table = matrix(
# #   c(nrow(Mcap_tableized_data),
# #     p.num_of_mcap,
# #     nrow(b.Mcap_tableized_data),
# #     b.num_of_mcap
# #     ), ncol = 2
# # )
# # colnames(Mcap_con_table) = c("Pathogenic", "Benign")
# # rownames(Mcap_con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
# 
# mcap_conf = raw_conf_matrix(p.num_of_mcap,Mcap_tableized_data,b.num_of_mcap,b.Mcap_tableized_data)


##New build 38 data
p.normal_initial = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
other_tools_file = other_tools_file[complete.cases(other_tools_file),]
other_tools_file = filter(other_tools_file, ID %in% p.normal_initial$n_occur$Var1)
p.num_of_revel = length(which(other_tools_file$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools_file$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools_file$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools_file$CADD_phred > 15))

b.normal_initial = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")

b.other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.other_tools_file = b.other_tools_file[complete.cases(b.other_tools_file),]
b.other_tools_file = filter(b.other_tools_file, ID %in% b.normal_initial$n_occur$Var1)
b.num_of_revel = length(which(b.other_tools_file$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools_file$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools_file$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools_file$CADD_phred > 15))

sift_conf = raw_conf_matrix(p.num_of_sift,other_tools_file,b.num_of_sift,b.other_tools_file)
mcap_conf = raw_conf_matrix(p.num_of_mcap,other_tools_file,b.num_of_mcap,b.other_tools_file)
revel_conf = raw_conf_matrix(p.num_of_revel,other_tools_file,b.num_of_revel,b.other_tools_file)
cadd_conf = raw_conf_matrix(p.num_of_cadd,other_tools_file,b.num_of_cadd,b.other_tools_file)

sift1_conf = conf_matrix(p.num_of_sift,other_tools_file,b.num_of_sift,b.other_tools_file)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools_file,b.num_of_mcap,b.other_tools_file)
revel1_conf = conf_matrix(p.num_of_revel,other_tools_file,b.num_of_revel,b.other_tools_file)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools_file,b.num_of_cadd,b.other_tools_file)


#PA
# p.normal_PA = Paralogous_var_align(Subset = c("ID",Revel_tableized_data$ID),"data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA = Paralogous_var_align(Subset = c("ID",b.Revel_tableized_data$ID),"data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)

p.normal_PA = Paralogous_var_align(Subset = c("ID",other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)




# p.normal_PA2 = Paralogous_var_align(Subset = c("ID",Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA2 = Paralogous_var_align(Subset = c("ID",b.Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)

p.normal_PA2 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)


# p.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# p.normal_PA3 = Paralogous_var_align(Subset = c("ID",Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA3 = Paralogous_var_align(Subset = c("ID",b.Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA3 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)


p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA4$Total_paralog_annotations = distinct(p.normal_PA4$Total_paralog_annotations)
p.normal_PA4$Unique_variant_annotations = p.normal_PA4$Total_paralog_annotations[!is.na(p.normal_PA4$Total_paralog_annotations$ID.y),]
p.normal_PA4$Unique_variant_annotations = distinct(p.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA4$true_num_of_paralog_anno = length(p.normal_PA4$Unique_variant_annotations$Variant_pos)
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4$Total_paralog_annotations = distinct(b.normal_PA4$Total_paralog_annotations)
b.normal_PA4$Unique_variant_annotations = b.normal_PA4$Total_paralog_annotations[!is.na(b.normal_PA4$Total_paralog_annotations$ID.y),]
b.normal_PA4$Unique_variant_annotations = distinct(b.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA4$true_num_of_paralog_anno = length(b.normal_PA4$Unique_variant_annotations$Variant_pos)
normal_PA4_conf = conf_matrix(p.normal_PA4$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA4$true_num_of_paralog_anno, b.normal_PA3$input_variants)

#QC3+QC1, no QC2  
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA5$Total_paralog_annotations = distinct(p.normal_PA5$Total_paralog_annotations)
p.normal_PA5$Unique_variant_annotations = p.normal_PA5$Total_paralog_annotations[!is.na(p.normal_PA5$Total_paralog_annotations$ID.y),]
p.normal_PA5$Unique_variant_annotations = distinct(p.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA5$true_num_of_paralog_anno = length(p.normal_PA5$Unique_variant_annotations$Variant_pos)
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5$Total_paralog_annotations = distinct(b.normal_PA5$Total_paralog_annotations)
b.normal_PA5$Unique_variant_annotations = b.normal_PA5$Total_paralog_annotations[!is.na(b.normal_PA5$Total_paralog_annotations$ID.y),]
b.normal_PA5$Unique_variant_annotations = distinct(b.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA5$true_num_of_paralog_anno = length(b.normal_PA5$Unique_variant_annotations$Variant_pos)
normal_PA5_conf = conf_matrix(p.normal_PA5$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA5$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# p.normal_PA = Subset_var_align_id(Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA = Subset_var_align_id(b.Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA_conf = conf_matrix(p.normal_PA$num_of_paralog_anno, p.normal_PA$paralog_data, b.normal_PA$num_of_paralog_anno, b.normal_PA$paralog_data)
# p.normal_PA2 = Subset_var_align_id(Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA2 = Subset_var_align_id(b.Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA2_conf = conf_matrix(p.normal_PA2$num_of_paralog_anno, p.normal_PA2$paralog_data, b.normal_PA2$num_of_paralog_anno, b.normal_PA2$paralog_data)
# p.normal_PA3 = Subset_var_align_id(Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# b.normal_PA3 = Subset_var_align_id(b.Revel_tableized_data$ID,"data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# normal_PA3_conf = conf_matrix(p.normal_PA3$num_of_paralog_anno, p.normal_PA3$paralog_data, b.normal_PA3$num_of_paralog_anno, b.normal_PA3$paralog_data)

#ParaZ
Sensitivity_list = vector()
Sens_CI_list = vector()
FPR_list = vector()
con_table_list = matrix(, ncol = 2)
PPV_list = vector()
PPV_CI_list = vector()
P_value_list = vector()
Specificity_list = vector()
##Table##
paraz_set = t(data.frame(Total = c(revel_conf$con_table[1],revel_conf$con_table[3],"NA","NA","NA","NA","NA","NA")))
colnames(paraz_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sensitivity CI","Specificity","P value")
####
for (i in seq(0,11)){
#noQC
#pathogenic to pathogenic
# p.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  # p.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
  p.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",other_tools_file$ID),
                                           # "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC",
                                           "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
  
#benign to pathogenic
# b.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
  # b.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",b.Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
  b.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",b.other_tools_file$ID),
                                           # "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC",
                                           "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
  
  
  paraz.conf = conf_matrix(p.paraz_var_align$true_num_of_paralog_anno, p.paraz_var_align$paralog_data, b.paraz_var_align$true_num_of_paralog_anno, b.paraz_var_align$paralog_data)
  Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
  Sens_CI_list = c(Sens_CI_list, 1.96*paraz.conf$Sensitivity_SE)
  FPR_list = c(FPR_list, paraz.conf$FPR)
  con_table_list = rbind(con_table_list, paraz.conf$con_table)
  PPV_list = c(PPV_list, paraz.conf$PPV)
  PPV_CI_list= c(PPV_CI_list, 1.96*paraz.conf$PPV_SE)
  P_value_list = c(P_value_list, paraz.conf$Pvalue$p.value)
  Specificity_list = c(Specificity_list, paraz.conf$Specificity)
  
  PA_threshold_row = c(format(paraz.conf$con_table[2], scientific = FALSE), format(paraz.conf$con_table[2,2], scientific = FALSE), round(paraz.conf$PPV, digits = 3), round(1.96*paraz.conf$PPV_SE, digits = 3), round(paraz.conf$Sensitivity, digits = 3), round(1.96*paraz.conf$Sensitivity_SE, digits =3), round(paraz.conf$Specificity,digits = 3), signif(paraz.conf$Pvalue$p.value, digits = 3))
  paraz_set = rbind(paraz_set, PA_threshold_row)
}

# Overall dataframe
other_tools_set = t(data.frame(
  # Total = c(sapply(c(revel_conf$con_table[1],revel_conf$con_table[3]),format,scientific = FALSE),"NA","NA","NA","NA"),
  Total = c(sapply(c(dim(other_tools_file)[1],dim(b.other_tools_file)[1]),format,scientific = FALSE),"NA","NA","NA","NA","NA","NA"),
  Sift = c(sapply(c(sift_conf$con_table[2],sift_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(sift_conf$PPV,sift1_conf$PPV_SE*1.96,sift_conf$Sensitivity,sift1_conf$Sensitivity_SE*1.96,sift_conf$Specificity,sift_conf$Pvalue$p.value),signif,digits=3)),
  MCap = c(sapply(c(mcap_conf$con_table[2],mcap_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(mcap_conf$PPV,mcap1_conf$PPV_SE*1.96,mcap_conf$Sensitivity,mcap1_conf$Sensitivity_SE*1.96,mcap_conf$Specificity,mcap_conf$Pvalue$p.value),signif,digits=3)),
  Cadd = c(sapply(c(cadd_conf$con_table[2],cadd_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(cadd_conf$PPV,cadd1_conf$PPV_SE*1.96,cadd_conf$Sensitivity,cadd1_conf$Sensitivity_SE*1.96,cadd_conf$Specificity,cadd_conf$Pvalue$p.value),signif,digits=3)),
  Revel = c(sapply(c(revel_conf$con_table[2],revel_conf$con_table[4]),format,scientific = FALSE),
            sapply(c(revel_conf$PPV,revel1_conf$PPV_SE*1.96,revel_conf$Sensitivity,revel1_conf$Sensitivity_SE*1.96,revel_conf$Specificity,revel_conf$Pvalue$p.value),signif,digits=3)),
  PA.noQC = c(sapply(c(normal_PA_conf$con_table[2],normal_PA_conf$con_table[4]),format,scientific = FALSE),
         sapply(c(normal_PA_conf$PPV,normal_PA_conf$PPV_SE*1.96,normal_PA_conf$Sensitivity,normal_PA_conf$Sensitivity_SE*1.96,normal_PA_conf$Specificity,normal_PA_conf$Pvalue$p.value),signif,digits=3)),
  PA.para_con = c(sapply(c(normal_PA2_conf$con_table[2],normal_PA2_conf$con_table[4]),format,scientific = FALSE),
                  sapply(c(normal_PA2_conf$PPV,normal_PA2_conf$PPV_SE*1.96,normal_PA2_conf$Sensitivity,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA2_conf$Specificity,normal_PA2_conf$Pvalue$p.value),signif,digits=3)),
  PA.all_con = c(sapply(c(normal_PA3_conf$con_table[2],normal_PA3_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA3_conf$PPV,normal_PA3_conf$PPV_SE*1.96,normal_PA3_conf$Sensitivity,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA3_conf$Specificity,normal_PA3_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con = c(sapply(c(normal_PA4_conf$con_table[2],normal_PA4_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA4_conf$PPV,normal_PA4_conf$PPV_SE*1.96,normal_PA4_conf$Sensitivity,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA4_conf$Specificity,normal_PA4_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con_w_para_con_no_all_con = c(sapply(c(normal_PA5_conf$con_table[2],normal_PA5_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA5_conf$PPV,normal_PA5_conf$PPV_SE*1.96,normal_PA5_conf$Sensitivity,normal_PA5_conf$Sensitivity_SE*1.96,normal_PA5_conf$Specificity,normal_PA5_conf$Pvalue$p.value),signif,digits=3))
))
colnames(other_tools_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sens CI","Specificity","P value")
paraz_set = paraz_set[-1,]
other_tools_set = rbind(other_tools_set,paraz_set)
rownames(other_tools_set) = c("Total","Sift","MCap","CADD","Revel","Paralogue Annotation","Paralogue Annotation (QC1)","Paralogue Annotation (QC2)", "Paralogue Annotation (QC3)", "Paralogue Annotation (QC3, no QC2)", "Para z_score cutoff 0","Para z_score cutoff 1","Para z_score cutoff 2","Para z_score cutoff 3","Para z_score cutoff 4","Para z_score cutoff 5","Para z_score cutoff 6","Para z_score cutoff 7","Para z_score cutoff 8","Para z_score cutoff 9","Para z_score cutoff 10","Para z_score cutoff 11")
# write.table(other_tools_set, file = "data/other_tools_set.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)
write.table(other_tools_set, file = "data/other_tools_set_w_paraz_afterQC1.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)


#for subsetting the massive dataframe above for easier graph making
other_tools_data = data.frame(
  Precision = c(as.numeric(other_tools_set[2:9,3])),
  Precision_CI = c(as.numeric(other_tools_set[2:9,4])),
  Sensitivity = c(as.numeric(other_tools_set[2:9,5])),
  Sensitivity_CI = c(as.numeric(other_tools_set[2:9,6]))
  )
other_tools_data$Type = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1" ,"QC1+QC2", "QC1+QC2+QC3")
# other_tools_data = rbind(other_tools_data, prc_data2[1:4,])
other_tools_data$Type = factor(other_tools_data$Type, levels = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1" ,"QC1+QC2", "QC1+QC2+QC3"))

gg4 = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  scale_shape_manual(values=c(0, 1, 2, 5, 15, 16, 17, 18)) +
  xlim(0,1) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(title = "Precision-Sensitivity space plot of Paralogue Annotation\ncompared with other tools", 
    x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.5)) #+ scale_color_discrete(breaks=c("Sift","MCap","Revel","noQC","QC1","QC2","QC3"))
ggsave("images/PRC_with_other_tools_plot.png")

other_tools_data = data.frame(
  Precision = c(as.numeric(other_tools_set[2:10,3])),
  Precision_CI = c(as.numeric(other_tools_set[2:10,4])),
  Sensitivity = c(as.numeric(other_tools_set[2:10,5])),
  Sensitivity_CI = c(as.numeric(other_tools_set[2:10,6]))
  )
other_tools_data$Type = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1", "QC1+QC2", "QC1+QC2+QC3", "QC1+QC3")
# other_tools_data = rbind(other_tools_data, prc_data2[1:4,])
other_tools_data$Type = factor(other_tools_data$Type, levels = c("Sift", "MCap", "CADD", "Revel","no QC", "QC1" ,"QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 20)
gg5 = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15, stroke=2) + 
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=1,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=1,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 20), 
    plot.title = element_text(size = 14)) + 
  theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(axis.text = element_text(size = 35)) + 
  theme(plot.title = element_text(hjust = 0.5)) + 
  theme(legend.position="none", axis.title.x=element_blank(), axis.title.y=element_blank())#+ scale_color_discrete(breaks=c("Sift","MCap","Revel","noQC","QC1","QC2","QC3"))
ggsave("images/PRC_with_other_tools_with_QC3_QC1_no_QC2.png", width = 10, height = 11) #THIS ONE IS BEING USED IN PAPER

other_tools_data = data.frame(
  Precision = c(as.numeric(other_tools_set[6:10,3])),
  Precision_CI = c(as.numeric(other_tools_set[6:10,4])),
  Sensitivity = c(as.numeric(other_tools_set[6:10,5])),
  Sensitivity_CI = c(as.numeric(other_tools_set[6:10,6]))
  )
other_tools_data$Type = c("noQC", "QC1" ,"QC2", "QC3", "QC3 (skipped QC2)")
# other_tools_data = rbind(other_tools_data, prc_data2[1:4,])
other_tools_data$Type = factor(other_tools_data$Type, levels = c("noQC", "QC1" ,"QC2", "QC3", "QC3 (skipped QC2)"))

gg6 = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  scale_shape_manual(values=c(15, 16, 17, 18, 19)) +
  xlim(0,1) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(title = "Precision-Sensitivity space plot of Paralogue Annotation\ncompared with other tools", 
    x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.5)) #+ scale_color_discrete(breaks=c("Sift","MCap","Revel","noQC","QC1","QC2","QC3"))
ggsave("images/PRC_with_QC3_QC1_no_QC2.png")

```

Looking at only SCN5A:
```{r cf_sift_revel_SCN%A, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
##New build 38 data
p.normal_initial = Paralogous_var_align(Subset = c("Gene", "SCN5A"),Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
other_tools_file = other_tools_file[complete.cases(other_tools_file),]
other_tools_file = filter(other_tools_file, ID %in% p.normal_initial$n_occur$Var1)
p.num_of_revel = length(which(other_tools_file$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools_file$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools_file$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools_file$CADD_phred > 15))

b.normal_initial = Paralogous_var_align(Subset = c("Gene", "SCN5A"),Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")

b.other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.other_tools_file = b.other_tools_file[complete.cases(b.other_tools_file),]
b.other_tools_file = filter(b.other_tools_file, ID %in% b.normal_initial$n_occur$Var1)
b.num_of_revel = length(which(b.other_tools_file$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools_file$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools_file$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools_file$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools_file,b.num_of_sift,b.other_tools_file)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools_file,b.num_of_mcap,b.other_tools_file)
revel1_conf = conf_matrix(p.num_of_revel,other_tools_file,b.num_of_revel,b.other_tools_file)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools_file,b.num_of_cadd,b.other_tools_file)


#PA


p.normal_PA = Paralogous_var_align(Subset = c("ID",other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)



p.normal_PA2 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)


p.normal_PA3 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)


p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA4$Total_paralog_annotations = distinct(p.normal_PA4$Total_paralog_annotations)
p.normal_PA4$Unique_variant_annotations = p.normal_PA4$Total_paralog_annotations[!is.na(p.normal_PA4$Total_paralog_annotations$ID.y),]
p.normal_PA4$Unique_variant_annotations = distinct(p.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA4$true_num_of_paralog_anno = length(p.normal_PA4$Unique_variant_annotations$Variant_pos)
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4$Total_paralog_annotations = distinct(b.normal_PA4$Total_paralog_annotations)
b.normal_PA4$Unique_variant_annotations = b.normal_PA4$Total_paralog_annotations[!is.na(b.normal_PA4$Total_paralog_annotations$ID.y),]
b.normal_PA4$Unique_variant_annotations = distinct(b.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA4$true_num_of_paralog_anno = length(b.normal_PA4$Unique_variant_annotations$Variant_pos)
normal_PA4_conf = conf_matrix(p.normal_PA4$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA4$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA5$Total_paralog_annotations = distinct(p.normal_PA5$Total_paralog_annotations)
p.normal_PA5$Unique_variant_annotations = p.normal_PA5$Total_paralog_annotations[!is.na(p.normal_PA5$Total_paralog_annotations$ID.y),]
p.normal_PA5$Unique_variant_annotations = distinct(p.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA5$true_num_of_paralog_anno = length(p.normal_PA5$Unique_variant_annotations$Variant_pos)
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5$Total_paralog_annotations = distinct(b.normal_PA5$Total_paralog_annotations)
b.normal_PA5$Unique_variant_annotations = b.normal_PA5$Total_paralog_annotations[!is.na(b.normal_PA5$Total_paralog_annotations$ID.y),]
b.normal_PA5$Unique_variant_annotations = distinct(b.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA5$true_num_of_paralog_anno = length(b.normal_PA5$Unique_variant_annotations$Variant_pos)
normal_PA5_conf = conf_matrix(p.normal_PA5$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA5$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# #ParaZ
# Sensitivity_list = vector()
# FPR_list = vector()
# con_table_list = matrix(, ncol = 2)
# PPV_list = vector()
# P_value_list = vector()
# Specificity_list = vector()
# ##Table##
# paraz_set = t(data.frame(Total = c(revel_conf$con_table[1],revel_conf$con_table[3],"NA","NA","NA","NA")))
# colnames(paraz_set) = c("Pathogenic variants","Benign variants","PPV","Sensitivity","Specificity","P value")
# ###
# for (i in seq(0,11)){
# #noQC
# #pathogenic to pathogenic
# # p.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#   # p.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#   p.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#   
# #benign to pathogenic
# # b.paraz_var_align = ParaZ_var_align(i,"data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "data/clinvar/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#   # b.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",b.Revel_tableized_data$ID), "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20181028_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#   b.paraz_var_align = Paralogous_var_align(paraz_cutoff = i, Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#   
#   
#   paraz.conf = conf_matrix(p.paraz_var_align$true_num_of_paralog_anno, p.paraz_var_align$paralog_data, b.paraz_var_align$true_num_of_paralog_anno, b.paraz_var_align$paralog_data)
#   Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
#   FPR_list = c(FPR_list, paraz.conf$FPR)
#   con_table_list = rbind(con_table_list, paraz.conf$con_table)
#   PPV_list = c(PPV_list, paraz.conf$PPV)
#   P_value_list = c(P_value_list, paraz.conf$Pvalue$p.value)
#   Specificity_list = c(Specificity_list, paraz.conf$Specificity)
#   
#   PA_threshold_row = c(format(paraz.conf$con_table[2], scientific = FALSE), format(paraz.conf$con_table[2,2], scientific = FALSE), round(paraz.conf$PPV, digits = 3), round(paraz.conf$Sensitivity, digits = 3), round(paraz.conf$Specificity,digits = 3), signif(paraz.conf$Pvalue$p.value, digits = 3))
#   paraz_set = rbind(paraz_set, PA_threshold_row)
# }

# Overall dataframe
other_tools_set = t(data.frame(
  # Total = c(sapply(c(revel_conf$con_table[1],revel_conf$con_table[3]),format,scientific = FALSE),"NA","NA","NA","NA"),
  Total = c(sapply(c(dim(other_tools_file)[1],dim(b.other_tools_file)[1]),format,scientific = FALSE),"NA","NA","NA","NA","NA","NA"),
  Sift = c(sapply(c(sift1_conf$con_table[2],sift1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(sift1_conf$PPV,sift1_conf$PPV_SE*1.96,sift1_conf$Sensitivity,sift1_conf$Sensitivity_SE*1.96,sift1_conf$Specificity,sift1_conf$Pvalue$p.value),signif,digits=3)),
  MCap = c(sapply(c(mcap1_conf$con_table[2],mcap1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(mcap1_conf$PPV,mcap1_conf$PPV_SE*1.96,mcap1_conf$Sensitivity,mcap1_conf$Sensitivity_SE*1.96,mcap1_conf$Specificity,mcap1_conf$Pvalue$p.value),signif,digits=3)),
  Cadd = c(sapply(c(cadd1_conf$con_table[2],cadd1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(cadd1_conf$PPV,cadd1_conf$PPV_SE*1.96,cadd1_conf$Sensitivity,cadd1_conf$Sensitivity_SE*1.96,cadd1_conf$Specificity,cadd1_conf$Pvalue$p.value),signif,digits=3)),
  Revel = c(sapply(c(revel1_conf$con_table[2],revel1_conf$con_table[4]),format,scientific = FALSE),
            sapply(c(revel1_conf$PPV,revel1_conf$PPV_SE*1.96,revel1_conf$Sensitivity,revel1_conf$Sensitivity_SE*1.96,revel1_conf$Specificity,revel1_conf$Pvalue$p.value),signif,digits=3)),
  PA.noQC = c(sapply(c(normal_PA_conf$con_table[2],normal_PA_conf$con_table[4]),format,scientific = FALSE),
         sapply(c(normal_PA_conf$PPV,normal_PA_conf$PPV_SE*1.96,normal_PA_conf$Sensitivity,normal_PA_conf$Sensitivity_SE*1.96,normal_PA_conf$Specificity,normal_PA_conf$Pvalue$p.value),signif,digits=3)),
  PA.para_con = c(sapply(c(normal_PA2_conf$con_table[2],normal_PA2_conf$con_table[4]),format,scientific = FALSE),
                  sapply(c(normal_PA2_conf$PPV,normal_PA2_conf$PPV_SE*1.96,normal_PA2_conf$Sensitivity,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA2_conf$Specificity,normal_PA2_conf$Pvalue$p.value),signif,digits=3)),
  PA.all_con = c(sapply(c(normal_PA3_conf$con_table[2],normal_PA3_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA3_conf$PPV,normal_PA3_conf$PPV_SE*1.96,normal_PA3_conf$Sensitivity,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA3_conf$Specificity,normal_PA3_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con = c(sapply(c(normal_PA4_conf$con_table[2],normal_PA4_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA4_conf$PPV,normal_PA4_conf$PPV_SE*1.96,normal_PA4_conf$Sensitivity,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA4_conf$Specificity,normal_PA4_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con_w_QC1_no_QC2 = c(sapply(c(normal_PA5_conf$con_table[2],normal_PA5_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA5_conf$PPV,normal_PA5_conf$PPV_SE*1.96,normal_PA5_conf$Sensitivity,normal_PA5_conf$Sensitivity_SE*1.96,normal_PA5_conf$Specificity,normal_PA5_conf$Pvalue$p.value),signif,digits=3))
))
colnames(other_tools_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sens CI","Specificity","P value")
##REMINDER TO FIX APPENDING OF PARAZ SET AS HAVE NOT YET ADDED CI FOR PPV AND SENS
# paraz_set = paraz_set[-1,]
# other_tools_set = rbind(other_tools_set,paraz_set)
# rownames(other_tools_set) = c("Total","Sift","MCap","CADD","Revel","Paralogue Annotation","Paralogue Annotation (QC1)","Paralogue Annotation (QC2)", "Paralogue Annotation (QC3)","Para z_score cutoff 0","Para z_score cutoff 1","Para z_score cutoff 2","Para z_score cutoff 3","Para z_score cutoff 4","Para z_score cutoff 5","Para z_score cutoff 6","Para z_score cutoff 7","Para z_score cutoff 8","Para z_score cutoff 9","Para z_score cutoff 10","Para z_score cutoff 11")
# write.table(other_tools_set, file = "data/other_tools_set.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)
# 

other_tools_data = data.frame(
  Precision = c(as.numeric(other_tools_set[2:10,3])),
  Precision_CI = c(as.numeric(other_tools_set[2:10,4])),
  Sensitivity = c(as.numeric(other_tools_set[2:10,5])),
  Sensitivity_CI = c(as.numeric(other_tools_set[2:10,6]))
  )
other_tools_data$Type = c("Sift", "MCap", "CADD", "Revel", "none", "conserved AA" ,"full conserved AA", "full conserved AA +\nconserved variants", "conserved AA +\nconserved variants")
# other_tools_data = rbind(other_tools_data, prc_data2[1:4,])
other_tools_data$Type = factor(other_tools_data$Type, levels = c("none", "conserved AA" , "full conserved AA", "conserved AA +\nconserved variants", "full conserved AA +\nconserved variants", "Sift", "MCap", "CADD", "Revel"))
# other_tools_data$Precision_CI_min = other_tools_data$Precision - other_tools_data$Precision_CI
# other_tools_data$Precision_CI_max = sapply(other_tools_data$Precision + other_tools_data$Precision_CI, min, 1)

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "none" = "#58D68D", "conserved AA"  = "#45B39D", "full conserved AA" = "#5DADE2", "conserved AA +\nconserved variants" = "#A569BD", "full conserved AA +\nconserved variants" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "none" = 15, "conserved AA" = 16, "full conserved AA" = 17, "conserved AA +\nconserved variants" = 18, "full conserved AA +\nconserved variants" = 20)
gg = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15,stroke=2) + 
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=1,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=1,height=0) +
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    plot.title = element_text(size = 14)) +labs(y = "PPV", x = "Sensitivity") +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(legend.position="none", axis.text = element_text(size = 35), axis.title.x=element_blank(), axis.title.y=element_blank()) + theme(legend.key.size = unit(2,"line"))
ggsave("images/PRC_of_SCN5A.png", width = 10, height = 11)

gg = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15) + 
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0.8,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(y = "PPV", x = "Sensitivity") +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(legend.text=element_text(size=25), legend.title=element_blank(), axis.text = element_text(size = 16), axis.title.x=element_blank(), axis.title.y=element_blank()) + theme(plot.title = element_text(hjust = 0.25)) + theme(legend.key.size = unit(5,"line"))
ggsave("images/PRC_of_SCN5A_with_legend.png", width = 10, height = 11)
```


Looking at UniFun and TP53-TA variants:
```{r cf_UniFun_TP53-TA, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
p.normal_PA = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.noQC", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.noQC", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.all_con", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.all_con", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)

other_tools = read.csv(file="data/TP53-TA/4704961/TP53-TA_deleterious.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/TP53-TA/4704961/TP53-TA_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))


# sift_conf = raw_conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
# mcap_conf = raw_conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
# revel_conf = raw_conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
# cadd_conf = raw_conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)
sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df1 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df1$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)
gg = ggplot(df1, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15,stroke=2) + 
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=1,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=1,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14),
    legend.position="none", 
    axis.text = element_text(size = 35), 
    axis.title.x=element_blank(), 
    axis.title.y=element_blank()) + 
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(plot.title = element_text(hjust = 0.25)) 
ggsave("images/PRC_of_TP53-TA_with_other_tools_plot.png", width = 10, height = 11)

#checking if any of the variants are in clinvar
TP53_variants = other_tools[,c("CHROM","POS","REF","ALT")]
TP53_variants$Chr = substr(TP53_variants$CHROM, 4, str_count(TP53_variants$CHROM))
TP53_variants$SUM = paste(TP53_variants$Chr, TP53_variants$POS, TP53_variants$REF, TP53_variants$ALT, sep = " ")
clinvar_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37.vcf_tableized_org", sep ="\t", header = T, stringsAsFactors = F)
clinvar_tableized$SUM = paste(clinvar_tableized$CHROM, clinvar_tableized$POS, clinvar_tableized$REF, clinvar_tableized$ALT, sep = " ")
clinvar_tableized = clinvar_tableized[clinvar_tableized$SUM %in% TP53_variants$SUM &
                                        !(grepl("Conflicting_interpretations_of_pathogenicity", clinvar_tableized$CLNSIG))
                                      ,]

##################
p.normal_PA = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_deleterious.out_paraloc_paralogs2.noQC", "data/UniFun/4704955/UniFun_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_benign.out_paraloc_paralogs2.noQC", "data/UniFun/4704955/UniFun_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_deleterious.out_paraloc_paralogs2.para_con", "data/UniFun/4704955/UniFun_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_benign.out_paraloc_paralogs2.para_con", "data/UniFun/4704955/UniFun_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_deleterious.out_paraloc_paralogs2.all_con", "data/UniFun/4704955/UniFun_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Overlap = 0, "data/UniFun/4704955/UniFun_benign.out_paraloc_paralogs2.all_con", "data/UniFun/4704955/UniFun_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)

other_tools = read.csv(file="data/UniFun/4704955/UniFun_deleterious.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/UniFun/4704955/UniFun_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))


# sift_conf = raw_conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
# mcap_conf = raw_conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
# revel_conf = raw_conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
# cadd_conf = raw_conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)
sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df2 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","noQC","QC1","QC2","QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96)
)

```

Looking at PPARG and SSC:
```{r cf_PPARG, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
f = read.csv("data/J_Li_data/J_Li_data.csv", sep = ",", stringsAsFactors = F, header = T)
PPARG = f[f$Data.source=="PPARG",-c(1,4)]
PPARG = add_column(PPARG, "ID" = paste0("custom_",1:dim(PPARG)[1]), .after = "Start")

PPARG$Chr = sapply(strsplit(PPARG$Chr, "chr"), "[", 2)
colnames(PPARG)[1:5] = c("#CHROM", "POS", "ID", "REF", "ALT")
PPARG_patho_vcf = PPARG[PPARG$Label == "pathogenic",1:5]
PPARG_benign_vcf = PPARG[PPARG$Label == "not pathogenic",1:5]
PPARG_patho_vcf$QUAl = "."
PPARG_patho_vcf$FILTER = "."
PPARG_patho_vcf$INFO = "."
PPARG_benign_vcf$QUAl = "."
PPARG_benign_vcf$FILTER = "."
PPARG_benign_vcf$INFO = "."
write.table(PPARG_patho_vcf, file = "data/J_Li_data/PPARG_patho.vcf", sep = "\t", quote = FALSE, row.names = FALSE)
write.table(PPARG_benign_vcf, file = "data/J_Li_data/PPARG_benign.vcf", sep = "\t", quote = FALSE, row.names = FALSE)


p.normal_PA = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.noQC", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.noQC", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.para_con", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.para_con", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.all_con", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.all_con", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)

other_tools = read.csv(file="data/J_Li_data/PPARG_patho.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/J_Li_data/PPARG_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df1 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df1$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)
gg = ggplot(df1, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15,stroke=2) + 
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=1,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=1,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) + theme(legend.position="none", axis.text = element_text(size = 35), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(plot.title = element_text(hjust = 0.25)) 
ggsave("images/PRC_of_PPARG_with_other_tools_plot.png", width = 10, height = 11)


#SSC
SSC = f[f$Data.source=="SSC",-c(1,4)]
SSC = add_column(SSC, "ID" = paste0("custom_",1:dim(SSC)[1]), .after = "Start")

SSC$Chr = sapply(strsplit(SSC$Chr, "chr"), "[", 2)
colnames(SSC)[1:5] = c("#CHROM", "POS", "ID", "REF", "ALT")
SSC_patho_vcf = SSC[SSC$Label == "asd",1:5]
SSC_benign_vcf = SSC[SSC$Label == "control",1:5]
SSC_patho_vcf$QUAl = "."
SSC_patho_vcf$FILTER = "."
SSC_patho_vcf$INFO = "."
SSC_benign_vcf$QUAl = "."
SSC_benign_vcf$FILTER = "."
SSC_benign_vcf$INFO = "."
write.table(SSC_patho_vcf, file = "data/J_Li_data/SSC_patho.vcf", sep = "\t", quote = FALSE, row.names = FALSE)
write.table(SSC_benign_vcf, file = "data/J_Li_data/SSC_benign.vcf", sep = "\t", quote = FALSE, row.names = FALSE)


p.normal_PA = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.noQC", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.noQC", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.para_con", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.para_con", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.all_con", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.all_con", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)


normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

other_tools = read.csv(file="data/J_Li_data/SSC_patho.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/J_Li_data/SSC_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df1 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df1$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)
gg = ggplot(df1, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=15,stroke=2) + 
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=1,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=1,height=0) +
  
  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) + theme(legend.position="none", axis.text = element_text(size = 35), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(plot.title = element_text(hjust = 0.25)) 
ggsave("images/PRC_of_SSC_with_other_tools_plot.png", width = 10, height = 11)

#checking if any of the variants are in clinvar
SSC_variants = f[f$Data.source=="SSC",-c(1,4)]
SSC_variants$CHROM = substr(SSC_variants$Chr, 4, str_count(SSC_variants$Chr))
SSC_variants$SUM = paste(SSC_variants$CHROM, SSC_variants$Start, SSC_variants$Ref, SSC_variants$Alt, sep = " ")
p.SSC_variants = SSC_variants[SSC_variants$Label=="asd",]
b.SSC_variants = SSC_variants[SSC_variants$Label=="control",]
clinvar_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37.vcf_tableized_org2", sep ="\t", header = T, stringsAsFactors = F)
clinvar_tableized$SUM = paste(clinvar_tableized$CHROM, clinvar_tableized$POS, clinvar_tableized$REF, clinvar_tableized$ALT, sep = " ")
p.clinvar_tableized = clinvar_tableized[clinvar_tableized$SUM %in% p.SSC_variants$SUM,]# &
                                        !(grepl("Conflicting_interpretations_of_pathogenicity", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Uncertain_significance", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Benign", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Likely_benign", clinvar_tableized$CLNSIG)),]
b.clinvar_tableized = clinvar_tableized[clinvar_tableized$SUM %in% b.SSC_variants$SUM,]# &
                                        !(grepl("Conflicting_interpretations_of_pathogenicity", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Uncertain_significance", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Pathogenic", clinvar_tableized$CLNSIG)) &
                                        !(grepl("Likely_pathogenic", clinvar_tableized$CLNSIG)),]
```

TP53-TA, PPARG and SSC combined all in one:
```{r cf_UniFun_TP53-TA, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
###TP53
p.normal_PA = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.noQC", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.noQC", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.para_con", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_paralogs2.all_con", "data/TP53-TA/4704961/TP53-TA_deleterious.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_paralogs2.all_con", "data/TP53-TA/4704961/TP53-TA_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)

other_tools = read.csv(file="data/TP53-TA/4704961/TP53-TA_deleterious.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/TP53-TA/4704961/TP53-TA_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))


# sift_conf = raw_conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
# mcap_conf = raw_conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
# revel_conf = raw_conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
# cadd_conf = raw_conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)
sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df1 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df1$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))
df1$test_set = "TP53"
cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)

###PPARG
f = read.csv("data/J_Li_data/J_Li_data.csv", sep = ",", stringsAsFactors = F, header = T)
PPARG = f[f$Data.source=="PPARG",-c(1,4)]
PPARG = add_column(PPARG, "ID" = paste0("custom_",1:dim(PPARG)[1]), .after = "Start")

PPARG$Chr = sapply(strsplit(PPARG$Chr, "chr"), "[", 2)
colnames(PPARG)[1:5] = c("#CHROM", "POS", "ID", "REF", "ALT")
PPARG_patho_vcf = PPARG[PPARG$Label == "pathogenic",1:5]
PPARG_benign_vcf = PPARG[PPARG$Label == "not pathogenic",1:5]
PPARG_patho_vcf$QUAl = "."
PPARG_patho_vcf$FILTER = "."
PPARG_patho_vcf$INFO = "."
PPARG_benign_vcf$QUAl = "."
PPARG_benign_vcf$FILTER = "."
PPARG_benign_vcf$INFO = "."
write.table(PPARG_patho_vcf, file = "data/J_Li_data/PPARG_patho.vcf", sep = "\t", quote = FALSE, row.names = FALSE)
write.table(PPARG_benign_vcf, file = "data/J_Li_data/PPARG_benign.vcf", sep = "\t", quote = FALSE, row.names = FALSE)


p.normal_PA = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.noQC", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.noQC", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.para_con", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.para_con", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/J_Li_data/PPARG_patho.out_paraloc_paralogs2.all_con", "data/J_Li_data/PPARG_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/J_Li_data/PPARG_benign.out_paraloc_paralogs2.all_con", "data/J_Li_data/PPARG_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)

other_tools = read.csv(file="data/J_Li_data/PPARG_patho.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/J_Li_data/PPARG_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df2 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df2$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))
df2$test_set = "PPARG"

cols = c("Sift" = "#4D5656", "MCap" = "#424949", "CADD" = "#1B2631", "Revel" = "#17202A", "no QC" = "#58D68D", "QC1" = "#45B39D", "QC1+QC2" = "#5DADE2", "QC1+QC3" = "#A569BD", "QC1+QC2+QC3" = "#EC7063")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 19)

###SSC
SSC = f[f$Data.source=="SSC",-c(1,4)]
SSC = add_column(SSC, "ID" = paste0("custom_",1:dim(SSC)[1]), .after = "Start")

SSC$Chr = sapply(strsplit(SSC$Chr, "chr"), "[", 2)
colnames(SSC)[1:5] = c("#CHROM", "POS", "ID", "REF", "ALT")
SSC_patho_vcf = SSC[SSC$Label == "asd",1:5]
SSC_benign_vcf = SSC[SSC$Label == "control",1:5]
SSC_patho_vcf$QUAl = "."
SSC_patho_vcf$FILTER = "."
SSC_patho_vcf$INFO = "."
SSC_benign_vcf$QUAl = "."
SSC_benign_vcf$FILTER = "."
SSC_benign_vcf$INFO = "."
write.table(SSC_patho_vcf, file = "data/J_Li_data/SSC_patho.vcf", sep = "\t", quote = FALSE, row.names = FALSE)
write.table(SSC_benign_vcf, file = "data/J_Li_data/SSC_benign.vcf", sep = "\t", quote = FALSE, row.names = FALSE)


p.normal_PA = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.noQC", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.noQC", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA2 = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.para_con", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.para_con", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA3 = Paralogous_var_align("data/J_Li_data/SSC_patho.out_paraloc_paralogs2.all_con", "data/J_Li_data/SSC_patho.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align("data/J_Li_data/SSC_benign.out_paraloc_paralogs2.all_con", "data/J_Li_data/SSC_benign.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA4 = Alt_con_PA(p.normal_PA4$Total_paralog_annotations,b.normal_PA4$Total_paralog_annotations)
p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
normal_PA5 = Alt_con_PA(p.normal_PA5$Total_paralog_annotations,b.normal_PA5$Total_paralog_annotations)


normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

other_tools = read.csv(file="data/J_Li_data/SSC_patho.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
p.num_of_revel = length(which(other_tools$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools$CADD_phred > 15))

b.other_tools = read.csv(file="data/J_Li_data/SSC_benign.vcf.hg19_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.num_of_revel = length(which(b.other_tools$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools,b.num_of_sift,b.other_tools)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools,b.num_of_mcap,b.other_tools)
revel1_conf = conf_matrix(p.num_of_revel,other_tools,b.num_of_revel,b.other_tools)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools,b.num_of_cadd,b.other_tools)

normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, other_tools, b.normal_PA$true_num_of_paralog_anno, b.other_tools)
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, other_tools, b.normal_PA2$true_num_of_paralog_anno, b.other_tools)
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, other_tools, b.normal_PA3$true_num_of_paralog_anno, b.other_tools)
normal_PA4_conf = conf_matrix(normal_PA4$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA4$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)
normal_PA5_conf = conf_matrix(normal_PA5$p.normal_PA4$true_num_of_paralog_anno, other_tools, normal_PA5$b.normal_PA4$true_num_of_paralog_anno, b.other_tools)

df3 = data.frame(
  Type = c("Sift","Mcap","Cadd","Revel","no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"),
  Precision = c(sift1_conf$PPV,mcap1_conf$PPV,cadd1_conf$PPV,revel1_conf$PPV,normal_PA_conf$PPV,normal_PA2_conf$PPV,normal_PA3_conf$PPV,normal_PA4_conf$PPV,normal_PA5_conf$PPV),
  Precision_CI = c(sift1_conf$PPV_SE*.196,mcap1_conf$PPV_SE*1.96,cadd1_conf$PPV_SE*1.96,revel1_conf$PPV_SE*1.96,normal_PA_conf$PPV_SE*1.96,normal_PA2_conf$PPV_SE*1.96,normal_PA3_conf$PPV_SE*1.96,normal_PA4_conf$PPV_SE*1.96,normal_PA5_conf$PPV_SE*1.96),
  Sensitivity = c(sift1_conf$Sensitivity,mcap1_conf$Sensitivity,cadd1_conf$Sensitivity,revel1_conf$Sensitivity,normal_PA_conf$Sensitivity,normal_PA2_conf$Sensitivity,normal_PA3_conf$Sensitivity,normal_PA4_conf$Sensitivity,normal_PA5_conf$Sensitivity),
  Sensitivity_CI = c(sift1_conf$Sensitivity_SE*1.96,mcap1_conf$Sensitivity_SE*1.96,cadd1_conf$Sensitivity_SE*1.96,revel1_conf$Sensitivity_SE*1.96,normal_PA_conf$Sensitivity_SE*1.96,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA5_conf$Sensitivity_SE*1.96)
)

df3$Type = factor(c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2","QC1+QC2+QC3", "QC1+QC3"), levels = c("Sift", "MCap", "CADD", "Revel", "no QC","QC1","QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))
df3$test_set = "ASD"

cols = c("TP53" = "#004D40", "PPARG" = "#1E88E5", "ASD" = "#D81B60")
shapes = c("Sift" = 0, "MCap" = 1, "CADD" = 2, "Revel" = 5, "no QC" = 15, "QC1" = 16, "QC1+QC2" = 17, "QC1+QC3" = 18, "QC1+QC2+QC3" = 20)

df_all = rbind(df1,df2,df3)

gg = ggplot(df_all, aes(x=Sensitivity, y=Precision, colour = test_set, shape = Type)) + geom_point(size=15,stroke=2,alpha=0.6) +
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +

  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +

  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1),
    axis.title = element_text(size = 16),
    plot.title = element_text(size = 14)) + theme(legend.position="none", axis.text = element_text(size = 35), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(plot.title = element_text(hjust = 0.25))
ggsave("images/PRC_of_TP53_PPARG_SSC_with_other_tools_plot.png", width = 10, height = 12)

gg = ggplot(df_all, aes(x=Sensitivity, y=Precision, colour = test_set, shape = Type)) + geom_point(size=15,stroke=2,alpha=0.6) +
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +

  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +

  scale_shape_manual(values = shapes) +
  scale_color_manual(values = cols) +
  xlim(0,1) + ylim(0,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1),
    axis.title = element_text(size = 16),
    plot.title = element_text(size = 14)) + theme(legend.key.size = unit(5,"line"),axis.text = element_text(size = 35), axis.title.x=element_blank(), axis.title.y=element_blank()) +
   theme(plot.background = element_blank(),
         panel.grid.minor = element_blank())+
  theme(axis.line.x = element_line(color="black", size = 2),
        axis.line.y = element_line(color="black", size = 2)) +
  theme(plot.title = element_text(hjust = 0.25),legend.text=element_text(size=25)) +guides(shape=FALSE)
ggsave("images/PRC_of_TP53_PPARG_SSC_with_other_tools_plot_with_legend.png", width = 10, height = 11)
```


## CALM genes example
PMID papers with pathogenic variants that have genetic evidence:
```{r CALM_genes, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
#for making the VCF file, only run once
CALM_variants = read.csv(file="data/CALM_genes/CALM_gene_variants.csv", sep = ",", header=TRUE, stringsAsFactors=FALSE) 
CALM_variants = CALM_variants[2:18,]
CALM_variants$ID = paste0("custom_",1:17)
CALM_variants$QUAL = "."
CALM_variants$FILTER = "."
# CALM_variants$INFO = paste(CALM_variants$PMID, CALM_variants$build.coord, CALM_variants$gene, CALM_variants$variant, sep = ",")
CALM_variants$INFO = "."
CALM_variants = dplyr::select(CALM_variants, select = -c("PMID","build.coord","gene","variant"))
names(CALM_variants)[1:4] = c("#CHROM","POS","REF","ALT")
CALM_variants = CALM_variants[c("#CHROM","POS","ID","REF","ALT","QUAL","FILTER","INFO")]
write.table(CALM_variants,file="data/CALM_genes/CALM_gene_variants.vcf", na="", row.names = FALSE, col.names = TRUE, sep = "\t", quote=FALSE)

#actual analysis
CALM_pa = Paralogous_var_align("data/CALM_genes/CALM_gene_variants.out_paraloc_paralogs2.noQC", "data/CALM_genes/CALM_gene_variants.out_paraloc_tableized",
                               "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
#full clinvar
p.normal_PA = Paralogous_var_align(Subset = c("Gene", "CALM1"),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Subset = c("Gene", "CALM1"),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)

p.normal_PA2 = Paralogous_var_align(Subset = c("Gene", "CALM1"), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Subset = c("Gene", "CALM1"), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)

p.normal_PA3 = Paralogous_var_align(Subset = c("Gene", "CALM1"), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Subset = c("Gene", "CALM1"), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA4$Total_paralog_annotations = distinct(p.normal_PA4$Total_paralog_annotations)
p.normal_PA4$Unique_variant_annotations = p.normal_PA4$Total_paralog_annotations[!is.na(p.normal_PA4$Total_paralog_annotations$ID.y),]
p.normal_PA4$Unique_variant_annotations = distinct(p.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA4$true_num_of_paralog_anno = length(p.normal_PA4$Unique_variant_annotations$Variant_pos)
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4$Total_paralog_annotations = distinct(b.normal_PA4$Total_paralog_annotations)
b.normal_PA4$Unique_variant_annotations = b.normal_PA4$Total_paralog_annotations[!is.na(b.normal_PA4$Total_paralog_annotations$ID.y),]
b.normal_PA4$Unique_variant_annotations = distinct(b.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA4$true_num_of_paralog_anno = length(b.normal_PA4$Unique_variant_annotations$Variant_pos)
normal_PA4_conf = conf_matrix(p.normal_PA4$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA4$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA5$Total_paralog_annotations = distinct(p.normal_PA5$Total_paralog_annotations)
p.normal_PA5$Unique_variant_annotations = p.normal_PA5$Total_paralog_annotations[!is.na(p.normal_PA5$Total_paralog_annotations$ID.y),]
p.normal_PA5$Unique_variant_annotations = distinct(p.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA5$true_num_of_paralog_anno = length(p.normal_PA5$Unique_variant_annotations$Variant_pos)
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5$Total_paralog_annotations = distinct(b.normal_PA5$Total_paralog_annotations)
b.normal_PA5$Unique_variant_annotations = b.normal_PA5$Total_paralog_annotations[!is.na(b.normal_PA5$Total_paralog_annotations$ID.y),]
b.normal_PA5$Unique_variant_annotations = distinct(b.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA5$true_num_of_paralog_anno = length(b.normal_PA5$Unique_variant_annotations$Variant_pos)
normal_PA5_conf = conf_matrix(p.normal_PA5$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA5$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# Overall dataframe
other_tools_set = t(data.frame(
  # Total = c(sapply(c(revel_conf$con_table[1],revel_conf$con_table[3]),format,scientific = FALSE),"NA","NA","NA","NA"),
  Total = c(sapply(c(dim(p.normal_PA$input_variants)[1],dim(b.normal_PA$input_variants)[1]),format,scientific = FALSE),"NA","NA","NA","NA","NA","NA"),
  PA.noQC = c(sapply(c(normal_PA_conf$con_table[2],normal_PA_conf$con_table[4]),format,scientific = FALSE),
         sapply(c(normal_PA_conf$PPV,normal_PA_conf$PPV_SE*1.96,normal_PA_conf$Sensitivity,normal_PA_conf$Sensitivity_SE*1.96,normal_PA_conf$Specificity,normal_PA_conf$Pvalue$p.value),signif,digits=3)),
  PA.para_con = c(sapply(c(normal_PA2_conf$con_table[2],normal_PA2_conf$con_table[4]),format,scientific = FALSE),
                  sapply(c(normal_PA2_conf$PPV,normal_PA2_conf$PPV_SE*1.96,normal_PA2_conf$Sensitivity,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA2_conf$Specificity,normal_PA2_conf$Pvalue$p.value),signif,digits=3)),
  PA.all_con = c(sapply(c(normal_PA3_conf$con_table[2],normal_PA3_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA3_conf$PPV,normal_PA3_conf$PPV_SE*1.96,normal_PA3_conf$Sensitivity,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA3_conf$Specificity,normal_PA3_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con = c(sapply(c(normal_PA4_conf$con_table[2],normal_PA4_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA4_conf$PPV,normal_PA4_conf$PPV_SE*1.96,normal_PA4_conf$Sensitivity,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA4_conf$Specificity,normal_PA4_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con_w_QC1_no_QC2 = c(sapply(c(normal_PA5_conf$con_table[2],normal_PA5_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA5_conf$PPV,normal_PA5_conf$PPV_SE*1.96,normal_PA5_conf$Sensitivity,normal_PA5_conf$Sensitivity_SE*1.96,normal_PA5_conf$Specificity,normal_PA5_conf$Pvalue$p.value),signif,digits=3))
))
colnames(other_tools_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sens CI","Specificity","P value")


```



## COL3A1
if reuse below, fix code to only get missense protein positions since haven't done that yet.
```{r COL3A1, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
#PA
p.normal_PA = Paralogous_var_align(Subset = c("Gene", "COL3A1"),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Subset = c("Gene", "COL3A1"),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)

p.normal_PA2 = Paralogous_var_align(Subset = c("Gene", "COL3A1"), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Subset = c("Gene", "COL3A1"), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# var_removed = var_rem_matrix(normal_PA_conf$con_table, normal_PA2_conf$con_table, p.normal_PA2$paralog_data, b.normal_PA2$paralog_data)

p.normal_PA3 = Paralogous_var_align(Subset = c("Gene", "COL3A1"), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Subset = c("Gene", "COL3A1"), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA4$Total_paralog_annotations = distinct(p.normal_PA4$Total_paralog_annotations)
p.normal_PA4$Unique_variant_annotations = p.normal_PA4$Total_paralog_annotations[!is.na(p.normal_PA4$Total_paralog_annotations$ID.y),]
p.normal_PA4$Unique_variant_annotations = distinct(p.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA4$true_num_of_paralog_anno = length(p.normal_PA4$Unique_variant_annotations$Variant_pos)
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4$Total_paralog_annotations = distinct(b.normal_PA4$Total_paralog_annotations)
b.normal_PA4$Unique_variant_annotations = b.normal_PA4$Total_paralog_annotations[!is.na(b.normal_PA4$Total_paralog_annotations$ID.y),]
b.normal_PA4$Unique_variant_annotations = distinct(b.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA4$true_num_of_paralog_anno = length(b.normal_PA4$Unique_variant_annotations$Variant_pos)
normal_PA4_conf = conf_matrix(p.normal_PA4$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA4$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA5$Total_paralog_annotations = distinct(p.normal_PA5$Total_paralog_annotations)
p.normal_PA5$Unique_variant_annotations = p.normal_PA5$Total_paralog_annotations[!is.na(p.normal_PA5$Total_paralog_annotations$ID.y),]
p.normal_PA5$Unique_variant_annotations = distinct(p.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA5$true_num_of_paralog_anno = length(p.normal_PA5$Unique_variant_annotations$Variant_pos)
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5$Total_paralog_annotations = distinct(b.normal_PA5$Total_paralog_annotations)
b.normal_PA5$Unique_variant_annotations = b.normal_PA5$Total_paralog_annotations[!is.na(b.normal_PA5$Total_paralog_annotations$ID.y),]
b.normal_PA5$Unique_variant_annotations = distinct(b.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA5$true_num_of_paralog_anno = length(b.normal_PA5$Unique_variant_annotations$Variant_pos)
normal_PA5_conf = conf_matrix(p.normal_PA5$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA5$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# Overall dataframe
other_tools_set = t(data.frame(
  # Total = c(sapply(c(revel_conf$con_table[1],revel_conf$con_table[3]),format,scientific = FALSE),"NA","NA","NA","NA"),
  Total = c(sapply(c(dim(p.normal_PA$input_variants)[1],dim(b.normal_PA$input_variants)[1]),format,scientific = FALSE),"NA","NA","NA","NA","NA","NA"),
  PA.noQC = c(sapply(c(normal_PA_conf$con_table[2],normal_PA_conf$con_table[4]),format,scientific = FALSE),
         sapply(c(normal_PA_conf$PPV,normal_PA_conf$PPV_SE*1.96,normal_PA_conf$Sensitivity,normal_PA_conf$Sensitivity_SE*1.96,normal_PA_conf$Specificity,normal_PA_conf$Pvalue$p.value),signif,digits=3)),
  PA.para_con = c(sapply(c(normal_PA2_conf$con_table[2],normal_PA2_conf$con_table[4]),format,scientific = FALSE),
                  sapply(c(normal_PA2_conf$PPV,normal_PA2_conf$PPV_SE*1.96,normal_PA2_conf$Sensitivity,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA2_conf$Specificity,normal_PA2_conf$Pvalue$p.value),signif,digits=3)),
  PA.all_con = c(sapply(c(normal_PA3_conf$con_table[2],normal_PA3_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA3_conf$PPV,normal_PA3_conf$PPV_SE*1.96,normal_PA3_conf$Sensitivity,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA3_conf$Specificity,normal_PA3_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con_w_QC1_no_QC2 = c(sapply(c(normal_PA5_conf$con_table[2],normal_PA5_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA5_conf$PPV,normal_PA5_conf$PPV_SE*1.96,normal_PA5_conf$Sensitivity,normal_PA5_conf$Sensitivity_SE*1.96,normal_PA5_conf$Specificity,normal_PA5_conf$Pvalue$p.value),signif,digits=3)),  
  PA.alt_con = c(sapply(c(normal_PA4_conf$con_table[2],normal_PA4_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA4_conf$PPV,normal_PA4_conf$PPV_SE*1.96,normal_PA4_conf$Sensitivity,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA4_conf$Specificity,normal_PA4_conf$Pvalue$p.value),signif,digits=3))
))
colnames(other_tools_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sens CI","Specificity","P value")
rownames(other_tools_set) = c("Total variants", "Paralogue Annotation", "Variants remaining after QC1", "Variants remaining after QC1+QC2", "Variants remaining after QC1+QC3", "Variants remaining after QC1+QC2+QC3")
write.table(other_tools_set, "data/all_possible_mutations/COL3A1/clinvar_variants_COL3A1.csv", quote = FALSE, sep = ",", row.names = TRUE, col.names = TRUE)

##all possible missense
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_paralogs2.noQC", "data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized = clinvar_P_LP_tableized[clinvar_P_LP_tableized$SYMBOL == "COL3A1",]
clinvar_P_LP_tableized_protein_pos = as.numeric(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & !(grepl("-", clinvar_P_LP_tableized$Protein_position)),"Protein_position"]))

all_protein_pos = unique(p.normal_PA$Total_paralog_annotations[,"Protein_position.x"])
patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])

aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,patho_protein_pos)

clinvar_COL3A1_variants = data.frame(
  name = "Clinvar_variants",
  source = "Jalview",
  type = "Clinvar_variants",
  start = clinvar_P_LP_tableized_protein_pos,
  end = clinvar_P_LP_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/clinvar_COL3A1_variants.gff")
write.table(clinvar_COL3A1_variants,"data/all_possible_mutations/COL3A1/clinvar_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

all_patho_pred_COL3A1_variants = data.frame(
  name = "all_patho_pred_variants",
  source = "Jalview",
  type = "all_patho_pred_variants",
  start = patho_protein_pos,
  end = patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/all_patho_pred_COL3A1_variants.gff")
write.table(all_patho_pred_COL3A1_variants,"data/all_possible_mutations/COL3A1/all_patho_pred_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_clinvar_and_patho_COL3A1_variants = data.frame(
  name = "aligned_clinvar_and_patho_protein_variants",
  source = "Jalview",
  type = "aligned_clinvar_and_patho_protein_variants",
  start = aligned_clinvar_and_patho_protein_pos,
  end = aligned_clinvar_and_patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/aligned_clinvar_and_patho_COL3A1_variants.gff")
write.table(aligned_clinvar_and_patho_COL3A1_variants,"data/all_possible_mutations/COL3A1/aligned_clinvar_and_patho_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
#benign
b.normal_PA = Paralogous_var_align("data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_paralogs2.noQC", "data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized")
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized = clinvar_B_LB_tableized[clinvar_B_LB_tableized$SYMBOL == "COL3A1",]
clinvar_B_LB_tableized_protein_pos = as.numeric(unique(clinvar_B_LB_tableized[!is.na(clinvar_B_LB_tableized$Protein_position) & !(grepl("-", clinvar_B_LB_tableized$Protein_position)),"Protein_position"]))
benign_protein_pos = unique(b.normal_PA$Total_paralog_annotations[!is.na(b.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])
aligned_benign_clinvar_and_benign_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos,benign_protein_pos)
benign_clinvar_COL3A1_variants = data.frame(
  name = "Benign_Clinvar_variants",
  source = "Jalview",
  type = "Benign_Clinvar_variants",
  start = clinvar_B_LB_tableized_protein_pos,
  end = clinvar_B_LB_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/benign_clinvar_COL3A1_variants.gff")
write.table(benign_clinvar_COL3A1_variants,"data/all_possible_mutations/COL3A1/benign_clinvar_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
all_benign_pred_COL3A1_variants = data.frame(
  name = "all_benign_pred_variants",
  source = "Jalview",
  type = "all_benign_pred_variants",
  start = benign_protein_pos,
  end = benign_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/all_benign_pred_COL3A1_variants.gff")
write.table(all_benign_pred_COL3A1_variants,"data/all_possible_mutations/COL3A1/all_benign_pred_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
aligned_benign_clinvar_and_benign_COL3A1_variants = data.frame(
  name = "aligned_benign_clinvar_and_benign_protein_variants",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_benign_protein_variants",
  start = aligned_benign_clinvar_and_benign_protein_pos,
  end = aligned_benign_clinvar_and_benign_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_benign_COL3A1_variants.gff")
write.table(aligned_benign_clinvar_and_benign_COL3A1_variants,"data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_benign_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_pred_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos, patho_protein_pos)
aligned_benign_clinvar_and_patho_pred_variants = data.frame(
  name = "aligned_benign_clinvar_and_patho_pred_variants",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_patho_pred_variants",
  start = aligned_benign_clinvar_and_patho_pred_protein_pos,
  end = aligned_benign_clinvar_and_patho_pred_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_patho_pred_COL3A1_variants.gff")
write.table(aligned_benign_clinvar_and_patho_pred_variants,"data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_patho_pred_COL3A1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

#QC1 patho
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_paralogs2.para_con", "data/all_possible_mutations/COL3A1/synthetic.vep.cov.table_chrom2_onlyCOL3A1_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized = clinvar_P_LP_tableized[clinvar_P_LP_tableized$SYMBOL == "COL3A1",]
clinvar_P_LP_tableized_protein_pos = as.numeric(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & !(grepl("-", clinvar_P_LP_tableized$Protein_position)),"Protein_position"]))
patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])
aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,patho_protein_pos)
clinvar_COL3A1_variants = data.frame(
  name = "Clinvar_variants_QC1",
  source = "Jalview",
  type = "Clinvar_variants_QC1",
  start = clinvar_P_LP_tableized_protein_pos,
  end = clinvar_P_LP_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/clinvar_COL3A1_QC1_variants.gff")
write.table(clinvar_COL3A1_variants,"data/all_possible_mutations/COL3A1/clinvar_COL3A1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

all_patho_pred_COL3A1_variants = data.frame(
  name = "all_patho_pred_variants_QC1",
  source = "Jalview",
  type = "all_patho_pred_variants_QC1",
  start = patho_protein_pos,
  end = patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/all_patho_pred_COL3A1_QC1_variants.gff")
write.table(all_patho_pred_COL3A1_variants,"data/all_possible_mutations/COL3A1/all_patho_pred_COL3A1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_clinvar_and_patho_COL3A1_variants = data.frame(
  name = "aligned_clinvar_and_patho_protein_variants_QC1",
  source = "Jalview",
  type = "aligned_clinvar_and_patho_protein_variants_QC1",
  start = aligned_clinvar_and_patho_protein_pos,
  end = aligned_clinvar_and_patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/aligned_clinvar_and_patho_COL3A1_QC1_variants.gff")
write.table(aligned_clinvar_and_patho_COL3A1_variants,"data/all_possible_mutations/COL3A1/aligned_clinvar_and_patho_COL3A1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_pred_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos, patho_protein_pos)
aligned_benign_clinvar_and_patho_pred_variants = data.frame(
  name = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
  start = aligned_benign_clinvar_and_patho_pred_protein_pos,
  end = aligned_benign_clinvar_and_patho_pred_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_patho_pred_COL3A1_QC1_variants.gff")
write.table(aligned_benign_clinvar_and_patho_pred_variants,"data/all_possible_mutations/COL3A1/aligned_benign_clinvar_and_patho_pred_COL3A1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_clinvar_and_patho_pred_protein_pos = intersect(aligned_benign_clinvar_and_patho_pred_protein_pos, aligned_clinvar_and_patho_protein_pos)


###Comparing to other tools
p.normal_initial = Paralogous_var_align(Subset = c("Gene", "COL3A1"),Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
other_tools_file = other_tools_file[complete.cases(other_tools_file),]
other_tools_file = filter(other_tools_file, ID %in% p.normal_initial$n_occur$Var1)
p.num_of_revel = length(which(other_tools_file$REVEL_score > 0.5))
p.num_of_sift = length(which(other_tools_file$SIFT_pred == "D"))
p.num_of_mcap = length(which(other_tools_file$M.CAP_pred == "D"))
p.num_of_cadd = length(which(other_tools_file$CADD_phred > 15))

b.normal_initial = Paralogous_var_align(Subset = c("Gene", "COL3A1"),Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized")

b.other_tools_file = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs_no_empty_lines.hg38_multianno.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE) 
b.other_tools_file = b.other_tools_file[complete.cases(b.other_tools_file),]
b.other_tools_file = filter(b.other_tools_file, ID %in% b.normal_initial$n_occur$Var1)
b.num_of_revel = length(which(b.other_tools_file$REVEL_score > 0.5))
b.num_of_sift = length(which(b.other_tools_file$SIFT_pred == "D"))
b.num_of_mcap = length(which(b.other_tools_file$M.CAP_pred == "D"))
b.num_of_cadd = length(which(b.other_tools_file$CADD_phred > 15))

sift1_conf = conf_matrix(p.num_of_sift,other_tools_file,b.num_of_sift,b.other_tools_file)
mcap1_conf = conf_matrix(p.num_of_mcap,other_tools_file,b.num_of_mcap,b.other_tools_file)
revel1_conf = conf_matrix(p.num_of_revel,other_tools_file,b.num_of_revel,b.other_tools_file)
cadd1_conf = conf_matrix(p.num_of_cadd,other_tools_file,b.num_of_cadd,b.other_tools_file)


#PA
p.normal_PA = Paralogous_var_align(Subset = c("ID",other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID),"data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA_conf = conf_matrix(p.normal_PA$true_num_of_paralog_anno, p.normal_PA$input_variants, b.normal_PA$true_num_of_paralog_anno, b.normal_PA$input_variants)

p.normal_PA2 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA2 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.para_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA2_conf = conf_matrix(p.normal_PA2$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA2$true_num_of_paralog_anno, b.normal_PA2$input_variants)

p.normal_PA3 = Paralogous_var_align(Subset = c("ID",other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
b.normal_PA3 = Paralogous_var_align(Subset = c("ID",b.other_tools_file$ID), "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_paralogs2.all_con", "data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
normal_PA3_conf = conf_matrix(p.normal_PA3$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA3$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA4 = list()
p.normal_PA4$Total_paralog_annotations = p.normal_PA3$Total_paralog_annotations[p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA4$Total_paralog_annotations = distinct(p.normal_PA4$Total_paralog_annotations)
p.normal_PA4$Unique_variant_annotations = p.normal_PA4$Total_paralog_annotations[!is.na(p.normal_PA4$Total_paralog_annotations$ID.y),]
p.normal_PA4$Unique_variant_annotations = distinct(p.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA4$true_num_of_paralog_anno = length(p.normal_PA4$Unique_variant_annotations$Variant_pos)
b.normal_PA4 = list()
b.normal_PA4$Total_paralog_annotations = b.normal_PA3$Total_paralog_annotations[b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA3$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA4$Total_paralog_annotations = distinct(b.normal_PA4$Total_paralog_annotations)
b.normal_PA4$Unique_variant_annotations = b.normal_PA4$Total_paralog_annotations[!is.na(b.normal_PA4$Total_paralog_annotations$ID.y),]
b.normal_PA4$Unique_variant_annotations = distinct(b.normal_PA4$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA4$true_num_of_paralog_anno = length(b.normal_PA4$Unique_variant_annotations$Variant_pos)
normal_PA4_conf = conf_matrix(p.normal_PA4$true_num_of_paralog_anno, p.normal_PA3$input_variants, b.normal_PA4$true_num_of_paralog_anno, b.normal_PA3$input_variants)

p.normal_PA5 = list()
p.normal_PA5$Total_paralog_annotations = p.normal_PA2$Total_paralog_annotations[p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
p.normal_PA5$Total_paralog_annotations = distinct(p.normal_PA5$Total_paralog_annotations)
p.normal_PA5$Unique_variant_annotations = p.normal_PA5$Total_paralog_annotations[!is.na(p.normal_PA5$Total_paralog_annotations$ID.y),]
p.normal_PA5$Unique_variant_annotations = distinct(p.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
p.normal_PA5$true_num_of_paralog_anno = length(p.normal_PA5$Unique_variant_annotations$Variant_pos)
b.normal_PA5 = list()
b.normal_PA5$Total_paralog_annotations = b.normal_PA2$Total_paralog_annotations[b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.x==b.normal_PA2$Total_paralog_annotations$ALT_Amino_acids.y,]
b.normal_PA5$Total_paralog_annotations = distinct(b.normal_PA5$Total_paralog_annotations)
b.normal_PA5$Unique_variant_annotations = b.normal_PA5$Total_paralog_annotations[!is.na(b.normal_PA5$Total_paralog_annotations$ID.y),]
b.normal_PA5$Unique_variant_annotations = distinct(b.normal_PA5$Unique_variant_annotations[c("Variant_pos", "ID.x", "REF.x", "ALT.x")])
b.normal_PA5$true_num_of_paralog_anno = length(b.normal_PA5$Unique_variant_annotations$Variant_pos)
normal_PA5_conf = conf_matrix(p.normal_PA5$true_num_of_paralog_anno, p.normal_PA2$input_variants, b.normal_PA5$true_num_of_paralog_anno, b.normal_PA2$input_variants)

# Overall dataframe
other_tools_set = t(data.frame(
  # Total = c(sapply(c(revel_conf$con_table[1],revel_conf$con_table[3]),format,scientific = FALSE),"NA","NA","NA","NA"),
  Total = c(sapply(c(dim(other_tools_file)[1],dim(b.other_tools_file)[1]),format,scientific = FALSE),"NA","NA","NA","NA","NA","NA"),
  Sift = c(sapply(c(sift1_conf$con_table[2],sift1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(sift1_conf$PPV,sift1_conf$PPV_SE*1.96,sift1_conf$Sensitivity,sift1_conf$Sensitivity_SE*1.96,sift1_conf$Specificity,sift1_conf$Pvalue$p.value),signif,digits=3)),
  MCap = c(sapply(c(mcap1_conf$con_table[2],mcap1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(mcap1_conf$PPV,mcap1_conf$PPV_SE*1.96,mcap1_conf$Sensitivity,mcap1_conf$Sensitivity_SE*1.96,mcap1_conf$Specificity,mcap1_conf$Pvalue$p.value),signif,digits=3)),
  Cadd = c(sapply(c(cadd1_conf$con_table[2],cadd1_conf$con_table[4]),format,scientific = FALSE),
           sapply(c(cadd1_conf$PPV,cadd1_conf$PPV_SE*1.96,cadd1_conf$Sensitivity,cadd1_conf$Sensitivity_SE*1.96,cadd1_conf$Specificity,cadd1_conf$Pvalue$p.value),signif,digits=3)),
  Revel = c(sapply(c(revel1_conf$con_table[2],revel1_conf$con_table[4]),format,scientific = FALSE),
            sapply(c(revel1_conf$PPV,revel1_conf$PPV_SE*1.96,revel1_conf$Sensitivity,revel1_conf$Sensitivity_SE*1.96,revel1_conf$Specificity,revel1_conf$Pvalue$p.value),signif,digits=3)),
  PA.noQC = c(sapply(c(normal_PA_conf$con_table[2],normal_PA_conf$con_table[4]),format,scientific = FALSE),
         sapply(c(normal_PA_conf$PPV,normal_PA_conf$PPV_SE*1.96,normal_PA_conf$Sensitivity,normal_PA_conf$Sensitivity_SE*1.96,normal_PA_conf$Specificity,normal_PA_conf$Pvalue$p.value),signif,digits=3)),
  PA.para_con = c(sapply(c(normal_PA2_conf$con_table[2],normal_PA2_conf$con_table[4]),format,scientific = FALSE),
                  sapply(c(normal_PA2_conf$PPV,normal_PA2_conf$PPV_SE*1.96,normal_PA2_conf$Sensitivity,normal_PA2_conf$Sensitivity_SE*1.96,normal_PA2_conf$Specificity,normal_PA2_conf$Pvalue$p.value),signif,digits=3)),
  PA.all_con = c(sapply(c(normal_PA3_conf$con_table[2],normal_PA3_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA3_conf$PPV,normal_PA3_conf$PPV_SE*1.96,normal_PA3_conf$Sensitivity,normal_PA3_conf$Sensitivity_SE*1.96,normal_PA3_conf$Specificity,normal_PA3_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con = c(sapply(c(normal_PA4_conf$con_table[2],normal_PA4_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA4_conf$PPV,normal_PA4_conf$PPV_SE*1.96,normal_PA4_conf$Sensitivity,normal_PA4_conf$Sensitivity_SE*1.96,normal_PA4_conf$Specificity,normal_PA4_conf$Pvalue$p.value),signif,digits=3)),
  PA.alt_con_w_QC1_no_QC2 = c(sapply(c(normal_PA5_conf$con_table[2],normal_PA5_conf$con_table[4]),format,scientific = FALSE),
                 sapply(c(normal_PA5_conf$PPV,normal_PA5_conf$PPV_SE*1.96,normal_PA5_conf$Sensitivity,normal_PA5_conf$Sensitivity_SE*1.96,normal_PA5_conf$Specificity,normal_PA5_conf$Pvalue$p.value),signif,digits=3))
))
colnames(other_tools_set) = c("Pathogenic variants","Benign variants","PPV","PPV CI","Sensitivity","Sens CI","Specificity","P value")
##REMINDER TO FIX APPENDING OF PARAZ SET AS HAVE NOT YET ADDED CI FOR PPV AND SENS
# paraz_set = paraz_set[-1,]
# other_tools_set = rbind(other_tools_set,paraz_set)
# rownames(other_tools_set) = c("Total","Sift","MCap","CADD","Revel","Paralogue Annotation","Paralogue Annotation (QC1)","Paralogue Annotation (QC2)", "Paralogue Annotation (QC3)","Para z_score cutoff 0","Para z_score cutoff 1","Para z_score cutoff 2","Para z_score cutoff 3","Para z_score cutoff 4","Para z_score cutoff 5","Para z_score cutoff 6","Para z_score cutoff 7","Para z_score cutoff 8","Para z_score cutoff 9","Para z_score cutoff 10","Para z_score cutoff 11")
# write.table(other_tools_set, file = "data/other_tools_set.csv", na="", row.names = TRUE, col.names = TRUE, sep = ",", quote=FALSE)
# 

other_tools_data = data.frame(
  Precision = c(as.numeric(other_tools_set[2:10,3])),
  Precision_CI = c(as.numeric(other_tools_set[2:10,4])),
  Sensitivity = c(as.numeric(other_tools_set[2:10,5])),
  Sensitivity_CI = c(as.numeric(other_tools_set[2:10,6]))
  )
other_tools_data$Type = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1" ,"QC1+QC2", "QC1+QC2+QC3", "QC1+QC3")
# other_tools_data = rbind(other_tools_data, prc_data2[1:4,])
other_tools_data$Type = factor(other_tools_data$Type, levels = c("Sift", "MCap", "CADD", "Revel", "no QC", "QC1" ,"QC1+QC2", "QC1+QC3", "QC1+QC2+QC3"))

gg = ggplot(other_tools_data, aes(x=Sensitivity, y=Precision, colour = Type, shape = Type)) + geom_point(size=4) + 
  # geom_errorbar(aes(ymin=Precision-Precision_CI, ymax=Precision+Precision_CI),colour="black",linetype=1,size=.2,width=.5) +
  # geom_errorbarh(aes(xmin=Sensitivity-Sensitivity_CI, xmax=Sensitivity+Sensitivity_CI),colour="black",linetype=1,size=.2,height=0) +
  
  geom_errorbar(aes(ymin=sapply(Precision-Precision_CI, max, 0), ymax=sapply(Precision+Precision_CI, min, 1)),colour="black",linetype=1,size=.2,width=0) +
  geom_errorbarh(aes(xmin=sapply(Sensitivity-Sensitivity_CI, max, 0), xmax=sapply(Sensitivity+Sensitivity_CI, min, 1)),colour="black",linetype=1,size=.2,height=0) +
  
  scale_shape_manual(values=c(0, 1, 2, 5, 15, 16, 17, 19, 18)) +
  xlim(0,1) + ylim(0.75,1)  + theme(plot.subtitle = element_text(vjust = 1),
    plot.caption = element_text(vjust = 1), 
    axis.title = element_text(size = 16), 
    plot.title = element_text(size = 14)) +labs(title = "Precision-Sensitivity space plot of Paralogue Annotation", 
    x = "Sensitivity") + theme(axis.text = element_text(size = 13)) + theme(plot.title = element_text(hjust = 0.25)) 
ggsave("images/PRC_of_COL3A1.png")

```


## SCN5A
```{r SCN5A, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
#all possible missense
#####
p.normal_PA = Paralogous_var_align(Overlap = 0, "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_paralogs2.noQC", "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

# clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized = clinvar_P_LP_tableized[clinvar_P_LP_tableized$SYMBOL == "SCN5A" &
                                                   !is.na(clinvar_P_LP_tableized$Protein_position) & 
                                                  !(grepl("-", clinvar_P_LP_tableized$Protein_position)) &
                                                  !(grepl("-", clinvar_P_LP_tableized$Amino_acids)) &
                                                  !(grepl("\\*", clinvar_P_LP_tableized$Amino_acids)) &
                                                  grepl("/", clinvar_P_LP_tableized$Amino_acids) &
                                                  str_count(clinvar_P_LP_tableized$REF) == str_count(clinvar_P_LP_tableized$ALT) &
                                                  str_count(clinvar_P_LP_tableized$Amino_acids) == 3,]
clinvar_P_LP_tableized_protein_pos = as.numeric(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & !(grepl("-", clinvar_P_LP_tableized$Protein_position)),"Protein_position"]))

all_protein_pos = unique(p.normal_PA$Total_paralog_annotations[,"Protein_position.x"])
patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])

aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,patho_protein_pos)

clinvar_SCN5A_variants = data.frame(
  name = "Clinvar_variants",
  source = "Jalview",
  type = "Clinvar_variants",
  start = clinvar_P_LP_tableized_protein_pos,
  end = clinvar_P_LP_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/clinvar_SCN5A_variants.gff")
write.table(clinvar_SCN5A_variants,"data/all_possible_mutations/SCN5A/clinvar_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

all_patho_pred_SCN5A_variants = data.frame(
  name = "all_patho_pred_variants",
  source = "Jalview",
  type = "all_patho_pred_variants",
  start = patho_protein_pos,
  end = patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/all_patho_pred_SCN5A_variants.gff")
write.table(all_patho_pred_SCN5A_variants,"data/all_possible_mutations/SCN5A/all_patho_pred_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_clinvar_and_patho_SCN5A_variants = data.frame(
  name = "aligned_clinvar_and_patho_protein_variants",
  source = "Jalview",
  type = "aligned_clinvar_and_patho_protein_variants",
  start = aligned_clinvar_and_patho_protein_pos,
  end = aligned_clinvar_and_patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/aligned_clinvar_and_patho_SCN5A_variants.gff")
write.table(aligned_clinvar_and_patho_SCN5A_variants,"data/all_possible_mutations/SCN5A/aligned_clinvar_and_patho_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
#benign
b.normal_PA = Paralogous_var_align(Overlap = 0, "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_paralogs2.noQC", "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized")
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized = clinvar_B_LB_tableized[clinvar_B_LB_tableized$SYMBOL == "SCN5A" &
                                                  !is.na(clinvar_B_LB_tableized$Protein_position) & 
                                                  !(grepl("-", clinvar_B_LB_tableized$Protein_position)) &
                                                  !(grepl("-", clinvar_B_LB_tableized$Amino_acids)) &
                                                  !(grepl("\\*", clinvar_B_LB_tableized$Amino_acids)) &
                                                  grepl("/", clinvar_B_LB_tableized$Amino_acids) &
                                                  str_count(clinvar_B_LB_tableized$REF) == str_count(clinvar_B_LB_tableized$ALT) &
                                                  str_count(clinvar_B_LB_tableized$Amino_acids) == 3,]
clinvar_B_LB_tableized_protein_pos = as.numeric(unique(clinvar_B_LB_tableized[!is.na(clinvar_B_LB_tableized$Protein_position) & !(grepl("-", clinvar_B_LB_tableized$Protein_position)),"Protein_position"]))
benign_protein_pos = unique(b.normal_PA$Total_paralog_annotations[!is.na(b.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])
aligned_benign_clinvar_and_benign_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos,benign_protein_pos)
benign_clinvar_SCN5A_variants = data.frame(
  name = "Benign_Clinvar_variants",
  source = "Jalview",
  type = "Benign_Clinvar_variants",
  start = clinvar_B_LB_tableized_protein_pos,
  end = clinvar_B_LB_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/benign_clinvar_SCN5A_variants.gff")
write.table(benign_clinvar_SCN5A_variants,"data/all_possible_mutations/SCN5A/benign_clinvar_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
all_benign_pred_SCN5A_variants = data.frame(
  name = "all_benign_pred_variants",
  source = "Jalview",
  type = "all_benign_pred_variants",
  start = benign_protein_pos,
  end = benign_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/all_benign_pred_SCN5A_variants.gff")
write.table(all_benign_pred_SCN5A_variants,"data/all_possible_mutations/SCN5A/all_benign_pred_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
aligned_benign_clinvar_and_benign_SCN5A_variants = data.frame(
  name = "aligned_benign_clinvar_and_benign_protein_variants",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_benign_protein_variants",
  start = aligned_benign_clinvar_and_benign_protein_pos,
  end = aligned_benign_clinvar_and_benign_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_benign_SCN5A_variants.gff")
write.table(aligned_benign_clinvar_and_benign_SCN5A_variants,"data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_benign_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_pred_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos, patho_protein_pos)
aligned_benign_clinvar_and_patho_pred_variants = data.frame(
  name = "aligned_benign_clinvar_and_patho_pred_variants",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_patho_pred_variants",
  start = aligned_benign_clinvar_and_patho_pred_protein_pos,
  end = aligned_benign_clinvar_and_patho_pred_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_patho_pred_SCN5A_variants.gff")
write.table(aligned_benign_clinvar_and_patho_pred_variants,"data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_patho_pred_SCN5A_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)
#####
#QC1 patho
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_paralogs2.para_con", "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_P_LP_tableized = clinvar_P_LP_tableized[clinvar_P_LP_tableized$SYMBOL == "SCN5A",]
# clinvar_P_LP_tableized_protein_pos = as.numeric(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & !(grepl("-", clinvar_P_LP_tableized$Protein_position)),"Protein_position"]))
patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])
aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,patho_protein_pos)
clinvar_SCN5A_variants = data.frame(
  name = "Clinvar_variants_QC1",
  source = "Jalview",
  type = "Clinvar_variants_QC1",
  start = clinvar_P_LP_tableized_protein_pos,
  end = clinvar_P_LP_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/clinvar_SCN5A_QC1_variants.gff")
write.table(clinvar_SCN5A_variants,"data/all_possible_mutations/SCN5A/clinvar_SCN5A_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

all_patho_pred_SCN5A_variants = data.frame(
  name = "all_patho_pred_variants_QC1",
  source = "Jalview",
  type = "all_patho_pred_variants_QC1",
  start = patho_protein_pos,
  end = patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/all_patho_pred_SCN5A_QC1_variants.gff")
write.table(all_patho_pred_SCN5A_variants,"data/all_possible_mutations/SCN5A/all_patho_pred_SCN5A_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_clinvar_and_patho_SCN5A_variants = data.frame(
  name = "aligned_clinvar_and_patho_protein_variants_QC1",
  source = "Jalview",
  type = "aligned_clinvar_and_patho_protein_variants_QC1",
  start = aligned_clinvar_and_patho_protein_pos,
  end = aligned_clinvar_and_patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/aligned_clinvar_and_patho_SCN5A_QC1_variants.gff")
write.table(aligned_clinvar_and_patho_SCN5A_variants,"data/all_possible_mutations/SCN5A/aligned_clinvar_and_patho_SCN5A_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_pred_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos, patho_protein_pos)
aligned_benign_clinvar_and_patho_pred_variants = data.frame(
  name = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
  source = "Jalview",
  type = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
  start = aligned_benign_clinvar_and_patho_pred_protein_pos,
  end = aligned_benign_clinvar_and_patho_pred_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_patho_pred_SCN5A_QC1_variants.gff")
write.table(aligned_benign_clinvar_and_patho_pred_variants,"data/all_possible_mutations/SCN5A/aligned_benign_clinvar_and_patho_pred_SCN5A_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_benign_clinvar_and_patho_clinvar_and_patho_pred_protein_pos = intersect(aligned_benign_clinvar_and_patho_pred_protein_pos, aligned_clinvar_and_patho_protein_pos)

##% increase of residues with "information"
num_clinvar_P_LP = length(clinvar_P_LP_tableized_protein_pos)
num_PA_pred_patho = length(patho_protein_pos)-length(aligned_clinvar_and_patho_protein_pos)
perc_inc = (num_PA_pred_patho + num_clinvar_P_LP)/num_clinvar_P_LP


#QC1+QC3 patho
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_paralogs2.para_con", "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

QC1_QC3_patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y) & p.normal_PA$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA$Total_paralog_annotations$ALT_Amino_acids.y,"Protein_position.x"])

QC1_QC3_aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,QC1_QC3_patho_protein_pos)

QC1_QC3_num_PA_pred_patho = length(QC1_QC3_patho_protein_pos)-length(QC1_QC3_aligned_clinvar_and_patho_protein_pos)
QC1_QC3_perc_inc = (QC1_QC3_num_PA_pred_patho + num_clinvar_P_LP)/num_clinvar_P_LP

#QC1+QC2+QC3
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_paralogs2.all_con", "data/all_possible_mutations/SCN5A/synthetic.vep.cov.table_chrom3_onlySCN5A_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

QC1_QC2_QC3_patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y) & p.normal_PA$Total_paralog_annotations$ALT_Amino_acids.x==p.normal_PA$Total_paralog_annotations$ALT_Amino_acids.y,"Protein_position.x"])

aligned_clinvar_and_patho_QC1_QC2_QC3_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,QC1_QC2_QC3_patho_protein_pos)

QC1_QC2_QC3_num_PA_pred_patho = length(QC1_QC2_QC3_patho_protein_pos)-length(aligned_clinvar_and_patho_QC1_QC2_QC3_protein_pos)
QC1_QC2_QC3_perc_inc = (QC1_QC2_QC3_num_PA_pred_patho + num_clinvar_P_LP)/num_clinvar_P_LP
```

## KCNQ1
```{r KCNQ1, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
#all possible missense
#QC1 patho
p.normal_PA = Paralogous_var_align("data/all_possible_mutations/KCNQ1/synthetic.vep.cov.table_chrom11_onlyKCNQ1_variants.out_paraloc_paralogs2.para_con", "data/all_possible_mutations/KCNQ1/synthetic.vep.cov.table_chrom11_onlyKCNQ1_variants.out_paraloc_tableized", "data/clinvar/clinvar_20190520_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_P_LP_tableized = clinvar_P_LP_tableized[clinvar_P_LP_tableized$SYMBOL == "KCNQ1" &
                                                   !is.na(clinvar_P_LP_tableized$Protein_position) & 
                                                  !(grepl("-", clinvar_P_LP_tableized$Protein_position)) &
                                                  !(grepl("-", clinvar_P_LP_tableized$Amino_acids)) &
                                                  !(grepl("\\*", clinvar_P_LP_tableized$Amino_acids)) &
                                                  grepl("/", clinvar_P_LP_tableized$Amino_acids) &
                                                  str_count(clinvar_P_LP_tableized$REF) == str_count(clinvar_P_LP_tableized$ALT) &
                                                  str_count(clinvar_P_LP_tableized$Amino_acids) == 3,]
clinvar_P_LP_tableized_protein_pos = as.numeric(unique(clinvar_P_LP_tableized[!is.na(clinvar_P_LP_tableized$Protein_position) & !(grepl("-", clinvar_P_LP_tableized$Protein_position)),"Protein_position"]))

patho_protein_pos = unique(p.normal_PA$Total_paralog_annotations[!is.na(p.normal_PA$Total_paralog_annotations$ID.y),"Protein_position.x"])

aligned_clinvar_and_patho_protein_pos = intersect(clinvar_P_LP_tableized_protein_pos,patho_protein_pos)

clinvar_KCNQ1_variants = data.frame(
  name = "Clinvar_variants",
  source = "Jalview",
  type = "Clinvar_variants",
  start = clinvar_P_LP_tableized_protein_pos,
  end = clinvar_P_LP_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/KCNQ1/clinvar_KCNQ1_variants.gff")
write.table(clinvar_KCNQ1_variants,"data/all_possible_mutations/KCNQ1/clinvar_KCNQ1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

all_patho_pred_KCNQ1_variants = data.frame(
  name = "all_patho_pred_variants_QC1",
  source = "Jalview",
  type = "all_patho_pred_variants_QC1",
  start = patho_protein_pos,
  end = patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/KCNQ1/all_patho_pred_KCNQ1_QC1_variants.gff")
write.table(all_patho_pred_KCNQ1_variants,"data/all_possible_mutations/KCNQ1/all_patho_pred_KCNQ1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

aligned_clinvar_and_patho_KCNQ1_variants = data.frame(
  name = "aligned_clinvar_and_patho_protein_variants_QC1",
  source = "Jalview",
  type = "aligned_clinvar_and_patho_protein_variants_QC1",
  start = aligned_clinvar_and_patho_protein_pos,
  end = aligned_clinvar_and_patho_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/KCNQ1/aligned_clinvar_and_patho_KCNQ1_QC1_variants.gff")
write.table(aligned_clinvar_and_patho_KCNQ1_variants,"data/all_possible_mutations/KCNQ1/aligned_clinvar_and_patho_KCNQ1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

#benign
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190520_GRCh37_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
clinvar_B_LB_tableized = clinvar_B_LB_tableized[clinvar_B_LB_tableized$SYMBOL == "KCNQ1" &
                                                  !is.na(clinvar_B_LB_tableized$Protein_position) & 
                                                  !(grepl("-", clinvar_B_LB_tableized$Protein_position)) &
                                                  !(grepl("-", clinvar_B_LB_tableized$Amino_acids)) &
                                                  !(grepl("\\*", clinvar_B_LB_tableized$Amino_acids)) &
                                                  grepl("/", clinvar_B_LB_tableized$Amino_acids) &
                                                  str_count(clinvar_B_LB_tableized$Codons) == 7 &
                                                  str_count(clinvar_B_LB_tableized$Amino_acids) == 3,]
clinvar_B_LB_tableized_protein_pos = as.numeric(unique(clinvar_B_LB_tableized[!is.na(clinvar_B_LB_tableized$Protein_position) & !(grepl("-", clinvar_B_LB_tableized$Protein_position)),"Protein_position"]))

benign_clinvar_KCNQ1_variants = data.frame(
  name = "Benign_Clinvar_variants",
  source = "Jalview",
  type = "Benign_Clinvar_variants",
  start = clinvar_B_LB_tableized_protein_pos,
  end = clinvar_B_LB_tableized_protein_pos,
  score = 0.0,
  strand = ".",
  phase = "."
) 
write("##gff-version 2","data/all_possible_mutations/KCNQ1/benign_clinvar_KCNQ1_variants.gff")
write.table(benign_clinvar_KCNQ1_variants,"data/all_possible_mutations/KCNQ1/benign_clinvar_KCNQ1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)


aligned_benign_clinvar_and_patho_pred_protein_pos = intersect(clinvar_B_LB_tableized_protein_pos, patho_protein_pos) #0 so cant use code below
# aligned_benign_clinvar_and_patho_pred_variants = data.frame(
#   name = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
#   source = "Jalview",
#   type = "aligned_benign_clinvar_and_patho_pred_variants_QC1",
#   start = aligned_benign_clinvar_and_patho_pred_protein_pos,
#   end = aligned_benign_clinvar_and_patho_pred_protein_pos,
#   score = 0.0,
#   strand = ".",
#   phase = "."
# ) 
write("##gff-version 2","data/all_possible_mutations/KCNQ1/aligned_benign_clinvar_and_patho_pred_KCNQ1_QC1_variants.gff")
# write.table(aligned_benign_clinvar_and_patho_pred_variants,"data/all_possible_mutations/KCNQ1/aligned_benign_clinvar_and_patho_pred_KCNQ1_QC1_variants.gff", quote = FALSE, sep = "\t", row.names = FALSE, col.names = FALSE, append = TRUE)

```




## Gnomad
There's 17.2 million variants in gnomad
```{r gnomad, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
# p.normal_PA = Paralogous_var_align("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split00_only_var_info.out_paraloc_paralogs2.noQC",
#                                    "data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split00_only_var_info.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# 
# p.normal_PA2 = Paralogous_var_align("data/gnomad_exome_r2.1.1/testing/gnomad.exomes.r2.1.1.sites_split00_only_var_info_forTESTING_with_customIDs.out_paraloc_paralogs2.noQC",
#                                    "data/gnomad_exome_r2.1.1/testing/gnomad.exomes.r2.1.1.sites_split00_only_var_info_forTESTING_with_customIDs.out_paraloc_tableized", "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")

num_of_var_pred_patho = 0
var_pred_patho = data.frame()
for (i in 0:9){
p.normal_PA = Paralogous_var_align(
  paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split0",i,"_only_var_info.out_paraloc_paralogs2.para_con"), paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split0",i,"_only_var_info.out_paraloc_tableized"), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
num_of_var_pred_patho = num_of_var_pred_patho + p.normal_PA$true_num_of_paralog_anno
var_pred_patho = rbind(var_pred_patho,p.normal_PA$Unique_variant_annotations[c("Variant_pos","REF.x","ALT.x")])
}
for (i in 10:17){
p.normal_PA = Paralogous_var_align(
  paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_paralogs2.para_con"), paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_tableized"), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
num_of_var_pred_patho = num_of_var_pred_patho + p.normal_PA$true_num_of_paralog_anno
var_pred_patho = rbind(var_pred_patho,p.normal_PA$Unique_variant_annotations[c("Variant_pos","REF.x","ALT.x")])
}
for (i in 18:25){
p.normal_PA = Paralogous_var_align(
  paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_paralogs2.para_con"), paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_tableized"), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
num_of_var_pred_patho = num_of_var_pred_patho + p.normal_PA$true_num_of_paralog_anno
var_pred_patho = rbind(var_pred_patho,p.normal_PA$Unique_variant_annotations[c("Variant_pos","REF.x","ALT.x")])
}
for (i in 26:34){
p.normal_PA = Paralogous_var_align(
  paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_paralogs2.para_con"), paste0("data/gnomad_exome_r2.1.1/gnomad.exomes.r2.1.1.sites_split",i,"_only_var_info.out_paraloc_tableized"), "data/clinvar/clinvar_20181028_GRCh37_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
num_of_var_pred_patho = num_of_var_pred_patho + p.normal_PA$true_num_of_paralog_anno
var_pred_patho = rbind(var_pred_patho,p.normal_PA$Unique_variant_annotations[c("Variant_pos","REF.x","ALT.x")])
}
num_of_var_pred_patho #66711 out of the 17.2 million gnomad variants are predicted to be pathogenic
var_pred_patho$ALL = paste(var_pred_patho$Variant_pos,var_pred_patho$REF.x,var_pred_patho$ALT.x,sep = " ")

# clinvar_P_LP_vcf_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.vcf_tableized_org", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# # clinvar_P_LP_vcf_tableized$Variant_pos = paste(clinvar_P_LP_vcf_tableized$CHROM,clinvar_P_LP_vcf_tableized$POS, sep = " ")
# clinvar_P_LP_vcf_tableized$ALL = paste(clinvar_P_LP_vcf_tableized$CHROM,clinvar_P_LP_vcf_tableized$POS,clinvar_P_LP_vcf_tableized$REF,clinvar_P_LP_vcf_tableized$ALT,sep = " ")

clinvar_P_LP_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_P_LP_tableized$Variant_pos = paste(clinvar_P_LP_tableized$CHROM,clinvar_P_LP_tableized$POS, sep = " ")
clinvar_P_LP_tableized$ALL = paste(clinvar_P_LP_tableized$CHROM,clinvar_P_LP_tableized$POS,clinvar_P_LP_tableized$REF,clinvar_P_LP_tableized$ALT,sep = " ")
clinvar_B_LB_tableized = read.csv(file="data/clinvar/clinvar_20190114_GRCh38_onlyBenign_and_Likely_benign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
# clinvar_B_LB_tableized$Variant_pos = paste(clinvar_B_LB_tableized$CHROM,clinvar_B_LB_tableized$POS, sep = " ")
clinvar_B_LB_tableized$ALL = paste(clinvar_B_LB_tableized$CHROM,clinvar_B_LB_tableized$POS,clinvar_B_LB_tableized$REF,clinvar_B_LB_tableized$ALT,sep = " ")

full_clinvar_tableized = read.csv(file="data/clinvar/clinvar_20190114.vcf_tableized_org", sep = "\t", header = TRUE, stringsAsFactors = FALSE)
# full_clinvar_tableized$Variant_pos = paste(full_clinvar_tableized$CHROM,full_clinvar_tableized$POS, sep = " ")
full_clinvar_tableized$ALL = paste(full_clinvar_tableized$CHROM,full_clinvar_tableized$POS,full_clinvar_tableized$REF,full_clinvar_tableized$ALT,sep = " ")


length(intersect(var_pred_patho$ALL,clinvar_P_LP_tableized$ALL)) #only 3 predicted patho variants are in clinvar P/LP
length(intersect(var_pred_patho$ALL,clinvar_B_LB_tableized$ALL)) #only 3 predicted patho variants are in clinvar B/LB
length(intersect(var_pred_patho$ALL,full_clinvar_tableized$ALL)) #27 all together, meaning 21 VUS

length(setdiff(var_pred_patho$ALL,clinvar_P_LP_tableized$ALL))
```
66711 out of the 17.2 million gnomad variants are predicted to be pathogenic
only 3 predicted patho variants are in clinvar P/LP
only 3 predicted patho variants are in clinvar B/LB
27 all together, meaning 21 VUS

## GO protein classes of clinvar P/LP variants with PA noQC

Using GO Panther enrichment analysis (http://pantherdb.org/) to identify common protein classes among the genes that had paralogue annotations.

```{r GO_protein_class, cache=TRUE, echo = FALSE, results='asis', message = FALSE, warning=FALSE}
#Panther GO
protein_classes = read.csv(file="data/clinvar/genes_with_paralogue_annotation_noQC_GO_protein_class.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE)
protein_classes$V4 = as.numeric(sub("%", "", protein_classes$V4))
protein_classes$V5 = as.numeric(sub("%", "", protein_classes$V5))
protein_classes = protein_classes[,2:5]
colnames(protein_classes) = c("Protein_class", "number_of_genes", "Percent_of_gene_hit_against_total_number_of_genes", "Percent_of_gene_hit_against_total_number_of_Protein_Class_hits")
protein_classes = protein_classes[order(protein_classes$number_of_genes, decreasing = T),] 
protein_classes$stage = "Only genes with\nParalogue Annotations"
gg_protein_class = ggplot(protein_classes, aes(y=number_of_genes, x=reorder(Protein_class, number_of_genes), fill = stage)) +
  geom_bar(stat="identity", position = "dodge") +
  coord_flip() +
  # geom_point() + 
  # geom_text(aes(label=Protein_class, position = "dodge"), position = position_dodge(width = 0), hjust =0 , vjust=0, angle = 20) +
  theme(
  #   axis.text.x = element_text(angle = -45, hjust = -0),
  #   plot.margin = unit(c(5.5, 5.5, 5.5, 10), "points")
    # axis.title.y=element_blank(),
    legend.title=element_blank()
    # legend.position="none"
    # axis.text.x=element_blank()
    # axis.ticks.x=element_blank()
    ) +
  scale_fill_manual(values=c("#BB8FCE")) + 
  guides(fill = guide_legend(reverse=TRUE)) +
  scale_y_continuous(breaks = c(0,50,100,150,200,250,300,350,400)) +
  xlab("PANTHER Protein Class") + theme(legend.key.size = unit(2,"line")) +ylab("Number of Genes")
ggsave("images/GO_protein_classes_of_only_genes_w_annotations.png", height = 8, width = 12)
###control - 3123 genes with paralogues and aligned positions
protein_classes2 = read.csv(file="data/clinvar/3123_genes_w_paralogues_w_aln_pos_GO_protein_class.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE)
protein_classes2$V4 = as.numeric(sub("%", "", protein_classes2$V4))
protein_classes2$V5 = as.numeric(sub("%", "", protein_classes2$V5))
protein_classes2 = protein_classes2[,2:5]
colnames(protein_classes2) = c("Protein_class", "number_of_genes", "Percent_of_gene_hit_against_total_number_of_genes", "Percent_of_gene_hit_against_total_number_of_Protein_Class_hits")
protein_classes2 = protein_classes2[order(protein_classes2$number_of_genes, decreasing = T),] 
protein_classes2$stage = "Only genes with\nParalogues"
###control - 4340 clinvar genes with only missense
protein_classes3 = read.csv(file="data/clinvar/4225_clinvar_genes_w_only_missense_GO_protein_class.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE)
protein_classes3$V4 = as.numeric(sub("%", "", protein_classes3$V4))
protein_classes3$V5 = as.numeric(sub("%", "", protein_classes3$V5))
protein_classes3 = protein_classes3[,2:5]
colnames(protein_classes3) = c("Protein_class", "number_of_genes", "Percent_of_gene_hit_against_total_number_of_genes", "Percent_of_gene_hit_against_total_number_of_Protein_Class_hits")
protein_classes3 = protein_classes3[order(protein_classes3$number_of_genes, decreasing = T),] 
protein_classes3$stage = "All genes with P/LP\nmissense Clinvar variants"

###control - 1947 disease clinvar genes with disease paralogues
protein_classes4 = read.csv(file="data/clinvar/1947_dis_clinvar_genes_w_dis_para_GO_protein_class.txt", sep = "\t", header = FALSE, stringsAsFactors = FALSE)
protein_classes4$V4 = as.numeric(sub("%", "", protein_classes4$V4))
protein_classes4$V5 = as.numeric(sub("%", "", protein_classes4$V5))
protein_classes4 = protein_classes4[,2:5]
colnames(protein_classes4) = c("Protein_class", "number_of_genes", "Percent_of_gene_hit_against_total_number_of_genes", "Percent_of_gene_hit_against_total_number_of_Protein_Class_hits")
protein_classes4 = protein_classes4[order(protein_classes4$number_of_genes, decreasing = T),] 
protein_classes4$stage = "Disease genes with at\nleast 1 disease paralogue"

###Total_plot with protein_classes, protein_classes2, protein_classes3
total_protein_class = rbind(protein_classes, protein_classes2, protein_classes3)
total_protein_class[total_protein_class$Protein_class=="transmembrane receptor regulatory/adaptor protein (PC00226)", "Protein_class"] = "transmembrane receptor\nregulatory/adaptor protein (PC00226)"
total_protein_class$stage = factor(total_protein_class$stage, levels = c("Only genes with\nParalogue Annotations", "Only genes with\nParalogues", "All genes with P/LP\nmissense Clinvar variants"))

gg_protein_class = ggplot(total_protein_class, aes(y=number_of_genes, x=reorder(Protein_class, number_of_genes), fill = stage)) +
  geom_bar(stat="identity", position = "dodge") +
  coord_flip() +
  # geom_point() + 
  # geom_text(aes(label=Protein_class, position = "dodge"), position = position_dodge(width = 0), hjust =0 , vjust=0, angle = 20) +
  theme(
  #   axis.text.x = element_text(angle = -45, hjust = -0),
  #   plot.margin = unit(c(5.5, 5.5, 5.5, 10), "points")
    # axis.title.y=element_blank(),
    legend.title=element_blank()
    # legend.position="none"
    # axis.text.x=element_blank()
    # axis.ticks.x=element_blank()
    ) +
  scale_fill_manual(values=c( "#BB8FCE", "#85C1E9","#76D7C4")) + 
  guides(fill = guide_legend(reverse=TRUE)) +
  scale_y_continuous(breaks = c(0,50,100,150,200,250,300,350,400)) +
  xlab("PANTHER Protein Class") + theme(legend.key.size = unit(2,"line")) +ylab("Number of Genes")
ggsave("images/GO_protein_classes_of_genes.png", height = 8, width = 12)

total_protein_class_filtered = total_protein_class[total_protein_class$Protein_class != "viral protein (PC00237)" &
                                                     total_protein_class$Protein_class != "surfactant (PC00212)" &
                                                     total_protein_class$Protein_class != "transmembrane receptor\nregulatory/adaptor protein (PC00226)" &
                                                     total_protein_class$Protein_class != "chaperone (PC00072)",]


gg_protein_class2 = ggplot(total_protein_class_filtered, aes(y=number_of_genes, x=reorder(Protein_class, number_of_genes), fill = stage)) +
  geom_bar(stat="identity", position = "dodge") +
  coord_flip() +
  theme(
    legend.title=element_blank()
    ) +
  scale_fill_manual(values=c( "#BB8FCE", "#85C1E9","#76D7C4")) + 
  guides(fill = guide_legend(reverse=TRUE)) +
  scale_y_continuous(breaks = c(0,50,100,150,200,250,300,350,400)) +
  xlab("PANTHER Protein Class") + theme(legend.key.size = unit(2,"line")) +ylab("Number of Genes")
ggsave("images/GO_protein_classes_of_genes_filtered.png", height = 8, width = 12)

###Total_plot with protein_classes, protein_classes4
total_protein_class = rbind(protein_classes, protein_classes4)
total_protein_class[total_protein_class$Protein_class=="transmembrane receptor regulatory/adaptor protein (PC00226)", "Protein_class"] = "transmembrane receptor\nregulatory/adaptor protein (PC00226)"
total_protein_class$stage = factor(total_protein_class$stage, levels = c("Only genes with\nParalogue Annotations", "Disease genes with at\nleast 1 disease paralogue"))

gg_protein_class = ggplot(total_protein_class, aes(y=number_of_genes, x=reorder(Protein_class, number_of_genes), fill = stage)) +
  geom_bar(stat="identity", position = "dodge") +
  coord_flip() +
  # geom_point() + 
  # geom_text(aes(label=Protein_class, position = "dodge"), position = position_dodge(width = 0), hjust =0 , vjust=0, angle = 20) +
  theme(
  #   axis.text.x = element_text(angle = -45, hjust = -0),
  #   plot.margin = unit(c(5.5, 5.5, 5.5, 10), "points")
    # axis.title.y=element_blank(),
    legend.title=element_blank()
    # legend.position="none"
    # axis.text.x=element_blank()
    # axis.ticks.x=element_blank()
    ) +
  scale_fill_manual(values= c(alpha("red",0.9), alpha("blue",0.4)), labels = c("Only genes with Paralogue\nAnnotations (n=535)","Disease genes with disease\nparalogues (n=1648)")) + 
  guides(fill = guide_legend(reverse=TRUE)) +
  scale_y_continuous(breaks = c(0,50,100,150,200), limits = c(0,205), expand = c(0,0), position = "right") +
  xlab("PANTHER Protein Class") + theme(legend.key.size = unit(3,"line")) +ylab("Number of Genes") +
  theme(axis.text.x = element_text(size = 16), 
        axis.text.y = element_text(size=11),
        legend.text = element_text(size=16),
        axis.title = element_text(size=15),
        legend.position = c(0.84,0.07)) + 
  guides(fill=guide_legend(
                 keywidth=0.2,
                 keyheight=0.5,
                 default.unit="inch",
                 reverse = TRUE))
ggsave("images/GO_protein_classes_of_genes2.png", height = 8, width = 12)

total_protein_class_poster = total_protein_class[total_protein_class$Protein_class != "transmembrane receptor\nregulatory/adaptor protein (PC00226)",]
gg_protein_class_poster = ggplot(total_protein_class_poster, aes(y=number_of_genes, x=reorder(Protein_class, number_of_genes), fill = stage)) +
  geom_bar(stat="identity", position = "dodge") +
  coord_flip() +
  # geom_point() + 
  # geom_text(aes(label=Protein_class, position = "dodge"), position = position_dodge(width = 0), hjust =0 , vjust=0, angle = 20) +
  theme(
  #   axis.text.x = element_text(angle = -45, hjust = -0),
  #   plot.margin = unit(c(5.5, 5.5, 5.5, 10), "points")
    # axis.title.y=element_blank(),
    legend.title=element_blank()
    # legend.position="none"
    # axis.text.x=element_blank()
    # axis.ticks.x=element_blank()
    ) +
  scale_fill_manual(values= c(alpha("red",0.9), alpha("blue",0.4)), labels = c("Genes with Paralogue\nAnnotations\n(hits=535)","Disease genes with\ndisease paralogues\n(hits=1648)")) + 
  guides(fill = guide_legend(reverse=TRUE)) +
  scale_y_continuous(breaks = c(0,50,100,150,200), limits = c(0,210), expand = c(0,0), position = "right") +
  xlab("PANTHER Protein Class") + theme(legend.key.size = unit(3,"line")) +ylab("Number of Genes") +
  theme(axis.text.x = element_text(size = 21), 
        axis.text.y = element_text(size=23),
        legend.text = element_text(size=25),
        axis.title = element_text(size=25),
        legend.position = c(0.7,0.165)) + 
  guides(fill=guide_legend(
                 keywidth=0.2,
                 keyheight=1.2,
                 default.unit="inch",
                 reverse = TRUE))
ggsave("images/GO_protein_classes_of_genes2_for_poster.png", height = 8, width = 12)

#gprofiler
# library(gprofiler2)
# p.paralogous_var_align = Paralogous_var_align(Overlap = 0, "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_paralogs2.noQC", "data/clinvar/clinvar_20190114_GRCh38_onlyPathogenic_and_Likely_pathogenic.out_paraloc_tableized")
# gostres = gost(unique(p.paralogous_var_align$Total_paralog_annotations[!is.na(p.paralogous_var_align$Total_paralog_annotations$ID.y),"Gene"]), user_threshold = 1e-16)
# gostplot(gostres, interactive = TRUE)
```

# References










