---
title: "Paralog Annotation Notes"
# output: rmarkdown::github_document
output: 
  html_document:
    keep_md: true

editor_options: 
  chunk_output_type: inline
bibliography: bibliography.bib
always_allow_html: yes
---

<!--Load Packages and function-->
```{r echo=FALSE, message=FALSE, warning=FALSE, include=FALSE, results='hide'}
source("src/Paralogous_var_align_functions.R")
```

### Aims
* Apply Paralogue annotation on other datasets
    + "Genome Wide" - Clinvar dataset and ALL possible exome variants (get from `/data/Mirror/ExAC_release/release0.3.1/manuscript_data/all_possible_variants`)
    + Cardiomyopathy genes - [MYH7, MYBPC3, TNNT2, TPM1, MYL2, MYL3, TNNI3, ACTC1] -CHECK CODE TO SEE IF ALL MISSENSE CM VARIANTS REALLY DO NOT APPEAR IN ANY OF THESE GENES AS EG. MYH6 SHOULD HAVE SOME
    + Channelopathy genes - [KCNQ1, KCNH2, SCN5A, KCNE1, KCNE2, RYR2]
* Improve precision via increasing conservation of ref/alt alleles
    + pairwise QC - ignore any individual pairwise alignments where ref alleles are not conserved
    + pairwise QC and family QC - ignore entire alignment columns if the entire family ref alleles are not conserved; NB analogous to para z scores
    + pairwise QC and family QC and alt allele QC - ignore entire alignment columns if family ref allele and alt allele isn't conserved
* investigate para z scores
* investigate pfam meta domains

### Some interesting things to look at maybe
* "Ohnologs"
* Perform PA on paralogs from CPG (from @Modos2016) and non-CPG and see difference? 
* Integrating ortholog data to increase confidence calling 

### OBSTACLES TO GET DONE:
* Debugging plugin
    + Have fixed "Can't call methods: start/location_from_column" by implementing `if` loops to catch errors
        - check out RBM20 and its "2" paralogues (MATR3 and MATR3...) that were example of genes which variants were causing these problems.
        - apparently there are other genes that have this same pattern where there are multiple genes with the same gene symbol but different IDs - James believes there exists a list of these genes somewhere (maybe ask Emily?)
        - see if can get a list of consistent variants that cause these specific issues for Erica.
    + "MySQL error has gone away" error seems to now have been fixed with the help from emily
        - issues seem to be due to server timeouts from running long jobs
        - sent me https://www.ensembl.org/Help/Faq?id=567 for help
    + ask emily about MART3...?
* Test if plugin works with build37 coords - would save effort in lift overs
    + if not ask emily in same email
* Make a list and write down overlapping genes that cause an issue like MART3, where only one of the overlapping genes has info reported back. Write it as an appendix.
    + Either report back in output file as special results or maybe TAKE OUT error catching for "Can't call methods: start/location_from_column" errors and see if server still times out. That way warnings will be reported.
* All possible missense vcf for exome
    + be clear between all possible amino acid substitution and all possible nucleotide substitution
        - the sythetic exome from ExAC should contain all possible SNV
    + would be good to also write up stat for how many aa sub there is for every nt sub in report
* Xiaolei's all possible missense cm vcf does not contain all possible snv
* Need to get all possible missense for genome as well, I.e. get a vcf containing all possible nt sub, not just all possible aa sub! 
    + NEED TO MAKE MY OWN
* Rebenchmark with bigger test set (up to 100,000?)
    + edit the benchmarking script and rerun
* Get setup on imperial hpc and make sure plugin works
* ~~Data from denis~~ 
    + DONE - denis says that data I have is most up to date
    + Convert Para Z scores to 1 file for faster lookup and addition to tableize data.
* Data from henrike
    + waiting, will send me data when ready
* Web tool
    + Keep design simple
    + Ask Mark
* Make an experiment plan for transfering the framework over to structure space from sequence space

### Manuscript Plan
* New tool to show (more likely Erica will write up)
* Contrast to previous studies, is genome wide validated
* Describe implementation and how to use
    + vep plugin lib
* Provide additional descriptive statistics of input (clinvar) data, e.g. number of genes with paralogues, number of disease genes etc.
* Paralogue annotate P/LP with P/LP; paralogue annotate B/LB with P/LP
    + generate confusion matrices for above
* can PA also predict benign variants as well as pathogenic?
* does paraZ score add additional benefit
* additional test/validation dataset
    + disease
    + ExAC
* ICC genes - EFs 
* Distributability 
    + plugin
    + R shiny - vep web tool; integrated browser
    + integrated into gnomad
* Pfam domains

### Introduction
With the advancements of sequencing technology, new potential variants are being discovered constantly. However to be able to identify said variants as pathogenic or benign requires supporting evidence, which does not always exists especially if the variant novel. 
Previously Ware *et al.* have developed **Paralogue Annotation** [@Ware2012; @Walsh2014], which utilizes information from paralogues (evolutionarily related genes from the same species) to help classify pathogenic variants. They verified its use in LQTS genes on variants acquired from patient cohorts.

Here **Paralogue Annotation** is tested further on a (Likely) Pathogenic/Benign varaint dataset from Clinvar.

<span style="color:red">Also have a look at @Barshir2018 for more info about paralogs in diseases.</span>


### Material and Methods Notes

The Paralog Annotation algorithm was wriiten by Erica as a perl script plugin (called __ParalogueAnno_plugin_cleanup.pm__) for Ensembl's VEP version 90 (https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html). 

The plugin has two arguments:

* the first parameter has 2 options:
    + ```variant``` (default) returns only the paralogous variants if any are present in the associated paralogs of the query gene found in the ensembl compara database
    + ```paraloc``` returns only paralog variant locations in the form of genomic coordinates of the corresponding codon in ALL paralogs;
* the second parameter has 2 options:
    + ```all``` for all variants;
    +  ```damaging``` (default) for only damaging variant.
The majority of the time ```paraloc``` mode is used.

The Ensembl team have touched up Erica's plugin and decrease runtime. The plugin is now called __ParalogueAnnotation.pm__

The initial output by VEP and the Plugin (VEP+Plugin) is not reader friendly for either the user nor if you want to parse informations. So a python wrapper, shwon below, for the VEP+Plugin was written to automatically parse the results, namely the paralogous variant information - __/data/Share/nick/Paralog_Anno/VEP_ParalogAnno.py__(Note the code below is not polished for release and is a WIP).
```{python eval = FALSE, include = FALSE}
import  os, sys, subprocess, re
import shlex, subprocess

#WRAPPER FOR VEP+PLUGIN

def VEP_Plugin_run(flavour, input_dir, input_file, genome_build, VEPversion):
	# flavour = int(sys.argv[1])	#0(no plugin), 1(variants), 2(paraloc)
	# input_dir = sys.argv[2]	#input directory
	# input_file = sys.argv[3]	#name of input file
	# # output_file = sys.argv[3]	#name of output file
	# genome_build = str(sys.argv[4]) #37, 38
	# VEPversion = int(sys.argv[5]) #83, 87, 90

	#e.g. python3 VEP_ParalogAnno.py 0 /data/Share/nick/Paralog_Anno/data_files 10_pathogenic.vcf 38 90

	# if not len(os.listdir("/data/Share/nick/Paralog_Anno/homo_sapiens/")) == 0:
	# 	sys.exit("Terminating script:/data/Share/nick/Paralog_Anno/homo_sapiens/ not empty. Please empty and try again")

	# print(type(flavour))
	if flavour == 0:
		# print("check")
		flavour = ""
		output_file = input_file.rsplit(".",1)[0]+".out_no_plugin"
		print(output_file)
	# elif flavour == 1:
	# 	flavour = "--plugin ParalogueAnno_plugin_cleanup"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_old"
	# 	print(output_file)
	# elif flavour == 2:
	# 	flavour = "--plugin ParalogueAnno_plugin_cleanup,paraloc"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_old_paraloc"
	# 	print(output_file)
	# elif flavour == 3:
	# 	flavour = "--plugin ParalogueAnnotation"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_new"
	# 	print(output_file)
	# elif flavour == 4:
	# 	flavour = "--plugin ParalogueAnnotation,paraloc"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_new_paraloc"
	# 	print(output_file)
	elif flavour == 1:
		flavour = "--plugin ParalogueAnnotation"
		output_file = input_file.rsplit(".",1)[0]+".out"
		print(output_file)
	elif flavour == 2:
		flavour = "--plugin ParalogueAnnotation,paraloc"
		output_file = input_file.rsplit(".",1)[0]+".out_paraloc"
		print(output_file)

	# print(input_file)

	if not input_dir.endswith("/"):
		input_dir = input_dir+"/"
	print(input_dir)


	'''if VEPversion == 83:
		#VEP version 83
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/83_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/83_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			"perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/variant_effect_predictor.pl --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/83_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)
	elif VEPversion == 87:
		#VEP version 87
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/87_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/87_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			"perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-tools-release-87/ensembl-tools/scripts/variant_effect_predictor/variant_effect_predictor.pl --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/87_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)'''
	if VEPversion == 90:
		#VEP version 90
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/90_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/90_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			# "perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-vep/vep --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			"perl -I /data/Share/nick/Paralog_Anno /data/Install/ensembl-vep/vep --force_overwrite --vcf --allele_number --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/90_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)
	
	# print(mv_command)
	# os.system(mv_command)
	print(command)
	os.system(command)
	# print(mv_back_command)
	# os.system(mv_back_command)
	
	with open(
		input_dir + output_file + 
		"_summary.html"
		) as infile:
		for line in infile:
			if "<td>Run time</td>" in line:
				if VEPversion == 83 or VEPversion == 87:
					#IF VEP87 AND 83
					m = re.search(r"<td>Run time<\/td> <td>.*second.<\/td>", line)
					# print(m.group())
					run_time = m.group().split("</td> <td>")[1].rstrip("</td>")
					print(run_time, "\n")
				elif VEPversion == 90:
					# #IF VEP90
					m = re.search(r"<td>Run time<\/td><td>.*second.<\/td>", line)
					# print(m.group())
					run_time = m.group().split("</td><td>")[1].rstrip("</td>")
					print(run_time, "\n")

	###TESTED THE BELOW WITH VEP90#### MIGHT NOT WORK WITH OTHER VERSIONS
	#reads through outfile of vep+plugin and tidies up data by extracting only variants that have paralogs and spits that out as another outfile
	out_file = open(input_dir + output_file + "_paralogs", "w")
	if not flavour == "":
		with open(
			input_dir + output_file
			) as infile:
			for line in infile:
				if not line.startswith("#"):
					line = line.split("CSQ=")
					CSQ = line[0]
					print(CSQ)	#print query variant
					line = line[1].split(",")
					# print(line, "\n")
					for x in line:
						y = x.split("|")
						# print(y, "\n")
						gene = y[3]
						paralog = y[23]
						if paralog == "" or paralog == "\n":
							paralog_check = 0
						else:
							paralog_check = 1
						if paralog_check == 1:
							print(x, "\n")
							out_file.write(
								str(CSQ)+"PARALOGS->"
								+str(x)+"\n"
								)
				else:
					out_file.write(line)
	out_file.close()
	
	# ###FOR CLINVAR VARIANTS AT LEAST###SCRAP ALL OF THIS, USE R TIDYVERSE JOIN METHOD INSTEAD
	# out_file = open(input_dir + output_file + "_aligned_vars", "w")
	# with open(input_dir + output_file + "_paralogs") as infile1:
	# 	infile2 = open(input_dir + input_file, "r") 
	# 	text_list = infile2.read()
	# 	infile2.close()
	# 	####TEST_BELOW####
	# 	# chromosome = 1
	# 	# pos1 = 1050574
	# 	# pos2 = 1050575
	# 	# pos3 = 1050576
	# 	# s1 = re.findall("^"+str(chromosome)+"\t"+str(pos1)+"\s.*$", text_list, re.MULTILINE)
	# 	# s2 = re.findall("^"+str(chromosome)+"\t"+str(pos2)+"\s.*$", text_list, re.MULTILINE)
	# 	# s3 = re.findall("^"+str(chromosome)+"\t"+str(pos3)+"\s.*$", text_list, re.MULTILINE)		
	# 	# if s1:
	# 	# 	print("YES")
	# 	# 	print(s1)
	# 	# if s2:
	# 	# 	print(s2)
	# 	# if s3:
	# 	# 	print(s3)
	# 	####END_OF_TEST####
	# 	for line in infile1:
	# 		if not line.strip() == "":
	# 			query_variant = line.split("PARALOGS->")[0]
	# 			out_file.write(query_variant)
	# 			paralog_line = line.split("PARALOGS->")[1].rstrip()
	# 			paralog_line = paralog_line.split("|&")[1]
	# 			paralog_line = paralog_line.split("&")
	# 			paralog_line = list(filter(None, paralog_line))
	# 			print(paralog_line)
	# 			out_file.write(str(paralog_line)+";")
	# 			for paralog in paralog_line:
	# 				if not paralog == "":
	# 					print(paralog)
	# 					location = paralog.split(":")[1]
	# 					chromosome = location.split("_")[0].lstrip("chr")
	# 					position = location.split("_")[1]
	# 					position1 = int(position.split("-")[0])
	# 					position2 = int(position.split("-")[1])
	# 					positions = list(range(position1,position2+1))
	# 					print(chromosome, type(positions), positions)
	# 					# p = re.compile(r"^"+str(chromosome)+r"\t"+positions+r"\t")
	# 					for pos in positions:
	# 						s = re.findall("^"+str(chromosome)+"\t"+str(pos)+"\s.*$",text_list, re.MULTILINE)
	# 						if s:
	# 							print(s)
	# 							out_file.write(str(s)+";")
	# 			out_file.write("\n")
	# out_file.close()
	# #################################################
	
# VEP_Plugin_run(2, "/data/Share/nick/Paralog_Anno/data_files", "new_extracted_clinvar_cm_intepretated_raw_variants_2.vcf", "38", 87)


flavour = int(sys.argv[1])	#0(no plugin), 1(variants), 2(paraloc),
input_dir = sys.argv[2]	#input directory
input_file = sys.argv[3]	#name of input file
genome_build = str(sys.argv[4]) #37, 38
VEPversion = int(sys.argv[5]) #83, 87, 90
VEP_Plugin_run(flavour, input_dir, input_file, genome_build, VEPversion)
```

An intermediate python script (__File_prep_for_R.py__) was used to prep the results into R friendly data. Furthermore it could also be used to perform pairwise and family QC. Incidentally, the pairwise QC could be performed directly in R after the raw results are processed by __tableize_vcf.py__ and tabulated (see below).

__/data/Share/nick/Paralog_Anno/File_prep_for_R.py__ - formats results from __VEP_ParalogAnno.py__ into tabulated format ready for R processing
```{python eval = FALSE, include = FALSE}
import  os, sys, subprocess, re

#SCRIPT FOR SEPARATING EACH LINE IN VCF OUTPUT FROM VEP_PARALOGANNO.PY INTO TAB SEP FILE READY FOR JOIN IN R
#don't use if VEP_ParalogAnno.py has been edited to already do this, check to see if same code exists in that script
###BELOW ONLY WORKS FOR CLINVAR VCF FILES (EXAC HAS NO INFO IN ID VARIABLE)

input_file = sys.argv[1]	#path of VEP_ParalogAnno.py paralogs output file
dir1 = input_file.rsplit("/", 1)[0]
input_file2 = sys.argv[2]	#path of vcf file containing only orginal query variants list used as input for VEP_ParalogAnno.py. I.e. the very first input file!
refid_flavour = sys.argv[3] #whether to consider refID or not. 0: not, 1: only consider paralogous pair alignments where refID = 1, 2: only consider total gene family paralogous alignments where all refID in that family = 1

if refid_flavour == "noQC":
	out_file = open(input_file+"2.noQC", "w")
elif refid_flavour == "para_con":
	out_file = open(input_file+"2.para_con", "w")
elif refid_flavour == "all_con":
	out_file = open(input_file+"2.all_con", "w")

max_split = 0	#check to see max number of columns for R import as wide format
max_paralog_codons = ""
with open(input_file, "r") as f:
	for line in f:
		if line != "\n":
			# print(line)
			chrom = line.split()[0]
			position = line.split()[1]
			ID = line.split()[2]
			print(chrom+" "+position+"\t"+str(ID)+"\t")

			out_file.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\t")
			
			line = line.rstrip()
			paralog_codons = line.split("PARALOGS->")[1]
			paralog_codons = paralog_codons.split("|&")[1]
			paralog_codons = paralog_codons.split("&")
			# paralog_codons.pop(0)
			paralog_codons.remove("")
			print(paralog_codons)
			refid_flav2_check = 0
			if refid_flavour == "all_con":
				if all(refids.endswith("REFID=1") for refids in paralog_codons) == True:
					refid_flav2_check = 1

			for codon in paralog_codons:
				print(codon)

				if refid_flavour == "noQC" or refid_flav2_check == 1:
					refID = codon.split(":")[-1]
					print(refID)
					codon = codon.split(":")[1]
					paralog_chrom = codon.split("_")[0].lstrip("chr")
					paralog_position = int(codon.split("-")[1])
					print(
						str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position)+"\t"
						)

					out_file.write(
						str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position)+"\t"
						)

				if refid_flavour == "para_con":
					refID = codon.split(":")[-1]
					if refID == "REFID=1":
						print(refID)
						codon = codon.split(":")[1]
						paralog_chrom = codon.split("_")[0].lstrip("chr")
						paralog_position = int(codon.split("-")[1])
						print(
							str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position)+"\t"
							)

						out_file.write(
							str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position)+"\t"
							)

				# if refid_flavour == 2:
				# 	refID = 

			out_file.write("\n")

			if len(paralog_codons)*3 + 2 > max_split:
				max_split = len(paralog_codons)*3 + 2
				# print(len(paralog_codons))
				# print(paralog_codons)
				max_paralog_codons = paralog_codons
			# newline[1] = "\t".join(newline[1])
			# newline = "\t".join(newline)
			# out_file.write(newline)
	print(max_split, max_paralog_codons)	#number of columns for dataframe in R; + 2 for (chrom poistion) and (ID)
out_file.close()

if input_file2 != "null":
	if input_file2.endswith(".vcf"):
		out_file2 = open(input_file2+"_onlyVariantslist", "w")
		with open(input_file2, "r") as f:
			for line in f:
				if not line.startswith("#"):
					chrom = line.split()[0]
					position = line.split()[1]
					ID = line.split()[2]
					out_file2.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\n")
	# elif input_file2.endswith("out_paraloc"):
	# 	out_file2 = open(input_file2+"_onlyVariantslist", "w")
	# 	with open(input_file2, "r") as f:
	# 		for line in f:
	# 			if not line.startswith("#"):
	# 				chrom = line.split()[0]
	# 				position = line.split()[1]
	# 				ID = line.split()[2]
	# 				amino_acids = line.split("CSQ=")[1]

	# 				out_file2.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\n")
	out_file2.close()
```

As paraloc mode only returns ref alleles. The alt alleles were extracted from the VEP information. This was done by using __tableize_vcf.py__.
__/data/Share/nick/Paralog_Anno/loftee/src/tableize_vcf.py__ was used to format the VEP output into table format for R processing. For example:
```{bash eval = FALSE, include = TRUE}
python /data/Share/nick/Paralog_Anno/loftee/src/tableize_vcf.py --vcf /data/Share/nick/Paralog_Anno/data_files/clinvar_20171029_onlyPathogenic.out_paraloc --out /data/Share/nick/Paralog_Anno/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized --do_not_minrep --include_id --vep_info SYMBOL,Amino_acids,Codons,Paralogue_Vars --split_by_transcript --canonical_only
```

If __--split_by_transcript__ is used then the code above is sufficient. Otherwise a python wrapper that includes additional formatting (__/data/Share/nick/Paralog_Anno/Tableize_wrapper.py__) that tableize couldn't do, i.e. separate variants that had multiple REF and ALT alleles was used to prepare the data for R. 
```{python eval = FALSE, include = FALSE}
import  os, sys, subprocess, re

#Tableize.py wrapper plus additional formatting that can't be done in tableize

input_file = sys.argv[1]	#path of paraloc file e.g. /data/Share/nick/Paralog_Anno/data_files/clinvar_20171029_onlyPathogenic.out_paraloc
dir1 = input_file.rsplit("/", 1)[0]

os.system("python /data/Share/nick/Paralog_Anno/loftee/src/tableize_vcf.py --vcf " + input_file + " --out " + input_file + "_tableized --do_not_minrep --include_id --vep_info SYMBOL,Amino_acids,Codons,Paralogue_Vars")

out_file = open(input_file + "_tableized_temp", "w")
with open(input_file + "_tableized") as f:
	for line in f:
		line = line.split()
		AAs = line[7].split(",")
		for column in AAs:
			out_file.write(line[0] + "\t" + line[1] +"\t" +line[2] +"\t" + line[3] +"\t" +line[4] +"\t" +line[5] +"\t" + line[6] + "\t" + column + "\t" + line[8] + "\t" + line[9] + "\n")
out_file.close()

os.system("mv " + input_file + "_tableized_temp " + input_file + "_tableized")
```

#### Datasets
Clinvar Likely Pathogenic/Pathogenic and Likely Benign/Benign variant vcf files were extracted and downloaded via the method developed by Zhang *et al.* [@Zhang2017]. 

#### Benchmarking performance of the plugin

__/data/Share/nick/Paralog_Anno/multi_vcf_extractor_benchmark.py__ is used to demonstrate speed at which VEP+Plugin takes to run
```{python eval = FALSE, include = FALSE}
#Script for benchmarking VEP with different flavours by auto extracting set number of variants and running them

import os, sys, subprocess, re
sys.path.insert(0, "/data/Install/Python-3.6.3/")
from VEP_ParalogAnno import *
import numpy as np
import pandas as pd
from scipy import stats, integrate
import matplotlib.pyplot as plt
import seaborn as sns

file = sys.argv[1]	#e.g. data_files/ExAC.r1.sites.vep_onlySCN5A_GRCh38coords_wRefAl.vcf
dir1 = file.split("/")
onlyfile = dir1.pop()
dir1 = "/".join(dir1)
if not dir1.endswith("/"):
	dir1 = dir1+"/"

flavour = int(sys.argv[2]) #e.g 0
VEPversion = int(sys.argv[3]) #e.g. 90

x_list = list(range(100,1100,100))
x_list.append(10)
x_list.sort()
y_list = []

for x_var in x_list:
	input_file = str(onlyfile.rstrip(".vcf"))+".first"+str(x_var)+"lines.vcf"
	os.system("head -n"+str(x_var)+" "+str(dir1)+str(onlyfile)+" > "+str(dir1)+str(input_file))

	VEP_Plugin_run(
		flavour, 
		"/data/Share/nick/Paralog_Anno/data_files", 
		input_file,
		"38",
		VEPversion)

	if flavour == 0:
		output_file = input_file.rsplit(".",1)[0]+".out_no_plugin"
		print(output_file)
	elif flavour == 1:
		output_file = input_file.rsplit(".",1)[0]+".out"
		print(output_file)
	elif flavour == 2:
		output_file = input_file.rsplit(".",1)[0]+".out_paraloc"
		print(output_file)

	with open(
		dir1 + output_file + 
		"_summary.html"
		) as infile:
		for line in infile:
			if "<td>Run time</td>" in line:
				if VEPversion == 83 or VEPversion == 87:
					#IF VEP87 AND 83
					m = re.search(r"<td>Run time<\/td> <td>.*second.<\/td>", line)
					run_time = m.group().split("</td> <td>")[1].rstrip("</td>")
					print(run_time, "\n")
				elif VEPversion == 90:
					# #IF VEP90
					m = re.search(r"<td>Run time<\/td><td>.*second.<\/td>", line)
					run_time = m.group().split("</td><td>")[1].rstrip("</td>")
					print(run_time, "\n")
				y_list.append(int(run_time.split()[0]))

x_list = [0] + x_list
y_list = [0] + y_list
print(x_list)
print(y_list)


fig, ax = plt.subplots(figsize=(15,10))
axes = plt.gca()
plt.plot(x_list, y_list, 'ro')
plt.xlabel("Number of variants in query", fontsize=15)
plt.ylabel("Time taken (seconds)", fontsize=20)
if flavour == 0:
	plt.title("Time taken for VEP (no plugin) to run against number of varaints being searched")
elif flavour == 1:
	plt.title("Time taken for VEP (plugin in '-varaints' mode) to run against number of varaints being searched")
elif flavour == 2:
	plt.title("Time taken for VEP (plugin in '-paraloc' mode) to run against number of varaints being searched")
# plt.show()
plt.savefig(dir1+"benchmark_"+str(flavour)+"_"+str(VEPversion)+".png")
plt.gcf().clear()
```

#### Scripts pipeline
<!-- old
vcf input file -> VEP_ParalogAnno.py -> File_prep_for_R.py -> Paralogous_var_align.R
vcf input file -> VEP_ParalogAnno.py -> paraloc file -> tableize_vcf.py (Tableize_wrapper.py) -> Paralogous_var_align.R
-->
```{r}
library(DiagrammeR)
pipeline = DiagrammeR::grViz("
digraph boxes_and_circles {
graph [overlap = true, fontsize = 10]

node [shape = plaintext, fillcolor = green, style=filled, fixedsize=false]
'VEP_ParalogAnno.py'; 'File_prep_for_R.py'; 'Tableize_wrapper.py'; 'R markdown'

node [shape = plaintext, fillcolor = orange, style=filled, fixedsize=false]
'vcf input file'; 'paralogs file'; 'paraloc file'; 'paralogs2 file'; 'paraloc_tableized file'

'vcf input file' -> 'VEP_ParalogAnno.py'; 'VEP_ParalogAnno.py' -> 'paralogs file'; 'VEP_ParalogAnno.py' -> 'paraloc file'; 'paralogs file' -> 'File_prep_for_R.py'; 'paraloc file' -> 'Tableize_wrapper.py'; 'File_prep_for_R.py' -> 'paralogs2 file'; 'Tableize_wrapper.py' -> 'paraloc_tableized file'; 'paralogs2 file' -> 'R markdown'; 'paraloc_tableized file' -> 'R markdown'

}")
pipeline
```

#### Statistical terms
In context of is there a pathogenic paralogue alignment? A TP = pathogenic query variant with a paralogous pathogenic hit; FP = benign query variant with a paralogous pathogenic hit; FN = pathogenic query variant with no paralogous pathogenic hit; and TN= benign query variant with no paralogous pathogenic hit.

Likewise for a benign paralogous alignment, a TP = benign query variant with a paralogous benign hit; FP = pathogenic query variant with a paralogous benign hit; FN = benign query variant with no paralogous benign hit; and TN = pathogenic query variant with no paralogous benign hit.

#### Annotation of Clinvar
The Clinvar file __clinvar_20171029.vcf__ was downloaded from ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/. Note that since the initial look at what was available there's been updated Clinvar files. 

NOTE that I have noticed some descrepencies between the plugin annotations which call REFID = 1/0 and that of comparing the REF amino acid by VEP in the dataset to itself. This is due to the fact that the paralogous variant VEP is referring to is simply not in the dataset that I am annotating back to. As a result, it is best to make sure that the ref alleles are indeed the same when processing in R.

The annotataion of the entire clinvar set as of 2018:
```{r eval = FALSE, include = FALSE}
#noQC
#pathogenic to pathogenic
p.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
# p.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized")
#benign to pathogenic
b.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#stats
conf = conf_matrix(p.paralogous_var_align$num_of_paralog_anno, p.paralogous_var_align$paralog_data, b.paralogous_var_align$num_of_paralog_anno, b.paralogous_var_align$paralog_data)

#para_con
p2.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b2.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.paralogous_var_align$num_of_paralog_anno, p2.paralogous_var_align$paralog_data, b2.paralogous_var_align$num_of_paralog_anno, b2.paralogous_var_align$paralog_data)

#all_con
p3.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b3.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.paralogous_var_align$num_of_paralog_anno, p3.paralogous_var_align$paralog_data, b3.paralogous_var_align$num_of_paralog_anno, b3.paralogous_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.paralogous_var_align$Total_paralog_annotations[p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# p4.Total_paralog_annotations = p4.Total_paralog_annotations[!duplicated(p4.Total_paralog_annotations$ID.x),]
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))

#benign to pathogenic
b4.Total_paralog_annotations = b3.paralogous_var_align$Total_paralog_annotations[b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.paralogous_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# b4.Total_paralog_annotations = b4.Total_paralog_annotations[!duplicated(b4.Total_paralog_annotations$ID.x),]
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))

conf4 = conf_matrix(p4.num_of_paralog_anno, p3.paralogous_var_align$paralog_data, b4.num_of_paralog_anno, b3.paralogous_var_align$paralog_data)
```

<!--Previous old code below, remove once above code works-->
The annotataion of the entire clinvar set as of 20171029:
```{r eval = FALSE, include = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)
#noQC
#pathogenic to pathogenic
# p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC")
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
# p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
# p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data = p.tableized_data[!is.na(p.tableized_data$Amino_acids),]

p.tableized_data$Variant_pos = paste(p.tableized_data$CHROM,p.tableized_data$POS, sep = " ")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],unlist)
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],function(x) x[1])
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],unlist)
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# p.tableized_data = p.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
p.ref_data = p.tableized_data
p.paralog_data = left_join(p.paralog_data, p.tableized_data, by = c("Variant_pos", "ID", "REF", "ALT", "Gene" = "SYMBOL"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:max_no_col, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
# b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC")
b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(b.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
# b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
# b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data = b.tableized_data[!is.na(b.tableized_data$Amino_acids),]

b.tableized_data$Variant_pos = paste(b.tableized_data$CHROM,b.tableized_data$POS, sep = " ")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],unlist)
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],function(x) x[1])
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],unlist)
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# b.tableized_data = b.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
b.ref_data = b.tableized_data
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos", "ID", "REF", "ALT", "Gene" = "SYMBOL"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:max_no_col, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table_TP = ptop.num_of_paralog_anno
con_table_FP = btop.num_of_paralog_anno
con_table_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table_PPV = con_table_TP/(con_table_TP+con_table_FP)
con_table_Sensitivty = con_table_TP/(con_table_TP+con_table_FN)
con_table = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table) = c("Pathogenic", "Benign")
rownames(con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table_p_value = fisher.test(con_table)

#para_con
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con")
max_no_col = (max(count.fields(b.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table2_TP = ptop.num_of_paralog_anno
con_table2_FP = btop.num_of_paralog_anno
con_table2_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table2_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table2_PPV = con_table2_TP/(con_table2_TP+con_table2_FP)
con_table2_Sensitivty = con_table2_TP/(con_table2_TP+con_table2_FN)
con_table2 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table2) = c("Pathogenic", "Benign")
rownames(con_table2) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table2_p_value = fisher.test(con_table2)
#Variants_removed_after_PA_QC1
var_rem_para_con_TP = con_table[2]-con_table2[2]
var_rem_para_con_FP = con_table[4]-con_table2[4]
var_rem_para_con_FN = length(p.paralog_data$ID)-var_rem_para_con_TP
var_rem_para_con_PPV = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FP)
var_rem_para_con_Sensitivity = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FN)
var_rem_para_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_para_con_TP,
    length(b.paralog_data$ID),
    var_rem_para_con_FP
  ), ncol = 2
)
colnames(var_rem_para_con) = c("Pathogenic", "Benign")
rownames(var_rem_para_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_para_con_p_value = fisher.test(var_rem_para_con)

#all_con
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.all_con")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.all_con")
max_no_col = (max(count.fields(b.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table3_TP = ptop.num_of_paralog_anno
con_table3_FP = btop.num_of_paralog_anno
con_table3_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table3_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table3_PPV = con_table3_TP/(con_table3_TP+con_table3_FP)
con_table3_Sensitivty = con_table3_TP/(con_table3_TP+con_table3_FN)
con_table3 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table3) = c("Pathogenic", "Benign")
rownames(con_table3) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table3_p_value = fisher.test(con_table3)
#Variants_removed_after_PA_QC2
var_rem_all_con_TP = con_table2[2]-con_table3[2]
var_rem_all_con_FP = con_table2[4]-con_table3[4]
var_rem_all_con_FN = length(p.paralog_data$ID)-var_rem_all_con_TP
var_rem_all_con_PPV = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FP)
var_rem_all_con_Sensitivity = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FN)
var_rem_all_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_all_con_TP,
    length(b.paralog_data$ID),
    var_rem_all_con_FP
  ), ncol = 2
)
colnames(var_rem_all_con) = c("Pathogenic", "Benign")
rownames(var_rem_all_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_all_con_p_value = fisher.test(var_rem_all_con)

#alt_con
#pathogenic to pathogenic
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$ALT_Amino_acids.x==ptop.Total_paralog_annotations$ALT_Amino_acids.y,]
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[!duplicated(ptop.Total_paralog_annotations$ID.x),]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$ALT_Amino_acids.x==btop.Total_paralog_annotations$ALT_Amino_acids.y,]
btop.Total_paralog_annotations = btop.Total_paralog_annotations[!duplicated(btop.Total_paralog_annotations$ID.x),]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table4_TP = ptop.num_of_paralog_anno
con_table4_FP = btop.num_of_paralog_anno
con_table4_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table4_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table4_PPV = con_table4_TP/(con_table4_TP+con_table4_FP)
con_table4_Sensitivty = con_table4_TP/(con_table4_TP+con_table4_FN)
con_table4 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table4) = c("Pathogenic", "Benign")
rownames(con_table4) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table4_p_value = fisher.test(con_table4)
#Variants_removed_after_PA_QC3
var_rem_alt_con_TP = con_table3[2]-con_table4[2]
var_rem_alt_con_FP = con_table3[4]-con_table4[4]
var_rem_alt_con_FN = length(p.paralog_data$ID)-var_rem_alt_con_TP
var_rem_alt_con_PPV = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FP)
var_rem_alt_con_Sensitivity = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FN)
var_rem_alt_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_alt_con_TP,
    length(b.paralog_data$ID),
    var_rem_alt_con_FP
  ), ncol = 2
)
colnames(var_rem_alt_con) = c("Pathogenic", "Benign")
rownames(var_rem_alt_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_alt_con_p_value = fisher.test(var_rem_alt_con)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.paralog_data$ID),length(b.paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(con_table[2], scientific = FALSE),format(con_table[4], scientific = FALSE),con_table2_PPV,con_table_Sensitivty,con_table_p_value$p.value)
Variants_remaining_after_PA_QC1 = c(format(con_table2[2], scientific = FALSE),format(con_table2[4], scientific = FALSE),con_table2_PPV,con_table2_Sensitivty,con_table2_p_value$p.value)
Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

Whole_clinvar_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )

```

Taking only the 8 sarcomeric genes:
```{r eval = FALSE, include = FALSE}
#noQC
p.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf = conf_matrix(p.sarco_var_align$num_of_paralog_anno, p.sarco_var_align$paralog_data, b.sarco_var_align$num_of_paralog_anno, b.sarco_var_align$paralog_data)
#para_con
p2.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b2.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.para_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf2 = conf_matrix(p2.sarco_var_align$num_of_paralog_anno, p2.sarco_var_align$paralog_data, b2.sarco_var_align$num_of_paralog_anno, b2.sarco_var_align$paralog_data)
#all_con
p3.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
b3.sarco_var_align = Subset_var_align(c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1"),"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.all_con", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
conf3 = conf_matrix(p3.sarco_var_align$num_of_paralog_anno, p3.sarco_var_align$paralog_data, b3.sarco_var_align$num_of_paralog_anno, b3.sarco_var_align$paralog_data)

#alt_con
#pathogenic to pathogenic
p4.Total_paralog_annotations = p3.sarco_var_align$Total_paralog_annotations[p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==p3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# p4.Total_paralog_annotations = p4.Total_paralog_annotations[!duplicated(p4.Total_paralog_annotations$ID.x),]
p4.num_of_paralog_anno = sum(!is.na(p4.Total_paralog_annotations$ID.y))

#benign to pathogenic
b4.Total_paralog_annotations = b3.sarco_var_align$Total_paralog_annotations[b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.x==b3.sarco_var_align$Total_paralog_annotations$ALT_Amino_acids.y,]
# b4.Total_paralog_annotations = b4.Total_paralog_annotations[!duplicated(b4.Total_paralog_annotations$ID.x),]
b4.num_of_paralog_anno = sum(!is.na(b4.Total_paralog_annotations$ID.y))

conf4 = conf_matrix(p4.num_of_paralog_anno, p3.sarco_var_align$paralog_data, b4.num_of_paralog_anno, b3.sarco_var_align$paralog_data)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.sarco_var_align$paralog_data$ID),length(b.sarco_var_align$paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(conf$con_table[2], scientific = FALSE),format(conf$con_table[4], scientific = FALSE),conf$PPV,conf$Sensitivty,conf$Pvalue$p.value)
Variants_remaining_after_PA_QC1 = c(format(conf2$con_table[2], scientific = FALSE),format(conf2$con_table[4], scientific = FALSE),conf2$PPV, conf2$Sensitivity, conf2$Pvalue$p.value)

Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

sarcomeric_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )
```
<!-- Old code below -->
```{r eval = FALSE, include = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)
#noQC
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% sarcomeric_gene)
p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data$Variant_pos = paste(p.tableized_data$CHROM,p.tableized_data$POS, sep = " ")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],unlist)
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],function(x) x[1])
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],unlist)
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# p.tableized_data = p.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
p.ref_data = p.tableized_data
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(b.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% sarcomeric_gene)
b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data$Variant_pos = paste(b.tableized_data$CHROM,b.tableized_data$POS, sep = " ")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],unlist)
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],function(x) x[1])
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],unlist)
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# b.tableized_data = b.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
b.ref_data = b.tableized_data
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table_TP = ptop.num_of_paralog_anno
con_table_FP = btop.num_of_paralog_anno
con_table_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table_PPV = con_table_TP/(con_table_TP+con_table_FP)
con_table_Sensitivty = con_table_TP/(con_table_TP+con_table_FN)
con_table = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table) = c("Pathogenic", "Benign")
rownames(con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table_p_value = fisher.test(con_table)

#para_con
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% sarcomeric_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con")
max_no_col = (max(count.fields(b.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% sarcomeric_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID", "REF" = "REF", "ALT" = "ALT"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table2_TP = ptop.num_of_paralog_anno
con_table2_FP = btop.num_of_paralog_anno
con_table2_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table2_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table2_PPV = con_table2_TP/(con_table2_TP+con_table2_FP)
con_table2_Sensitivty = con_table2_TP/(con_table2_TP+con_table2_FN)
con_table2 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table2) = c("Pathogenic", "Benign")
rownames(con_table2) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table2_p_value = fisher.test(con_table2)
#Variants_removed_after_PA_QC1
var_rem_para_con_TP = con_table[2]-con_table2[2]
var_rem_para_con_FP = con_table[4]-con_table2[4]
var_rem_para_con_FN = length(p.paralog_data$ID)-var_rem_para_con_TP
var_rem_para_con_PPV = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FP)
var_rem_para_con_Sensitivity = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FN)
var_rem_para_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_para_con_TP,
    length(b.paralog_data$ID),
    var_rem_para_con_FP
  ), ncol = 2
)
colnames(var_rem_para_con) = c("Pathogenic", "Benign")
rownames(var_rem_para_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_para_con_p_value = fisher.test(var_rem_para_con)

#all_con
#pathogenic to pathogenic
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% sarcomeric_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
sarcomeric_gene = c("MYH7","MYBPC3","TNNT2","TPM1","MYL2","MYL3","TNNI3","ACTC1")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% sarcomeric_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table3_TP = ptop.num_of_paralog_anno
con_table3_FP = btop.num_of_paralog_anno
con_table3_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table3_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table3_PPV = con_table3_TP/(con_table3_TP+con_table3_FP)
con_table3_Sensitivty = con_table3_TP/(con_table3_TP+con_table3_FN)
con_table3 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table3) = c("Pathogenic", "Benign")
rownames(con_table3) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table3_p_value = fisher.test(con_table3)
#Variants_removed_after_PA_QC2
var_rem_all_con_TP = con_table2[2]-con_table3[2]
var_rem_all_con_FP = con_table2[4]-con_table3[4]
var_rem_all_con_FN = length(p.paralog_data$ID)-var_rem_all_con_TP
var_rem_all_con_PPV = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FP)
var_rem_all_con_Sensitivity = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FN)
var_rem_all_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_all_con_TP,
    length(b.paralog_data$ID),
    var_rem_all_con_FP
  ), ncol = 2
)
colnames(var_rem_all_con) = c("Pathogenic", "Benign")
rownames(var_rem_all_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_all_con_p_value = fisher.test(var_rem_all_con)

#alt_con
#pathogenic to pathogenic
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$ALT_Amino_acids.x==ptop.Total_paralog_annotations$ALT_Amino_acids.y,]
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[!duplicated(ptop.Total_paralog_annotations$ID.x),]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$ALT_Amino_acids.x==btop.Total_paralog_annotations$ALT_Amino_acids.y,]
btop.Total_paralog_annotations = btop.Total_paralog_annotations[!duplicated(btop.Total_paralog_annotations$ID.x),]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table4_TP = ptop.num_of_paralog_anno
con_table4_FP = btop.num_of_paralog_anno
con_table4_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table4_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table4_PPV = con_table4_TP/(con_table4_TP+con_table4_FP)
con_table4_Sensitivty = con_table4_TP/(con_table4_TP+con_table4_FN)
con_table4 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table4) = c("Pathogenic", "Benign")
rownames(con_table4) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table4_p_value = fisher.test(con_table4)
#Variants_removed_after_PA_QC3
var_rem_alt_con_TP = con_table3[2]-con_table4[2]
var_rem_alt_con_FP = con_table3[4]-con_table4[4]
var_rem_alt_con_FN = length(p.paralog_data$ID)-var_rem_alt_con_TP
var_rem_alt_con_PPV = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FP)
var_rem_alt_con_Sensitivity = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FN)
var_rem_alt_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_alt_con_TP,
    length(b.paralog_data$ID),
    var_rem_alt_con_FP
  ), ncol = 2
)
colnames(var_rem_alt_con) = c("Pathogenic", "Benign")
rownames(var_rem_alt_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_alt_con_p_value = fisher.test(var_rem_alt_con)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.paralog_data$ID),length(b.paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(con_table[2], scientific = FALSE),format(con_table[4], scientific = FALSE),con_table2_PPV,con_table_Sensitivty,con_table_p_value$p.value)
Variants_remaining_after_PA_QC1 = c(format(con_table2[2], scientific = FALSE),format(con_table2[4], scientific = FALSE),con_table2_PPV,con_table2_Sensitivty,con_table2_p_value$p.value)
Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

sarcomeric_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )

```

Using only the 8 sarcomeric genes and joining to the whole clinvar dataset did not provide many annotations which could suggest either PA does not perform well on sarcomeric genes (paralogues to sarcomeric genes are not involed in disease) or that there is a lack of data. Therefore, it is not yet certain that PA does not work on sarcomeric genes and annotataion of additional sarcomeric data is required. See below.

Taking only the 5 channelopathy genes:
```{r eval = FALSE, include = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)
#noQC
#pathogenic to pathogenic
p.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(p.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "Ref", "Alt", and "\n"
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data$Variant_pos = paste(p.tableized_data$CHROM,p.tableized_data$POS, sep = " ")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],unlist)
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],function(x) x[1])
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],unlist)
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# p.tableized_data = p.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
p.ref_data = p.tableized_data
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.noQC", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data$Variant_pos = paste(b.tableized_data$CHROM,b.tableized_data$POS, sep = " ")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],unlist)
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],function(x) x[1])
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],unlist)
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
# b.tableized_data = b.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]
b.ref_data = b.tableized_data
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table_TP = ptop.num_of_paralog_anno
con_table_FP = btop.num_of_paralog_anno
con_table_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table_PPV = con_table_TP/(con_table_TP+con_table_FP)
con_table_Sensitivty = con_table_TP/(con_table_TP+con_table_FN)
con_table = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table) = c("Pathogenic", "Benign")
rownames(con_table) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table_p_value = fisher.test(con_table)

#para_con
#pathogenic to pathogenic
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.para_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table2_TP = ptop.num_of_paralog_anno
con_table2_FP = btop.num_of_paralog_anno
con_table2_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table2_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table2_PPV = con_table2_TP/(con_table2_TP+con_table2_FP)
con_table2_Sensitivty = con_table2_TP/(con_table2_TP+con_table2_FN)
con_table2 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table2) = c("Pathogenic", "Benign")
rownames(con_table2) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table2_p_value = fisher.test(con_table2)
#Variants_removed_after_PA_QC1
var_rem_para_con_TP = con_table[2]-con_table2[2]
var_rem_para_con_FP = con_table[4]-con_table2[4]
var_rem_para_con_FN = length(p.paralog_data$ID)-var_rem_para_con_TP
var_rem_para_con_PPV = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FP)
var_rem_para_con_Sensitivity = var_rem_para_con_TP/(var_rem_para_con_TP+var_rem_para_con_FN)
var_rem_para_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_para_con_TP,
    length(b.paralog_data$ID),
    var_rem_para_con_FP
  ), ncol = 2
)
colnames(var_rem_para_con) = c("Pathogenic", "Benign")
rownames(var_rem_para_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_para_con_p_value = fisher.test(var_rem_para_con)

#all_con
#pathogenic to pathogenic
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
p.paralog_data = dplyr::filter(p.paralog_data, Gene %in% channelopathy_gene)
p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$REF_Amino_acids.x==ptop.Total_paralog_annotations$REF_Amino_acids.y,]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
b.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyBenign.out_paraloc_paralogs2.all_con", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", "Gene", paste("paralog", 1:132, sep = "")))
channelopathy_gene = c("KCNQ1","KCNH2","SCN5A","KCNE1","KCNE2","RYR2")
b.paralog_data = dplyr::filter(b.paralog_data, Gene %in% channelopathy_gene)
b.paralog_data = left_join(b.paralog_data,b.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
b.gathered_paralog_data = filter(gather(b.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
btop.Total_paralog_annotations = left_join(b.gathered_paralog_data, p.ref_data, by = c("paralog_pos" = "Variant_pos"))
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$REF_Amino_acids.x==btop.Total_paralog_annotations$REF_Amino_acids.y,]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table3_TP = ptop.num_of_paralog_anno
con_table3_FP = btop.num_of_paralog_anno
con_table3_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table3_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table3_PPV = con_table3_TP/(con_table3_TP+con_table3_FP)
con_table3_Sensitivty = con_table3_TP/(con_table3_TP+con_table3_FN)
con_table3 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table3) = c("Pathogenic", "Benign")
rownames(con_table3) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table3_p_value = fisher.test(con_table3)
#Variants_removed_after_PA_QC2
var_rem_all_con_TP = con_table2[2]-con_table3[2]
var_rem_all_con_FP = con_table2[4]-con_table3[4]
var_rem_all_con_FN = length(p.paralog_data$ID)-var_rem_all_con_TP
var_rem_all_con_PPV = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FP)
var_rem_all_con_Sensitivity = var_rem_all_con_TP/(var_rem_all_con_TP+var_rem_all_con_FN)
var_rem_all_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_all_con_TP,
    length(b.paralog_data$ID),
    var_rem_all_con_FP
  ), ncol = 2
)
colnames(var_rem_all_con) = c("Pathogenic", "Benign")
rownames(var_rem_all_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_all_con_p_value = fisher.test(var_rem_all_con)

#alt_con
#pathogenic to pathogenic
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[ptop.Total_paralog_annotations$ALT_Amino_acids.x==ptop.Total_paralog_annotations$ALT_Amino_acids.y,]
ptop.Total_paralog_annotations = ptop.Total_paralog_annotations[!duplicated(ptop.Total_paralog_annotations$ID.x),]
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))
#benign to pathogenic
btop.Total_paralog_annotations = btop.Total_paralog_annotations[btop.Total_paralog_annotations$ALT_Amino_acids.x==btop.Total_paralog_annotations$ALT_Amino_acids.y,]
btop.Total_paralog_annotations = btop.Total_paralog_annotations[!duplicated(btop.Total_paralog_annotations$ID.x),]
btop.num_of_paralog_anno = sum(!is.na(btop.Total_paralog_annotations$ID.y))
#stats
con_table4_TP = ptop.num_of_paralog_anno
con_table4_FP = btop.num_of_paralog_anno
con_table4_TN = length(b.paralog_data$ID)-btop.num_of_paralog_anno
con_table4_FN = length(p.paralog_data$ID)-ptop.num_of_paralog_anno
con_table4_PPV = con_table4_TP/(con_table4_TP+con_table4_FP)
con_table4_Sensitivty = con_table4_TP/(con_table4_TP+con_table4_FN)
con_table4 = matrix(
  c(length(p.paralog_data$ID),
    ptop.num_of_paralog_anno,
    length(b.paralog_data$ID),
    btop.num_of_paralog_anno
  ), ncol = 2
)
colnames(con_table4) = c("Pathogenic", "Benign")
rownames(con_table4) = c("Number of variants in total", "Number of variants predicted as pathogenic")
con_table4_p_value = fisher.test(con_table4)
#Variants_removed_after_PA_QC3
var_rem_alt_con_TP = con_table3[2]-con_table4[2]
var_rem_alt_con_FP = con_table3[4]-con_table4[4]
var_rem_alt_con_FN = length(p.paralog_data$ID)-var_rem_alt_con_TP
var_rem_alt_con_PPV = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FP)
var_rem_alt_con_Sensitivity = var_rem_alt_con_TP/(var_rem_alt_con_TP+var_rem_alt_con_FN)
var_rem_alt_con = matrix(
  c(length(p.paralog_data$ID),
    var_rem_alt_con_TP,
    length(b.paralog_data$ID),
    var_rem_alt_con_FP
  ), ncol = 2
)
colnames(var_rem_alt_con) = c("Pathogenic", "Benign")
rownames(var_rem_alt_con) = c("Number of variants in total", "Number of variants predicted as pathogenic")
var_rem_alt_con_p_value = fisher.test(var_rem_alt_con)

#Overall_table
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c(length(p.paralog_data$ID),length(b.paralog_data$ID),"NA","NA","NA")
Paralogue_Annotation_no_QC = c(format(con_table[2], scientific = FALSE),format(con_table[4], scientific = FALSE),con_table2_PPV,con_table_Sensitivty,con_table_p_value$p.value)
Variants_remaining_after_PA_QC1 = c(format(con_table2[2], scientific = FALSE),format(con_table2[4], scientific = FALSE),con_table2_PPV,con_table2_Sensitivty,con_table2_p_value$p.value)
Variants_removed_after_PA_QC1 = c(var_rem_para_con[2],var_rem_para_con[4],var_rem_para_con_PPV,"NA",var_rem_para_con_p_value$p.value)
Variants_remaining_after_PA_QC2 = c(format(con_table3[2], scientific = FALSE),format(con_table3[4], scientific = FALSE),con_table3_PPV,con_table3_Sensitivty,con_table3_p_value$p.value)
Variants_removed_after_PA_QC2 = c(var_rem_all_con[2],var_rem_all_con[4],var_rem_all_con_PPV,"NA",var_rem_all_con_p_value$p.value)
Variants_remaining_after_PA_QC3 = c(format(con_table4[2], scientific = FALSE),format(con_table4[4], scientific = FALSE),con_table4_PPV,con_table4_Sensitivty,con_table4_p_value$p.value)
Variants_removed_after_PA_QC3 = c(var_rem_alt_con[2],var_rem_alt_con[4],var_rem_alt_con_PPV,"NA",var_rem_alt_con_p_value$p.value)

channelopathy_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )

```

On the other hand, channelopathy genes did annotate well suggesting that their paralogues are involved in disease. 

Looking at alt alleles. Taking only pairwise alignments where the alt allele is conserved leaves only 1115 individual pairwise alignments. The number of actual unique variants this equates to is less - 825.

#### Annotation of all possible missense variations in the 8 sarcomeric genes and calculation of EF

For calculating the EFs, run the all possible missense variants through VEP+plugin and return paraloc locations. Then join those locations with pathogenic clinvar variants as before. This indicates which variants from all possible missense variants are likely to be pathogenic. Then we check to see if any of these variants are present in the cases and controls. Hopefully the controls will be less but there is more control data than cases bare in mind. Calculate the EFs using that. Remember though the EFs are based on how many times an allele is seen, not the number of different alleles by themselves.

```{r eval = FALSE, include = FALSE}
#noQC
#all missense data
art.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/remapped_cm_mutation_vep_canon_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_art_var = art.paralogous_var_align$Total_paralog_annotations[!is.na(art.paralogous_var_align$Total_paralog_annotations$ID.y),]
#case data
case.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.vcf", sep = "\t", skip = 4)
case.data$Variant_pos = paste(case.data$X.CHROM, case.data$POS, sep = " ")
patho_case.data = case.data[case.data$Variant_pos %in% patho_art_var$Variant_pos,]
#control data
control.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.vcf", sep = "\t", skip = 4)
control.data$Variant_pos = paste(control.data$X.CHROM, control.data$POS, sep = " ")
patho_control.data = control.data[control.data$Variant_pos %in% patho_art_var$Variant_pos,]

#case data VEP
case.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_HCM_missense_LMM_OMGL_chr_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_case_var = case.paralogous_var_align$Total_paralog_annotations[!is.na(case.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_case_var_IDs = patho_case_var$ID.x
# write(patho_case_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_case_var_IDs.txt", sep = "\t", ncolumns = 1)
case.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/HCM_missense_LMM_OMGL_chr.csv", sep = ",")
patho_case.data = case.data[case.data$mut_id %in% patho_case_var_IDs,]
case_cases = sum(patho_case.data$sum.vs_case_count.)
Total_case_cases = sum(case.data$sum.vs_case_count.)

#control data VEP
control.paralogous_var_align = Paralogous_var_align("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/remapped_ExAC_missense_sarcomeric_chr_wRefAl.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
patho_control_var = control.paralogous_var_align$Total_paralog_annotations[!is.na(control.paralogous_var_align$Total_paralog_annotations$ID.y),]
patho_control_var_IDs = patho_control_var$ID.x
# write(patho_control_var_IDs, file = "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/patho_control_var_IDs.txt", sep = "\t", ncolumns = 1)
control.data = read.csv("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/ExAC_missense_sarcomeric_chr.csv", sep = ",")
patho_control.data = control.data[control.data$mut_id %in% patho_control_var_IDs,]
control_cases = sum(patho_control.data$mut_exac_count)
Total_control_cases = sum(control.data$mut_exac_count)

```
<!--
```{r eval = FALSE, include = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)
#noQC
#all missense variants join to cases
#all missense data
art.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/cm_mutation_vep_canon.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(art.paralog_data, sep = "\t"))-6) #-6 for "Variant_pos", "ID", "Gene", "REF", "ALT", and "\n"
art.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/cm_mutation_vep_canon.out_paraloc_paralogs2.noQC", header = FALSE, sep = "\t", col.names = c("Variant_pos", "ID", "Gene", "REF", "ALT", paste("paralog", 1:max_no_col, sep = "")))
art.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/all_possible_mutation_icc/cm_mutation_vep_canon.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
art.tableized_data$Variant_pos = paste(art.tableized_data$CHROM,art.tableized_data$POS, sep = " ")
art.tableized_data$REF_Amino_acids = sapply(art.tableized_data[,"Amino_acids"],strsplit, "/")
art.tableized_data$REF_Amino_acids = sapply(art.tableized_data[,"REF_Amino_acids"],unlist)
art.tableized_data$REF_Amino_acids = sapply(art.tableized_data[,"REF_Amino_acids"],function(x) x[1])
art.tableized_data$ALT_Amino_acids = sapply(art.tableized_data[,"Amino_acids"],strsplit, "/")
art.tableized_data$ALT_Amino_acids = sapply(art.tableized_data[,"ALT_Amino_acids"],unlist)
art.tableized_data$ALT_Amino_acids = sapply(art.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
art.ref_data = art.tableized_data
art.paralog_data = left_join(art.paralog_data,art.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
art.gathered_paralog_data = filter(gather(art.paralog_data, paralog, paralog_pos, paste("paralog", 1:max_no_col, sep = ""), factor_key = TRUE), paralog_pos != "")
#pathogenic clinvar data


#case data
case.paralog_data = file("/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/HCM_missense_LMM_OMGL_chr.out_paraloc_paralogs2.noQC")
max_no_col = (max(count.fields(case.paralog_data, sep = "\t"))-4)
case.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/HCM_missense_LMM_OMGL_chr.out_paraloc_paralogs2.noQC", header = FALSE, sep = "\t", col.names =  c("Variant_pos", "ID", "Gene", paste("paralog", 1:max_no_col, sep = "")))
case.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/HCM_missense_LMM_OMGL_chr.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors = FALSE)
case.tableized_data$Variant_pos = paste(case.tableized_data$CHROM,case.tableized_data$POS, sep = " ")
case.tableized_data$REF_Amino_acids = sapply(case.tableized_data[,"Amino_acids"],strsplit, "/")
case.tableized_data$REF_Amino_acids = sapply(case.tableized_data[,"REF_Amino_acids"],unlist)
case.tableized_data$REF_Amino_acids = sapply(case.tableized_data[,"REF_Amino_acids"],function(x) x[1])
case.tableized_data$ALT_Amino_acids = sapply(case.tableized_data[,"Amino_acids"],strsplit, "/")
case.tableized_data$ALT_Amino_acids = sapply(case.tableized_data[,"ALT_Amino_acids"],unlist)
case.tableized_data$ALT_Amino_acids = sapply(case.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
case.ref_data = case.tableized_data
case.paralog_data = left_join(case.paralog_data,case.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))
case.gathered_paralog_data = filter(gather(case.paralog_data, paralog, paralog_pos, paste("paralog", 1:max_no_col, sep = ""), factor_key = TRUE), paralog_pos != "")
#all missense to case
arttocase.Total_paralog_annotations = left_join(art.gathered_paralog_data,case.ref_data, by = c("paralog_pos" = "Variant_pos"))
arttocase.num_of_paralog_anno = sum(!is.na(arttocase.Total_paralog_annotations$ID.y))


ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.num_of_paralog_anno = sum(!is.na(ptop.Total_paralog_annotations$ID.y))

#all missense variants join to controls
b.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/case_controls/ExAC_missense_sarcomeric_chr.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
b.tableized_data$Variant_pos = paste(b.tableized_data$CHROM,b.tableized_data$POS, sep = " ")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],unlist)
b.tableized_data$REF_Amino_acids = sapply(b.tableized_data[,"REF_Amino_acids"],function(x) x[1])
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"Amino_acids"],strsplit, "/")
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],unlist)
b.tableized_data$ALT_Amino_acids = sapply(b.tableized_data[,"ALT_Amino_acids"],function(x) x[2])
b.ref_data = b.tableized_data
ptob.Total_paralog_annotations = left_join(p.gathered_paralog_data,b.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptob.num_of_paralog_anno = sum(!is.na(ptob.Total_paralog_annotations$ID.y))


```
-->

Total cases: 6140
number of affected cases: 39

Total controls: 60678
number of affected controls: 28

Odds ratio 	13.7648
95 % CI:	8.4648 to 22.3833
z statistic	10.570
Significance level	P < 0.0001
Attributable Risk Percent: 92.7% 
95 % CI: 79.6 to 100

#### Paralogue stats
According to ensembl, 92096 protein coding genes are defined to have paralogues. While 7958 protein genes do not have paralogues


#### Plan for overall annotations table
REF              | All | Alt matches |Alt no match
-----------------|-----|-------------|-------------
                 |     |             |
No QC            |     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
Paralog Conserved|     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
All Conserved    |     |             |
                 |     |             |
                 
#### Para-Z scores
For the para-z scores, will need to extract amino acid position from VEP output as well. Then look up the gene in question in para-z score folder, and using the position identify the para-z score. From my understanding, the para-z score is the same across aligned amino acids in the same gene family. Therefore, we could use a cut-off threshold to further improve our confidence in calling variants pathogenic etc. We could also then calculate ROC curves by altering the cut-off to see how that affects sensitivity/PPV.

```{r eval = FALSE, include = FALSE}
Sensitivity_list = vector()
FPR_list = vector()
con_table_list = matrix(, ncol = 2)
for (i in seq(0,11)){
#noQC
#pathogenic to pathogenic
p.paraz_var_align = ParaZ_var_align(i,"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
#benign to pathogenic
b.paraz_var_align = ParaZ_var_align(i,"/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_paralogs2.noQC", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Benign_and_LikelyBenign.out_paraloc_tableized", "/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_alleles.single.b38.Pathogenic_and_LikelyPathogenic.out_paraloc_tableized")
paraz.conf = conf_matrix(p.paraz_var_align$num_of_paralog_anno, p.paraz_var_align$paralog_data, b.paraz_var_align$num_of_paralog_anno, b.paraz_var_align$paralog_data)
Sensitivity_list = c(Sensitivity_list, paraz.conf$Sensitivity)
FPR_list = c(FPR_list, paraz.conf$FPR)
con_table_list = rbind(con_table_list, paraz.conf$con_table)
}
roc_data = do.call(rbind, Map(data.frame, Sensitivity = Sensitivity_list, FPR = FPR_list, cutoff = seq(0,11)))
gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = cutoff)) + geom_point() + xlim(0,1) + ylim(0,1)
gg = gg + geom_abline(intercept = 0, slope = 1, linetype = "dotted")

gg = ggplot(roc_data, aes(x=FPR, y=Sensitivity, colour = cutoff)) + geom_point() + xlim(0,0.25) + ylim(0.7,1)
```

#### Ohnologs
The "2R"" hypothesis states that some 500 million years ago, early vertebrates went through 2 rounds of whole genome duplication (WGD)[@Ohno1968]. Paralogues that arose from this WGD are known as ohnologs. @Singh2014 showed that monogenic disease genes to be enriched in ohnologs than other paralogs that arose from small scale duplications.