---
title: "Paralog Annotation Notes"
output:
  html_document:
    df_print: paged
---
### Aims
* Apply Paralogue annotation on other datasets
    + "Genome Wide" - Clinvar dataset
    + Cardiomyopathy genes - [MYH7, MYBPC3, TNNT2, TPM1, MYL2, MYL3, TNNI3, ACTC1]
    + Channelopathy genes - [KCNQ1, KCNH2, SCN5A, KCNE1, KCNE2, RYR2]
* Improve precision via increasing conservation of ref/alt alleles
    + pairwise QC - ignore any individual pairwise alignments where ref alleles are not conserved
    + pairwise QC and family QC - ignore entire alignment columns if the entire family ref alleles are not conserved; NB analogous to para z scores
    + pairwise QC and family QC and alt allele QC - ignore entire alignment columns if family ref allele and alt allele isn't conserved
* investigate para z scores
* investigate pfam meta domains

### Material and Methods Notes

The Paralog Annotation algorithm was wriiten by Erica as a perl script plugin (called __ParalogueAnno_plugin_cleanup.pm__) for Ensembl's VEP version 90 (https://www.ensembl.org/info/docs/tools/vep/script/vep_options.html). 

The plugin has two arguments:

* the first parameter has 2 options:
    + ```variant``` (default) returns only the paralogous variants if any are present in the associated paralogs of the query gene found in the ensembl compara database
    + ```paraloc``` returns only paralog variant locations in the form of genomic coordinates of the corresponding codon in ALL paralogs;
* the second parameter has 2 options:
    + ```all``` for all variants;
    +  ```damaging``` (default) for only damaging variant.
The majority of the time ```paraloc``` mode is used.

The Ensembl team have touched up Erica's plugin and decrease runtime. The plugin is now called __ParalogueAnnotation.pm__

The initial output by VEP and the Plugin (VEP+Plugin) is not reader friendly for either the user nor if you want to parse informations. So a python wrapper, shwon below, for the VEP+Plugin was written to automatically parse the results, namely the paralogous variant information - __/data/Share/nick/Paralog_Anno/VEP_ParalogAnno.py__(Note the code below is not polished for release and is a WIP).
```{python eval = FALSE, include = FALSE}
import  os, sys, subprocess, re
import shlex, subprocess

#WRAPPER FOR VEP+PLUGIN

def VEP_Plugin_run(flavour, input_dir, input_file, genome_build, VEPversion):
	# flavour = int(sys.argv[1])	#0(no plugin), 1(variants), 2(paraloc)
	# input_dir = sys.argv[2]	#input directory
	# input_file = sys.argv[3]	#name of input file
	# # output_file = sys.argv[3]	#name of output file
	# genome_build = str(sys.argv[4]) #37, 38
	# VEPversion = int(sys.argv[5]) #83, 87, 90

	#e.g. python3 VEP_ParalogAnno.py 0 /data/Share/nick/Paralog_Anno/data_files 10_pathogenic.vcf 38 90

	# if not len(os.listdir("/data/Share/nick/Paralog_Anno/homo_sapiens/")) == 0:
	# 	sys.exit("Terminating script:/data/Share/nick/Paralog_Anno/homo_sapiens/ not empty. Please empty and try again")

	# print(type(flavour))
	if flavour == 0:
		# print("check")
		flavour = ""
		output_file = input_file.rsplit(".",1)[0]+".out_no_plugin"
		print(output_file)
	# elif flavour == 1:
	# 	flavour = "--plugin ParalogueAnno_plugin_cleanup"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_old"
	# 	print(output_file)
	# elif flavour == 2:
	# 	flavour = "--plugin ParalogueAnno_plugin_cleanup,paraloc"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_old_paraloc"
	# 	print(output_file)
	# elif flavour == 3:
	# 	flavour = "--plugin ParalogueAnnotation"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_new"
	# 	print(output_file)
	# elif flavour == 4:
	# 	flavour = "--plugin ParalogueAnnotation,paraloc"
	# 	output_file = input_file.rsplit(".",1)[0]+".out_new_paraloc"
	# 	print(output_file)
	elif flavour == 1:
		flavour = "--plugin ParalogueAnnotation"
		output_file = input_file.rsplit(".",1)[0]+".out"
		print(output_file)
	elif flavour == 2:
		flavour = "--plugin ParalogueAnnotation,paraloc"
		output_file = input_file.rsplit(".",1)[0]+".out_paraloc"
		print(output_file)

	# print(input_file)

	if not input_dir.endswith("/"):
		input_dir = input_dir+"/"
	print(input_dir)


	'''if VEPversion == 83:
		#VEP version 83
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/83_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/83_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			"perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/variant_effect_predictor.pl --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/83_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)
	elif VEPversion == 87:
		#VEP version 87
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/87_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/87_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			"perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-tools-release-87/ensembl-tools/scripts/variant_effect_predictor/variant_effect_predictor.pl --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/87_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)'''
	if VEPversion == 90:
		#VEP version 90
		# rm_command = ("rm -rf /data/Share/nick/Paralog_Anno/homo_sapiens/90_GRCh" + genome_build)
		mv_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens_non_use/90_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens"
			)
		command = (
			# "perl -I /data/Install/ensembl-tools-release-83/scripts/variant_effect_predictor/.vep/Plugins /data/Install/ensembl-vep/vep --force_overwrite --vcf --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			"perl -I /data/Share/nick/Paralog_Anno /data/Install/ensembl-vep/vep --force_overwrite --vcf --allele_number --offline --cache --dir_cache /data/Share/nick/Paralog_Anno/" +
			" -i " + input_dir + input_file +
			" -o " + input_dir + output_file + 
			" " + flavour
			)
		mv_back_command = (
			"mv /data/Share/nick/Paralog_Anno/homo_sapiens/90_GRCh" + genome_build + " /data/Share/nick/Paralog_Anno/homo_sapiens_non_use"
			)
	
	# print(mv_command)
	# os.system(mv_command)
	print(command)
	os.system(command)
	# print(mv_back_command)
	# os.system(mv_back_command)
	
	with open(
		input_dir + output_file + 
		"_summary.html"
		) as infile:
		for line in infile:
			if "<td>Run time</td>" in line:
				if VEPversion == 83 or VEPversion == 87:
					#IF VEP87 AND 83
					m = re.search(r"<td>Run time<\/td> <td>.*second.<\/td>", line)
					# print(m.group())
					run_time = m.group().split("</td> <td>")[1].rstrip("</td>")
					print(run_time, "\n")
				elif VEPversion == 90:
					# #IF VEP90
					m = re.search(r"<td>Run time<\/td><td>.*second.<\/td>", line)
					# print(m.group())
					run_time = m.group().split("</td><td>")[1].rstrip("</td>")
					print(run_time, "\n")

	###TESTED THE BELOW WITH VEP90#### MIGHT NOT WORK WITH OTHER VERSIONS
	#reads through outfile of vep+plugin and tidies up data by extracting only variants that have paralogs and spits that out as another outfile
	out_file = open(input_dir + output_file + "_paralogs", "w")
	if not flavour == "":
		with open(
			input_dir + output_file
			) as infile:
			for line in infile:
				if not line.startswith("#"):
					line = line.split("CSQ=")
					CSQ = line[0]
					print(CSQ)	#print query variant
					line = line[1].split(",")
					# print(line, "\n")
					for x in line:
						y = x.split("|")
						# print(y, "\n")
						gene = y[3]
						paralog = y[23]
						if paralog == "" or paralog == "\n":
							paralog_check = 0
						else:
							paralog_check = 1
						if paralog_check == 1:
							print(x, "\n")
							out_file.write(
								str(CSQ)+"PARALOGS->"
								+str(x)+"\n"
								)
				else:
					out_file.write(line)
	out_file.close()
	
	# ###FOR CLINVAR VARIANTS AT LEAST###SCRAP ALL OF THIS, USE R TIDYVERSE JOIN METHOD INSTEAD
	# out_file = open(input_dir + output_file + "_aligned_vars", "w")
	# with open(input_dir + output_file + "_paralogs") as infile1:
	# 	infile2 = open(input_dir + input_file, "r") 
	# 	text_list = infile2.read()
	# 	infile2.close()
	# 	####TEST_BELOW####
	# 	# chromosome = 1
	# 	# pos1 = 1050574
	# 	# pos2 = 1050575
	# 	# pos3 = 1050576
	# 	# s1 = re.findall("^"+str(chromosome)+"\t"+str(pos1)+"\s.*$", text_list, re.MULTILINE)
	# 	# s2 = re.findall("^"+str(chromosome)+"\t"+str(pos2)+"\s.*$", text_list, re.MULTILINE)
	# 	# s3 = re.findall("^"+str(chromosome)+"\t"+str(pos3)+"\s.*$", text_list, re.MULTILINE)		
	# 	# if s1:
	# 	# 	print("YES")
	# 	# 	print(s1)
	# 	# if s2:
	# 	# 	print(s2)
	# 	# if s3:
	# 	# 	print(s3)
	# 	####END_OF_TEST####
	# 	for line in infile1:
	# 		if not line.strip() == "":
	# 			query_variant = line.split("PARALOGS->")[0]
	# 			out_file.write(query_variant)
	# 			paralog_line = line.split("PARALOGS->")[1].rstrip()
	# 			paralog_line = paralog_line.split("|&")[1]
	# 			paralog_line = paralog_line.split("&")
	# 			paralog_line = list(filter(None, paralog_line))
	# 			print(paralog_line)
	# 			out_file.write(str(paralog_line)+";")
	# 			for paralog in paralog_line:
	# 				if not paralog == "":
	# 					print(paralog)
	# 					location = paralog.split(":")[1]
	# 					chromosome = location.split("_")[0].lstrip("chr")
	# 					position = location.split("_")[1]
	# 					position1 = int(position.split("-")[0])
	# 					position2 = int(position.split("-")[1])
	# 					positions = list(range(position1,position2+1))
	# 					print(chromosome, type(positions), positions)
	# 					# p = re.compile(r"^"+str(chromosome)+r"\t"+positions+r"\t")
	# 					for pos in positions:
	# 						s = re.findall("^"+str(chromosome)+"\t"+str(pos)+"\s.*$",text_list, re.MULTILINE)
	# 						if s:
	# 							print(s)
	# 							out_file.write(str(s)+";")
	# 			out_file.write("\n")
	# out_file.close()
	# #################################################
	
# VEP_Plugin_run(2, "/data/Share/nick/Paralog_Anno/data_files", "new_extracted_clinvar_cm_intepretated_raw_variants_2.vcf", "38", 87)


flavour = int(sys.argv[1])	#0(no plugin), 1(variants), 2(paraloc),
input_dir = sys.argv[2]	#input directory
input_file = sys.argv[3]	#name of input file
genome_build = str(sys.argv[4]) #37, 38
VEPversion = int(sys.argv[5]) #83, 87, 90
VEP_Plugin_run(flavour, input_dir, input_file, genome_build, VEPversion)
```

An intermediate python script (__File_prep_for_R.py__) was used to prep the results into R friendly data. Furthermore it could also be used to perform pairwise and family QC. Incidentally, the pairwise QC could be performed directly in R after the raw results are processed by __tableize_vcf.py__ and tabulated (see below).

__/data/Share/nick/Paralog_Anno/File_prep_for_R.py__ - formats results from __VEP_ParalogAnno.py__ into tabulated format ready for R processing
```{python eval = FALSE, include = FALSE}
import  os, sys, subprocess, re

#SCRIPT FOR SEPARATING EACH LINE IN VCF OUTPUT FROM VEP_PARALOGANNO.PY INTO TAB SEP FILE READY FOR JOIN IN R
#don't use if VEP_ParalogAnno.py has been edited to already do this, check to see if same code exists in that script
###BELOW ONLY WORKS FOR CLINVAR VCF FILES (EXAC HAS NO INFO IN ID VARIABLE)

input_file = sys.argv[1]	#path of VEP_ParalogAnno.py paralogs output file
dir1 = input_file.rsplit("/", 1)[0]
input_file2 = sys.argv[2]	#path of vcf file containing only orginal query variants list used as input for VEP_ParalogAnno.py. I.e. the very first input file!
refid_flavour = sys.argv[3] #whether to consider refID or not. 0: not, 1: only consider paralogous pair alignments where refID = 1, 2: only consider total gene family paralogous alignments where all refID in that family = 1

if refid_flavour == "noQC":
	out_file = open(input_file+"2.noQC", "w")
elif refid_flavour == "para_con":
	out_file = open(input_file+"2.para_con", "w")
elif refid_flavour == "all_con":
	out_file = open(input_file+"2.all_con", "w")

max_split = 0	#check to see max number of columns for R import as wide format
max_paralog_codons = ""
with open(input_file, "r") as f:
	for line in f:
		if line != "\n":
			# print(line)
			chrom = line.split()[0]
			position = line.split()[1]
			ID = line.split()[2]
			print(chrom+" "+position+"\t"+str(ID)+"\t")

			out_file.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\t")
			
			line = line.rstrip()
			paralog_codons = line.split("PARALOGS->")[1]
			paralog_codons = paralog_codons.split("|&")[1]
			paralog_codons = paralog_codons.split("&")
			# paralog_codons.pop(0)
			paralog_codons.remove("")
			print(paralog_codons)
			refid_flav2_check = 0
			if refid_flavour == "all_con":
				if all(refids.endswith("REFID=1") for refids in paralog_codons) == True:
					refid_flav2_check = 1

			for codon in paralog_codons:
				print(codon)

				if refid_flavour == "noQC" or refid_flav2_check == 1:
					refID = codon.split(":")[-1]
					print(refID)
					codon = codon.split(":")[1]
					paralog_chrom = codon.split("_")[0].lstrip("chr")
					paralog_position = int(codon.split("-")[1])
					print(
						str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position)+"\t"
						)

					out_file.write(
						str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
						str(paralog_chrom)+" "+str(paralog_position)+"\t"
						)

				if refid_flavour == "para_con":
					refID = codon.split(":")[-1]
					if refID == "REFID=1":
						print(refID)
						codon = codon.split(":")[1]
						paralog_chrom = codon.split("_")[0].lstrip("chr")
						paralog_position = int(codon.split("-")[1])
						print(
							str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position)+"\t"
							)

						out_file.write(
							str(paralog_chrom)+" "+str(paralog_position-2)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position-1)+"\t"+
							str(paralog_chrom)+" "+str(paralog_position)+"\t"
							)

				# if refid_flavour == 2:
				# 	refID = 

			out_file.write("\n")

			if len(paralog_codons)*3 + 2 > max_split:
				max_split = len(paralog_codons)*3 + 2
				# print(len(paralog_codons))
				# print(paralog_codons)
				max_paralog_codons = paralog_codons
			# newline[1] = "\t".join(newline[1])
			# newline = "\t".join(newline)
			# out_file.write(newline)
	print(max_split, max_paralog_codons)	#number of columns for dataframe in R; + 2 for (chrom poistion) and (ID)
out_file.close()

if input_file2 != "null":
	if input_file2.endswith(".vcf"):
		out_file2 = open(input_file2+"_onlyVariantslist", "w")
		with open(input_file2, "r") as f:
			for line in f:
				if not line.startswith("#"):
					chrom = line.split()[0]
					position = line.split()[1]
					ID = line.split()[2]
					out_file2.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\n")
	# elif input_file2.endswith("out_paraloc"):
	# 	out_file2 = open(input_file2+"_onlyVariantslist", "w")
	# 	with open(input_file2, "r") as f:
	# 		for line in f:
	# 			if not line.startswith("#"):
	# 				chrom = line.split()[0]
	# 				position = line.split()[1]
	# 				ID = line.split()[2]
	# 				amino_acids = line.split("CSQ=")[1]

	# 				out_file2.write(str(chrom)+" "+str(position)+"\t"+str(ID)+"\n")
	out_file2.close()
```

As paraloc mode only returns ref alleles. The alt alleles were extracted from the VEP information. This was done by using __tableize_vcf.py__.
__/data/Share/nick/Paralog_Anno/loftee/src/tableize_vcf.py__ was used to format the VEP output into table format for R processing. For example:
```{bash eval = FALSE, include = TRUE}
python /data/Install/LOFTEE/loftee-master/src/tableize_vcf.py --vcf /data/Share/nick/Paralog_Anno/data_files/clinvar_20171029_onlyPathogenic.out_paraloc --out /data/Share/nick/Paralog_Anno/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized --do_not_minrep --include_id --vep_info Amino_acids,Codons,Paralogue_Vars
```
However, a python wrapper plus additional formatting (__/data/Share/nick/Paralog_Anno/Tableize_wrapper.py__) that tableize couldn't do, i.e. separate variants that had multiple REF and ALT alleles was used instead to prepare the data for R.
```{python eval = FALSE, include = FALSE}

```

#### Benchmarking performance of the plugin

__/data/Share/nick/Paralog_Anno/multi_vcf_extractor_benchmark.py__ is used to demonstrate speed at which VEP+Plugin takes to run
```{python eval = FALSE, include = FALSE}
#Script for benchmarking VEP with different flavours by auto extracting set number of variants and running them

import os, sys, subprocess, re
sys.path.insert(0, "/data/Install/Python-3.6.3/")
from VEP_ParalogAnno import *
import numpy as np
import pandas as pd
from scipy import stats, integrate
import matplotlib.pyplot as plt
import seaborn as sns

file = sys.argv[1]	#e.g. data_files/ExAC.r1.sites.vep_onlySCN5A_GRCh38coords_wRefAl.vcf
dir1 = file.split("/")
onlyfile = dir1.pop()
dir1 = "/".join(dir1)
if not dir1.endswith("/"):
	dir1 = dir1+"/"

flavour = int(sys.argv[2]) #e.g 0
VEPversion = int(sys.argv[3]) #e.g. 90

x_list = list(range(100,1100,100))
x_list.append(10)
x_list.sort()
y_list = []

for x_var in x_list:
	input_file = str(onlyfile.rstrip(".vcf"))+".first"+str(x_var)+"lines.vcf"
	os.system("head -n"+str(x_var)+" "+str(dir1)+str(onlyfile)+" > "+str(dir1)+str(input_file))

	VEP_Plugin_run(
		flavour, 
		"/data/Share/nick/Paralog_Anno/data_files", 
		input_file,
		"38",
		VEPversion)

	if flavour == 0:
		output_file = input_file.rsplit(".",1)[0]+".out_no_plugin"
		print(output_file)
	elif flavour == 1:
		output_file = input_file.rsplit(".",1)[0]+".out"
		print(output_file)
	elif flavour == 2:
		output_file = input_file.rsplit(".",1)[0]+".out_paraloc"
		print(output_file)

	with open(
		dir1 + output_file + 
		"_summary.html"
		) as infile:
		for line in infile:
			if "<td>Run time</td>" in line:
				if VEPversion == 83 or VEPversion == 87:
					#IF VEP87 AND 83
					m = re.search(r"<td>Run time<\/td> <td>.*second.<\/td>", line)
					run_time = m.group().split("</td> <td>")[1].rstrip("</td>")
					print(run_time, "\n")
				elif VEPversion == 90:
					# #IF VEP90
					m = re.search(r"<td>Run time<\/td><td>.*second.<\/td>", line)
					run_time = m.group().split("</td><td>")[1].rstrip("</td>")
					print(run_time, "\n")
				y_list.append(int(run_time.split()[0]))

x_list = [0] + x_list
y_list = [0] + y_list
print(x_list)
print(y_list)


fig, ax = plt.subplots(figsize=(15,10))
axes = plt.gca()
plt.plot(x_list, y_list, 'ro')
plt.xlabel("Number of variants in query", fontsize=15)
plt.ylabel("Time taken (seconds)", fontsize=20)
if flavour == 0:
	plt.title("Time taken for VEP (no plugin) to run against number of varaints being searched")
elif flavour == 1:
	plt.title("Time taken for VEP (plugin in '-varaints' mode) to run against number of varaints being searched")
elif flavour == 2:
	plt.title("Time taken for VEP (plugin in '-paraloc' mode) to run against number of varaints being searched")
# plt.show()
plt.savefig(dir1+"benchmark_"+str(flavour)+"_"+str(VEPversion)+".png")
plt.gcf().clear()
```

#### Annotation of Clinvar
The Clinvar file __clinvar_20171029.vcf__ was downloaded from ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/. Note that since the initial look at what was available there's been updated Clinvar files. 

NOTE that I have noticed some descrepencies between the plugin annotations which call REFID = 1/0 and that of comparing the REF amino acid by VEP in the dataset to itself. For exmaple:
Looking at variant ID 101429 (located at chr 2 position 189007510) and its paralogous variant ID 444000 (located at chr 17 position 50187974), according to VEP 101429 ref amino acid is G and 444000 ref amino acid is Q.But the plugin still gives a REFID=1
```

```

```{r eval = FALSE, include = FALSE}
Packages = c("tidyverse", "dplyr", "ggplot2", "ggsignif", "huxtable")
lapply(Packages, library, character.only = TRUE)

#Pathogenic variants
#python processed files
p.paralog_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_paralogs2", sep="\t", header=FALSE, col.names = c("Variant_pos", "ID", paste("paralog", 1:132, sep = "")))

#tableized files
p.tableized_data = read.csv(file="/media/nick/Data/Users/N/Documents/PhD/Paralogues/data_files/clinvar_20171029_onlyPathogenic.out_paraloc_tableized", sep = "\t", header=TRUE, stringsAsFactors=FALSE)
p.tableized_data$Variant_pos = paste(p.tableized_data$CHROM,p.tableized_data$POS, sep = " ")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],unlist)
p.tableized_data$REF_Amino_acids = sapply(p.tableized_data[,"REF_Amino_acids"],function(x) x[1])
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"Amino_acids"],strsplit, "/")
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],unlist)
p.tableized_data$ALT_Amino_acids = sapply(p.tableized_data[,"ALT_Amino_acids"],function(x) x[2])

p.tableized_data = p.tableized_data[,c("Variant_pos","ID","REF","ALT","REF_Amino_acids","ALT_Amino_acids","Codons")]

p.paralog_data = left_join(p.paralog_data,p.tableized_data, by = c("Variant_pos" = "Variant_pos", "ID" = "ID"))

p.ref_data = p.tableized_data

p.gathered_paralog_data = filter(gather(p.paralog_data, paralog, paralog_pos, paste("paralog", 1:132, sep = ""), factor_key = TRUE), paralog_pos != "")
ptop.Total_paralog_annotations = left_join(p.gathered_paralog_data,p.ref_data, by = c("paralog_pos" = "Variant_pos"))
ptop.Total_paralog_annotations[is.na(ptop.Total_paralog_annotations)] = 0
```

Looking at alt alleles. Taking only pairwise alignments where the alt allele is conserved leaves only 1115 individual pairwise alignments. The number of actual unique variants this equates to is less - 825.

#### Paralogue stats
According to ensembl, 92096 protein coding genes are defined to have paralogues. While 7958 protein genes do not have paralogues


#### Plan for overall annotations table
REF              | All | Alt matches |Alt no match
-----------------|-----|-------------|-------------
                 |     |             |
No QC            |     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
Paralog Conserved|     |             |
                 |     |             |
-----------------|-----|-------------|-------------
                 |     |             |
All Conserved    |     |             |
                 |     |             |
                 
```{r include = TRUE, eval = FALSE}
Variant = c("Pathogenic variants","Benign variants","PPV","Sensitivity","P value")
Total = c("","","NA","NA","")
Paralogue_Annotation_no_QC = c("","","","","")
Variants_remaining_after_PA_QC1 = c("","","","","")
Variants_removed_after_PA_QC1 = c("","","","NA","")
Variants_remaining_after_PA_QC2 = c("","","","","")
Variants_removed_after_PA_QC2 = c("","","","NA","")
Variants_remaining_after_PA_QC3 = c("","","","","")
Variants_removed_after_PA_QC3 = c("","","","NA","")

Whole_clinvar_set = data.frame(Variant,
                               Total,
                               Paralogue_Annotation_no_QC,
                               Variants_remaining_after_PA_QC1,
                               Variants_removed_after_PA_QC1,
                               Variants_remaining_after_PA_QC2,
                               Variants_removed_after_PA_QC2,
                               Variants_remaining_after_PA_QC3,
                               Variants_removed_after_PA_QC3
                               )

```